---
title: "2018_seed_fungicide_16S_13021"
author: "Fang Liu"
date: "6/21/2019"
output:
  html_document: default
  pdf_document: default
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load required libraries to working environment

```{r echo=FALSE}
library(vegan)
library(phyloseq)
library(ggplot2)
library(reshape2)
library(dplyr)
library(plyr)
library(metagenomeSeq)
library(Hmisc)
library(directlabels)
library(stringr)
library(broom)
library(permute)
```

## Set working directory

```{r}
setwd('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021')
getwd()
```

-------------------------------------------------
Using 97% similarity based OTU clustering 
-------------------------------------------------

## Upload mothur .shared and cons.taxonomy file

```{r}
# Read in the original strigolactone output OTU table and taxonomy table for sequencing depth barplot

Seed_phyloseq<-import_mothur(mothur_shared_file = "/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Seed.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.shared",mothur_constaxonomy_file = "/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Seed.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.0.03.cons.taxonomy")

Seed_phyloseq #144821 taxa and 164 samples
min(sample_sums(Seed_phyloseq)) #252
max(sample_sums(Seed_phyloseq)) #488368
sum(sample_sums(Seed_phyloseq)) #13079622 reads remained 
sort(sample_sums(Seed_phyloseq))

# Read in the meta data
Seed_meta<-read.csv('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Seed_meta.csv',row.names = 1,header = TRUE)
head(Seed_meta)
identical(rownames(Seed_meta),sample_names(Seed_phyloseq))
# update meta data by adding in read depth data
Seed_meta_up<-Seed_meta
Seed_meta_up$Treatment<-factor(Seed_meta_up$Treatment,levels = c("CT","CM","EE"))
Seed_meta_up$Compartment<-factor(Seed_meta_up$Compartment,levels=c("Soil","Bulk","Rhizosphere","Endosphere","SEED"))
Seed_meta_up$Time<-factor(Seed_meta_up$Time,levels = c('Presowing','1wk','3wk','4wk','SEED'))
Seed_meta_up$Read_depth<-sample_sums(Seed_phyloseq)
Seed_meta_up[1:5,]

# Generate phyloseq object by merging Seed_phyloseq with sample data
# Here, before merge always check if the sample order in meta data match with that from phyloseq object
identical(rownames(Seed_meta_up),sample_names(Seed_phyloseq)) # TRUE
Seed_meta_up_phyloseq<-sample_data(Seed_meta_up)

#>>>>> Seed_phyloseq_up >>>>> with sample data integrated

Seed_phyloseq_up<-merge_phyloseq(Seed_phyloseq,Seed_meta_up_phyloseq)
Seed_phyloseq_up
colnames(tax_table(Seed_phyloseq_up))<-c('Kingdom','Phylum','Class','Order','Family','Genus')
sample_names(Seed_phyloseq_up)[1:5]
Seed_phyloseq_up # 144821 taxa and 164 samples

r_Seed_phyloseq_up<-transform_sample_counts(Seed_phyloseq_up,function(x) x/sum(x))

```

## Read depth barplot

```{r}
tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Read_depth_barplot.tiff",units = 'in',width = 16,height = 8,res = 300,bg='White',compression = 'lzw')

ggplot(data=Seed_meta_up,aes(x=Sample_ID,y=Read_depth))+geom_bar(width=0.7,stat = "identity") +xlab("Sample_ID")+ylab("Sequencing depth") +scale_y_continuous(labels = scales::comma)+theme(panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),text=element_text(family="Arial"), plot.title = element_text(family='Arial', hjust = .5,vjust = 3,size = 24,face="bold"), axis.text.x = element_text(family="Arial",size=8,angle=90,vjust = .6),axis.text.y = element_text(family="Arial",size=14),axis.title.x = element_text(family="Arial",size=20,face="bold"),axis.title.y = element_text(family="Arial",size=20,face="bold"),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

# reorder the Sample_ID based on Compartment and time
Seed_meta_up2<-Seed_meta_up[with(Seed_meta_up,order(Compartment, Time)),]
Seed_meta_up2$Sample_ID<-factor(Seed_meta_up2$Sample_ID,levels =as.character(Seed_meta_up2$Sample_ID))

tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Read_depth_barplot reordered based on Compartment and time.tiff",units = 'in',width = 16,height = 8,res = 300,bg='White',compression = 'lzw')

ggplot(data=Seed_meta_up2,aes(x=Sample_ID,y=Read_depth))+geom_bar(width=0.7,stat = "identity") +xlab("Sample_ID")+ylab("Sequencing depth") +
  scale_y_continuous(labels = scales::comma)+theme(panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),text=element_text(family="Arial"), plot.title = element_text(family='Arial', hjust = .5,vjust = 3,size = 24,face="bold"), axis.text.x = element_text(family="Arial",size=8,angle=90,vjust = .6),axis.text.y = element_text(family="Arial",size=14),axis.title.x = element_text(family="Arial",size=20,face="bold"),axis.title.y = element_text(family="Arial",size=20,face="bold"),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

##PCoA including SEED samples

set.seed(1013)
adonis2(t(otu_table(r_Seed_phyloseq_up))~Compartment,data=data.frame(sample_data(r_Seed_phyloseq_up)),permutations=999,by="margin") 

#PCoA plot between compartments including SEED

# Define compartment colors
color_compartment<-c("#CD9BCD","#C84248","orange","#00CDCD","#ABAD29")
label_compartment<-factor(levels(Seed_meta_up$Compartment),levels = c("Soil","Bulk","Rhizosphere","Endosphere","SEED"))
#tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Compartment_corlor_pie.tiff",units = 'in',width=10,height = 8,res=300,compression = 'lzw')
pie(rep (1,5),col=color_compartment,labels=paste(label_compartment,color_compartment,sep=":"),radius = 1.0)
#dev.off()


Seed_up_bray<-vegdist(t(otu_table(r_Seed_phyloseq_up)),method="bray",binary = FALSE)
Seed_up_PCoA<-ordinate(r_Seed_phyloseq_up,method = "PCoA",Seed_up_bray)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA across all samples between compartments.tiff', units="in", width=7, height=5, res=300)

plot_ordination(r_Seed_phyloseq_up,Seed_up_PCoA,type="samples",color="Compartment")+labs(title="PCoA cross all samples ",x="PCoA1 [34.2%]",y="PCoA2 [13.5%]")+scale_color_manual(values =c("#CD9BCD","#C84248","orange","#00CDCD","#ABAD29") )+
  geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title.x=element_text(size = 15,face="bold",family = "Arial",margin = margin(t=5,r=0,b=0,l=0,unit = 'pt') ),axis.title.y=element_text(size = 15,face="bold",family = "Arial",margin =  margin(t=0,r=5,b=0,l=0,unit = 'pt')),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

## PERMANOVA test

#-- Compartment vs read_depth

set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = paste(data.frame(sample_data(r_Seed_phyloseq_up))$Plot,data.frame(sample_data(r_Seed_phyloseq_up))$Time,sep = "_")),within=Within(type='free'))
adonis2(t(otu_table(r_Seed_phyloseq_up))~Read_depth+Compartment,data=data.frame(sample_data(r_Seed_phyloseq_up)),permutations=999,by="margin")

# -- Compartment only---

set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = paste(data.frame(sample_data(r_Seed_phyloseq_up))$Plot,data.frame(sample_data(r_Seed_phyloseq_up))$Time,sep = "_")),within=Within(type='free'))
adonis2(t(otu_table(r_Seed_phyloseq_up))~Compartment,data=data.frame(sample_data(r_Seed_phyloseq_up)),permutations=999,by="margin")

```

## >>>>> Seed_13021>>>>>>>>
## rarefied at 13021 and remove singletons: due the low sequencing depth, all the seed samples were removed

```{r}
Seed_13021<-import_mothur(mothur_shared_file = "/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Seed.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.0.03.subsample.0.03.pick.shared",mothur_constaxonomy_file = "/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Seed.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.0.03.subsample.0.03.pick.0.03.cons.taxonomy")
Seed_13021 # 21355 taxa and 150 samples
sum(sample_sums(Seed_13021))
sample_sums(Seed_13021)
Seed_13021_up<-merge_phyloseq(Seed_13021,Seed_meta_up_phyloseq)
Seed_13021_up # 21355 taxa and 150 samples
colnames(tax_table(Seed_13021_up))<-c('Kingdom','Phylum','Class','Order','Family','Genus')
# Here to remove the sample - Seed_CM_4
Seed_13021_up<-prune_samples(sample_names(Seed_13021_up)[1:149],Seed_13021_up)
```

## Diversity

```{r}
Seed_13021_up
Diversity<-data.frame(estimate_richness(Seed_13021_up,split=TRUE,measures = NULL),sample_data(Seed_13021_up))
dim(Diversity)
Diversity[1:5,]
identical(substring((rownames(Diversity)),2),sample_names(Seed_13021_up))
rownames(Diversity)<-sample_names(Seed_13021_up)
  
#system("sed 's/X//g' file")
# Define color for treatment
Diversity$Treat<-factor(Diversity$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','Seed'))

Treat_checklist<-data.frame(Treat_colors=c('#9b9696','#b6cc0e','#ed5567','#07aeba','#3a44ff','#f936f6','#723434','#316022','#05fff2','#b105fc'),Treat_labels=c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk'))
Treat_colors<-factor(Treat_checklist$Treat_colors,levels = as.character(Treat_checklist$Treat_colors))
Treat_labels<-factor(Treat_checklist$Treat_labels,levels = as.character(Treat_checklist$Treat_labels))

tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Treat_colour_pie.tiff",units = 'in',width=10,height = 8,res=300,compression = 'lzw')
pie(rep (1,10),col=as.character(Treat_colors),labels=paste(as.character(Treat_labels),as.character(Treat_colors),sep=":"),radius = 1.0)
dev.off()

## Shannon diversity ---- between treats

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Shannon diversity between treats.tiff', units="in", width=7, height=5, res=300)

boxplot(Shannon~Treat, data=Diversity, main="Diversity between treats") 

ggplot(Diversity,aes(x=Treat,y=Shannon,fill=Treat))+geom_boxplot(size=0.5)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = as.character(Treat_colors),name="Treat")+labs(title="Shannon Diversity",x="Treatment",y="Shannon Index")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=17,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## ANOVA

# First, test the normality of dependence
res<-resid(lm(Shannon~Treat,data=Diversity))
shapiro.test(res) # p-value = 5.35e-13
plot(density(Diversity$Shannon))

# Pairwise comparison
kruskal.test(Shannon~Treat, data=Diversity)
dunn.test::dunn.test(x = Diversity$Shannon,g = Diversity$Treat,method ='bh',kw = TRUE,label = TRUE,wrap = TRUE,table = TRUE,alpha = 0.1)

# ------ between compartments----

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Shannon diversity between compartments.tiff', units="in", width=6, height=4, res=300)

boxplot(Shannon~Compartment, data=Diversity, main="Diversity between compartments") 

ggplot(Diversity,aes(x=Compartment,y=Shannon,fill=Compartment))+geom_boxplot(size=0.5)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Compartment))+scale_fill_manual(values = as.character(Treat_colors),name="Treat")+labs(title="Shannon Diversity",x="Treatment",y="Shannon Index")+ylim(c(2.5,7.5))+scale_x_discrete(labels=c("Soil" = "Soil", "Bulk" = "Bulk",
                              "Rhizosphere" = "Rhi",'Endosphere'="Endo",'SEED'='SEED'))+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=17,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## ANOVA

# First, test the normality of dependence
res<-resid(lm(Shannon~Compartment,data=Diversity))
shapiro.test(res) # p-value = 5.35e-13
plot(density(Diversity$Shannon))

# Pairwise comparison
kruskal.test(Shannon~Compartment, data=Diversity)
dunn.test::dunn.test(x = Diversity$Shannon,g = Diversity$Compartment,method ='bh',kw = TRUE,label = TRUE,wrap = TRUE,table = TRUE,alpha = 0.1)

## Here, the diversity change could be interpreted as below:

#In the rhizosphere at week 1, some competive bacteria taxa kind take advangage of the nutrients or resources probobaly opportunists. Then other taxas comes after and become competive when the rhizosphere nutrients becomes less abundant. Or when cross-feed interactions happens. However, inside of the endosphere, diverse bacteria kind of colonize soybean root during its germination maybe in the first week, then those adapted to the microenvironment within the plant started to be dominant and stable. 

## Test fungicide treatment impacts
```

## Rarefaction plot at maximum sequencing depth

```{r}
## >>>>>> at maximum depth >>>

# This rarefaction table is generated from mothur using raw OTU table and rarefied with interval of 100
rarefaction_df<-read.csv('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Seed.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.groups.rarefaction.csv',header = TRUE)
head(rarefaction_df)
rarefaction_df<-rarefaction_df[c(1,which(rarefaction_df$numsampled%%100==0)),]
rarefaction_df_up<-rarefaction_df[,-c(grep("lci|hci",colnames(rarefaction_df),perl=TRUE))]
head(rarefaction_df_up)
# To change treatment name to regular expression instead of X0.03.CT_1
colnames<-gsub('X0.03.','',colnames(rarefaction_df_up))
colnames(rarefaction_df_up)<-colnames
head(rarefaction_df_up)
rarefaction_df_up<-rarefaction_df_up[,c('numsampled',as.character(sample_data(Seed_phyloseq_up)$Sample_ID))]
head(rarefaction_df_up)  
  
melt_cultivar_otu_rarefy_max<-melt(rarefaction_df_up,id.vars = 'numsampled')
head(melt_cultivar_otu_rarefy_max)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Rarefaction plot with maximum depth.tiff', units="in", width=16, height=8, res=600)
ggplot(melt_cultivar_otu_rarefy_max,aes(x=numsampled,y=value,colour=variable))+geom_line(aes(x=numsampled,y=value,group=variable),size=0.4)+geom_dl(aes(label = variable), method = list(dl.combine("last.points"), cex = 0.2))+ylab("OTU numbers")+xlab("Sequencing depth")+theme(panel.background = element_rect(fill="white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(), title=(element_text(size=15,family="Arial",face="bold")),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family="Arial"),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=5,family="Arial")),legend.position="none")+ggtitle("Rarefaction curve with maximum read depth")
dev.off()

## >>>>>> at minimum depth >>>

# This rarefaction table is generated from mothur using raw OTU table and rarefied with interval of 100
rarefaction_df<-read.csv('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/minimum_depth.rarefaction.csv',header = TRUE)
head(rarefaction_df)
rarefaction_df<-rarefaction_df[c(1,which(rarefaction_df$numsampled%%100==0)),]
rarefaction_df_up<-rarefaction_df[,-c(grep("lci|hci",colnames(rarefaction_df),perl=TRUE))]
head(rarefaction_df_up)
# To change treatment name to regular expression instead of X0.03.CT_1
colnames<-gsub('X0.03.','',colnames(rarefaction_df_up))
colnames(rarefaction_df_up)<-colnames
head(rarefaction_df_up)
rarefaction_df_up<-rarefaction_df_up[,c('numsampled',as.character(sample_names(Seed_13021)))]
rarefaction_df_up[1:5,1:5]
  
melt_cultivar_otu_rarefy_max<-melt(rarefaction_df_up,id.vars = 'numsampled')
head(melt_cultivar_otu_rarefy_max)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Rarefaction plot with minimum depth.tiff', units="in", width=16, height=8, res=600)
ggplot(melt_cultivar_otu_rarefy_max,aes(x=numsampled,y=value,colour=variable))+geom_line(aes(x=numsampled,y=value,group=variable),size=0.4)+
geom_dl(aes(label = variable), method = list(dl.combine("last.points"), cex = 0.2))+ylab("OTU numbers")+xlab("Sequencing depth")+theme(panel.background = element_rect(fill="white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(), title=(element_text(size=15,family="Arial",face="bold")),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family="Arial"),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=5,family="Arial")),legend.position="none")+ggtitle("Rarefaction curve with maximum read depth")
dev.off()

```


## stacked barplot with all samples 

```{r}
phylum_Seed_13021<-tax_glom(Seed_13021_up,taxrank = rank_names(Seed_13021_up)[2],NArm = TRUE)
phylum_Seed_13021 #39 taxa and 149 samples

r_phylum_Seed_13021<-transform_sample_counts(phylum_Seed_13021 ,function(x) x/sum(x))
filter_r_phylum_Seed_13021<- filter_taxa(r_phylum_Seed_13021, function(x) sum(x>0.01)>0.2*length(x),prune = TRUE)
filter_r_phylum_Seed_13021 # got 14 phyla across 149 sampels left
sample_sums(filter_r_phylum_Seed_13021)
tax_table(filter_r_phylum_Seed_13021)[,2]
Others<-1-sample_sums(filter_r_phylum_Seed_13021)
phylum_seed<-rbind(otu_table(filter_r_phylum_Seed_13021),Others)
phylum_seed<-as.data.frame(phylum_seed)
phylum_seed[,1:5]
rownames(phylum_seed)<-c(tax_table(filter_r_phylum_Seed_13021)[,2],"Others")
phylum_seed
#phylum_rarefied_ST<-data.frame(Treat=as.factor(sample_data(r_phylum_Seed_13021)$Treatment),t(phylum_rarefied_ST))
phylum_seed[,1:5]
#phylum_seed<-phylum_seed%>% group_by(Sample_ID) %>%
#  summarise_all(funs(mean))
phylum_Seed_t<-as.data.frame(t(phylum_seed))


phylum_Seed_t %>%
  mutate(dominant=phylum_Seed_t%>%
           select(c(Proteobacteria,Acidobacteria,Bacteroidetes,Actinobacteria,Verrucomicrobia,Planctomycetes))%>%
           rowSums()) %>% select(dominant)

rowSums(phylum_Seed_t[,1:dim(phylum_Seed_t)[2]])
phylum_Seed_t[1:5,1:5]
phylum_Seed_t$Sample_ID<-rownames(phylum_Seed_t)
identical(phylum_Seed_t$Sample_ID,as.character(sample_data(Seed_13021_up)$Sample_ID))
phylum_Seed_t$Compartment<-sample_data(Seed_13021_up)$Compartment
phylum_Seed_t$Time<-sample_data(Seed_13021_up)$Time
phylum_Seed_t[1:5,]
phylum_Seed_t_melt<-melt(phylum_Seed_t,id.vars = c("Sample_ID","Compartment","Time"))
dim(phylum_Seed_t_melt)
head(phylum_Seed_t_melt)
colnames(phylum_Seed_t_melt)<-c('Sample_ID','Compartment','Time','Phylum','Relative_abundance')
phylum_Seed_t_melt[1:5,]

# Define Phylum colors

Bacteria_phylum_vs_color_checklist<-data.frame(phylum=c('Proteobacteria','Actinobacteria','Bacteria_unclassified','Acidobacteria','Nitrospirae','Chloroflexi','Candidatus_Saccharibacteria','Firmicutes','Bacteroidetes','Verrucomicrobia','candidate_division_WPS-1','Gemmatimonadetes','Planctomycetes','BRC1','Latescibacteria','Chlamydiae','Others','Patescibacteria','Rokubacteria','Cyanobacteria'),colors=c('#ff9082','#98d66d','#fcef5d','#c45cc0','#45b1f9','#f76fc3','#85c1b8','#a584ff','#ffb444','#7ebfe5','#cec0c9','#467584','#005ff9','#bc8c38','#bcba6d','#91749e','#b2798d','#343837','#b23301','#235931'))
phylum_colors<-as.character(Bacteria_phylum_vs_color_checklist$colors)
phylum_labels<-as.character(Bacteria_phylum_vs_color_checklist$phylum)

    
tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Phylum_colour_pie.tiff",units = 'in',width=10,height = 8,res=300,compression = 'lzw')
pie(rep (1,20),col=phylum_colors,labels=paste(phylum_labels,phylum_colors,sep=":"),radius = 1.0)
dev.off()

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Phylum level stacked barplot using rarefied_ST.tiff', units="in", width=15, height=5, res=300)

phylum_Seed_t_melt$Phylum<-factor(phylum_Seed_t_melt$Phylum,levels=phylum_labels[match(c(as.character(unique(phylum_Seed_t_melt$Phylum))),phylum_labels)])

ggplot(phylum_Seed_t_melt, aes(x =Sample_ID , y = Relative_abundance,fill=Phylum))+geom_bar(stat='identity',size=0.1)+labs(x="Treatment",y="Relative Abundance")+scale_fill_manual(values=phylum_colors[match(levels(phylum_Seed_t_melt$Phylum),phylum_labels)])+theme(panel.background = element_rect(fill='white',color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",hjust = 0.5,vjust = 3)),axis.text.x = element_text(size = 6,family="Arial"),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=6,family="Arial",angle = 90),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Phylum stacked barplot with reordered sample_ID

phylum_Seed_t2<-phylum_Seed_t[with(phylum_Seed_t,order(Compartment,Time)),]
phylum_Seed_t2$Sample_ID<-factor(phylum_Seed_t2$Sample_ID,levels = as.character(phylum_Seed_t2$Sample_ID))
phylum_Seed_t_melt2<-melt(phylum_Seed_t2,id.vars = c("Sample_ID","Compartment","Time"))
dim(phylum_Seed_t_melt2)
head(phylum_Seed_t_melt2)
colnames(phylum_Seed_t_melt2)<-c('Sample_ID','Compartment','Time','Phylum','Relative_abundance')
dim(phylum_Seed_t_melt2)
phylum_Seed_t_melt2[1:5,]

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Phylum level stacked barplot using rarefied_ST_reordered_sampleID1.tiff', units="in", width=15, height=5, res=300)

phylum_Seed_t_melt2$Phylum<-factor(phylum_Seed_t_melt2$Phylum,levels=phylum_labels[match(c(as.character(unique(phylum_Seed_t_melt2$Phylum))),phylum_labels)])

ggplot(phylum_Seed_t_melt2, aes(x =Sample_ID , y = Relative_abundance,fill=Phylum))+geom_bar(stat='identity',size=0.1)+labs(x="Treatment",y="Relative Abundance")+scale_fill_manual(values=phylum_colors[match(c(as.character(unique(phylum_Seed_t_melt2$Phylum))),phylum_labels)])+theme(panel.background = element_rect(fill='white',color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",hjust = 0.5,vjust = 3)),axis.text.x = element_text(size = 6,family="Arial"),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=6,family="Arial",angle = 90),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Barplot after grouping based on Treat

filter_r_phylum_Seed_13021
Others<-1-sample_sums(filter_r_phylum_Seed_13021)
phylum_seed<-rbind(otu_table(filter_r_phylum_Seed_13021),Others)
phylum_seed[,1:5]
rownames(phylum_seed)<-c(tax_table(filter_r_phylum_Seed_13021)[,2],"Others")
phylum_seed
identical(rownames(t(phylum_seed)),sample_names(filter_r_phylum_Seed_13021)) # TRUE
phylum_group_seed<-data.frame(Treat=sample_data(filter_r_phylum_Seed_13021)$Treat,t(phylum_seed))
phylum_group_seed<-phylum_group_seed%>% group_by(Treat) %>%
  summarise_all(funs(mean))
phylum_group_seed
rowSums(phylum_group_seed[,2:16])
phylum_group_Seed_melt<-melt(phylum_group_seed,id.vars = "Treat")
colnames(phylum_group_Seed_melt)<-c('Treat','Phylum','Relative_abundance')
phylum_group_Seed_melt

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Phylum level stacked barplot grouped.tiff', units="in", width=7, height=5, res=300)

phylum_group_Seed_melt$Phylum<-factor(phylum_group_Seed_melt$Phylum,levels=phylum_labels[match(c(as.character(unique(phylum_group_Seed_melt$Phylum))),phylum_labels)])

phylum_group_Seed_melt$Treat<-factor(phylum_group_Seed_melt$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','Seed'))

ggplot(phylum_group_Seed_melt, aes(x =Treat , y = Relative_abundance,fill=Phylum))+geom_bar(stat='identity',size=0.1)+labs(x="Treatment",y="Relative Abundance")+scale_fill_manual(values=phylum_colors[match(levels(phylum_Seed_t_melt$Phylum),phylum_labels)])+theme(panel.background = element_rect(fill='white',color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",hjust = 0.5,vjust = 3)),axis.text.x = element_text(size = 12,family="Arial",angle = 90),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()
```


## ----- PERMANOVA test ------- 

```{r}
# >>>> Read_depth >>>>>

r_Seed_13021_up<-transform_sample_counts(Seed_13021_up,function(x) x/sum(x))
set.seed(1013)
adonis2(t(otu_table(r_Seed_13021_up))~Read_depth,data=data.frame(sample_data(r_Seed_13021_up)),permutations=999,by="margin") 

#            Df SumOfSqs      R2      F Pr(>F)  
#Read_depth   1    0.478 0.01323 1.9714  0.067 .
#Residual   147   35.644 0.98677                
#Total      148   36.122 1.00000

# >>>> Compartment >>>>>
set.seed(1013)
adonis2(t(otu_table(r_Seed_13021_up))~Compartment,data=data.frame(sample_data(r_Seed_13021_up)),permutations=999,by="margin")
#adonis2(formula = t(otu_table(r_Seed_13021_up)) ~ Compartment, data = data.frame(sample_data(r_Seed_13021_up)), permutations = 999, by = "margin")
#             Df SumOfSqs      R2      F Pr(>F)    
#Compartment   3   16.759 0.46396 41.834  0.001 ***
#Residual    145   19.363 0.53604                  
#Total       148   36.122 1.00000                  

# >>>> Compartment vs Read_depth >>>>>
set.seed(1013)
adonis2(t(otu_table(r_Seed_13021_up))~Compartment+Read_depth,data=data.frame(sample_data(r_Seed_13021_up)),permutations=999,by="margin")

#adonis2(formula = t(otu_table(r_Seed_13021_up)) ~ Compartment + Read_depth, data = data.frame(sample_data(r_Seed_13021_up)), permutations = 999, by = "margin")
#             Df SumOfSqs      R2       F Pr(>F)    
#Compartment   3   16.389 0.45372 40.8571  0.001 ***
#Read_depth    1    0.108 0.00299  0.8089  0.524    
#Residual    144   19.255 0.53304                   
#Total       148   36.122 1.00000
```

# >>>>> PCoA before subset >>>> using r_Seed_13021_up

```{r}
# >>> Treatment_vs_Time >>>

dim(otu_table(r_Seed_13021_up)) # 21355   149
Seed_13021_up_bray<-vegdist(t(otu_table(r_Seed_13021_up)),method="bray",binary = FALSE)
Seed_13021_up_PCoA<-ordinate(r_Seed_13021_up,method = "PCoA", Seed_13021_up_bray)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Seed_13021_up_Treatment_vs_Time.tiff', units="in", width=7, height=5, res=300)

plot_ordination(Seed_13021_up,Seed_13021_up_PCoA,type="samples",shape="Time",color="Treatment")+labs(title="PCoA Seed_13021_up_Treatment_vs_Time",x="PCoA1 [39.8%]",y="PCoA2 [13.4%]")+geom_point(size=2)+scale_shape_manual(values = c(0,1,2,4,8))+scale_color_manual(values = c("#84a8a4","#c143a2","#e59112"))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
#p$layers<-p$layers[-1]
#p
dev.off()

      # >>> Define Plots cplors >>>
      
      color_plot<-c("#8569D5","#00CDCD","orange","#DA5724","#508578","#CD9BCD","#AD6F3B","#673770","#D14285","#652926","#C84248","#CBD588","#D3D3ED","#BAFFE2","#000099","grey")
      label_plot<-factor(unique(sample_data(Seed_13021_up)$Plot),levels = unique(sample_data(Seed_13021_up)$Plot))
      tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Plot_corlor_pie.tiff",units = 'in',width=10,height = 8,res=300,compression = 'lzw')
      pie(rep (1,16),col=color_plot,labels=paste(label_plot,color_plot,sep=":"),radius = 1.0)
      dev.off()

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Seed_13021_up_between_Plots.tiff', units="in", width=7, height=5, res=300)

plot_ordination(Seed_13021_up,Seed_13021_up_PCoA,type="samples",color="Plot")+labs(title="PCoA Seed_13021_up_between_Plots",x="PCoA1 [39.8%]",y="PCoA2 [13.4%]")+geom_point(size=2)+scale_color_manual(values = c("#8569D5","#00CDCD","orange","#DA5724","#508578","#CD9BCD","#AD6F3B","#673770","#D14285","#652926","#C84248","#CBD588","#D3D3ED","#BAFFE2","#000099","grey"))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()


# >>> Compartment vs Treatment >>>

# First, I want to define the colors for different compartment to make sure both my ITS and 16S data use the same annotations

# Define compartment colors
color_compartment<-c("#CD9BCD","#C84248","orange","#00CDCD","#ABAD29")
label_compartment<-factor(levels(Seed_meta_up$Compartment),levels = c("Soil","Bulk","Rhizosphere","Endosphere","Seed"))
tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Compartment_corlor_pie.tiff",units = 'in',width=10,height = 8,res=300,compression = 'lzw')
pie(rep (1,5),col=color_compartment,labels=paste(label_compartment,color_compartment,sep=":"),radius = 1.0)
dev.off()

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Seed_13021_up_Compartment vs Treatment.tiff', units="in", width=7, height=5, res=300)

plot_ordination(Seed_13021_up,Seed_13021_up_PCoA,type="samples",shape="Treatment",color="Compartment")+labs(title="PCoA Seed_13021_up_Compartment vs Treatment",x="PCoA1 [39.8%]",y="PCoA2 [13.4%]")+scale_color_manual(values =c("#CD9BCD","#C84248","orange","#00CDCD","#ABAD29") )+
  geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()


# >>> Compartment vs Time >>>

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Seed_13021_up_Compartment vs Time.tiff', units="in", width=7, height=5, res=300)

plot_ordination(Seed_13021_up,Seed_13021_up_PCoA,type="samples",color="Compartment",shape="Time",)+labs(title="PCoA Seed_13021_up_Compartment vs Time",x="PCoA1 [39.8%]",y="PCoA2 [13.4%]")+geom_point(size=2)+scale_color_manual(values = c("#CD9BCD","#C84248","orange","#00CDCD","#ABAD29"))+scale_shape_manual(values = c(0,1,2,4,8))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
#p$layers<-p$layers[-1]
#p
dev.off()
```

## Subset phyloseq to Soil- this test turned out that no significant difference between indigenous bacteria community between plots that were assinged to CT, CM and EE 

```{r}
## Soil samples

Soil<-subset_samples(Seed_13021_up,Compartment=="Soil")
Soil #21355 taxa and 15 samples

Soil_up<-filter_taxa(Soil, function(x) sum(x)>1,prune = TRUE)
Soil_up #7796 taxa and 15 samples

r_Soil_up<-transform_sample_counts(Soil_up,function(x) x/sum(x))

# PERMANOVA 

## Treatment impact---  no significant difference between treatment.

set.seed(1013)
adonis2(t(otu_table(r_Soil_up))~Treatment,data=data.frame(sample_data(r_Soil_up)),permutations=999,by="margin") 

## Read_depth impact check -- insignificant

set.seed(1013)
adonis2(t(otu_table(r_Soil_up))~Read_depth,data=data.frame(sample_data(r_Soil_up)),permutations=999,by="margin")

#adonis2(formula = t(otu_table(r_Soil_up)) ~ Read_depth, data = data.frame(sample_data(r_Soil_up)), permutations = 999, by = "margin")
#           Df SumOfSqs      R2      F Pr(>F)
#Read_depth  1  0.07297 0.05986 0.8277  0.734
#Residual   13  1.14612 0.94014              
#Total      14  1.21909 1.00000

# PCoA plot
Soil_up_bray<-vegdist(t(otu_table(Soil_up)),method = 'bray',binary=FALSE)

r_Soil_up_bray<-vegdist(t(otu_table(r_Soil_up)),method="bray",binary = FALSE)
r_Soil_up_PCoA<-ordinate(r_Soil_up,method = "PCoA", r_Soil_up_bray)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_Soil_up_Treatment.tiff', units="in", width=7, height=5, res=300)

plot_ordination(r_Soil_up,r_Soil_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="PCoA r_Soil_up_Treatment",x="PCoA1 [20.1%]",y="PCoA2 [14.9%]")+
  geom_point(size=2)+ theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()
```


##SB_CT

```{r}
SB_CT<-subset_samples(Seed_13021_up,(Compartment=="Soil"&Treatment=="CT")|(Treatment=="CT"&Compartment=="Bulk"))
sample_names(SB_CT)
SB_CT #21355 taxa and 20 samples

SB_CT_up<-filter_taxa(SB_CT,function(x) sum(x)>1,prune = TRUE)
SB_CT_up # 9403 taxa and 20 samples

r_SB_CT_up<-transform_sample_counts(SB_CT_up,function(x) x/sum(x))

#PERMANOVA

#------- Time ---- significant and R2=23.85%
set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_SB_CT_up))$Plot),
          within=Within(type='free'))
adonis2(t(otu_table(r_SB_CT_up))~Time,data=data.frame(sample_data(r_SB_CT_up)),permutations=perm,by="margin") 

#adonis2(formula = t(otu_table(r_SB_CT_up)) ~ Time, data = data.frame(sample_data(r_SB_CT_up)), permutations = perm, by = "margin")
#         Df SumOfSqs      R2      F Pr(>F)    
#Time      3  0.42175 0.23849 1.6703  0.001 ***
#Residual 16  1.34668 0.76151                  
#Total    19  1.76842 1.00000

# --- Read_depth impact ---

set.seed(1013)
adonis2(t(otu_table(r_SB_CT_up))~Read_depth,data=data.frame(sample_data(r_SB_CT_up)),permutations=999,by="margin") 

## PCoA plot

SB_CT_up_bray<-vegdist(t(otu_table(r_SB_CT_up)),method="bray",binary = FALSE)
SB_CT_up_PCoA<-ordinate(r_SB_CT_up,method = "PCoA",SB_CT_up_bray)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Bulk 1wk 3wk 4wk and presowing in CT plots.tiff', units="in", width=7, height=6, res=300)

plot_ordination(r_SB_CT_up,SB_CT_up_PCoA,type="samples",color="Time",shape = "Treatment")+labs(title="SB_CT_13021_up_Time",x="PCoA1 [18%]",y="PCoA2 [11.5%]")+geom_point(size=2)+scale_shape_manual(values = c(0))+scale_color_manual(values=c("#456780","#b04f13","#a037ad","#fa667a"))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=10,face="bold",family = "Arial"),legend.text = (element_text(size=8,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,2,2,1),"cm"))
dev.off()
```

# As bacterial community composition significant different between compartment, the dataset will be subsetted to specific compartment and then analyze Plot, Time and treatment impacts

#>>>>> Before subset samples to different compartment, test if bulk soil and pre-planting soil significantly different in terms of bacterial community composition

```{r}
BS_13021 <-subset_samples(Seed_13021_up,Compartment=="Soil"|Compartment=="Bulk")
BS_13021 # 21355 taxa and 60 samples
BS_13021_up<-filter_taxa(BS_13021,function(x) sum(x)>1,prune = TRUE)
BS_13021_up # 16148 taxa and 60 samples
r_BS_13021_up<-transform_sample_counts(BS_13021_up,function(x) x/sum(x))

# PERMANOVA test between bulk soil and pre-planting soil

# --- Plot --- signifcant and could explain 44.06% of variations
set.seed(1013)  
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_BS_13021_up))$Time),
          within=Within(type='free'))
adonis2(t(otu_table(r_BS_13021_up))~Plot,data=data.frame(sample_data(r_BS_13021_up )),permutations=999,by="margin")

#         Df SumOfSqs      R2      F Pr(>F)    
#Plot     14   2.2565 0.44059 2.4753  0.001 ***
#Residual 44   2.8650 0.55941                  
#Total    58   5.1215 1.00000

# -----Time ----here time factor mixed with compartment impact as pre_sow time cource are represented by all soil compartment
set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_BS_13021_up))$Plot),
          within=Within(type='free'))
adonis2(t(otu_table(r_BS_13021_up))~Time,data=data.frame(sample_data(r_BS_13021_up)),permutations=perm,by="margin")

# Plot vs Time
set.seed(1013)
adonis2(t(otu_table(r_BS_13021_up))~Plot+Time,data=data.frame(sample_data(r_BS_13021_up )),permutations=999,by="margin")

#adonis2(formula = t(otu_table(r_BS_13021_up)) ~ Plot + Time, data = data.frame(sample_data(r_BS_13021_up)), permutations = 999, by = "margin")
#         Df SumOfSqs      R2      F Pr(>F)    
#Plot     14   2.2592 0.43990 3.0181  0.001 ***
#Time      3   0.6803 0.13247 4.2412  0.001 ***
#Residual 41   2.1922 0.42685                  
#Total    58   5.1357 1.00000

# OKay, since bulk soil and pre-plant soil bacterial community composition indicated to be significantly different, I will seperate these two parts intead of combine them together.

# PCoA of BS_13021_up

BS_13021_up_bray<-vegdist(t(otu_table(r_BS_13021_up)),method="bray",binary = FALSE)
BS_13021_up_PCoA<-ordinate(r_BS_13021_up,method = "PCoA",BS_13021_up_bray)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA BS_13021_up_Bulk_vs_PreSoil between plots.tiff', units="in", width=7, height=6, res=300)

plot_ordination(r_BS_13021_up,BS_13021_up_PCoA,type="samples",shape="Time",color="Plot")+labs(title="BS_13021_up_Plots_vs_Compartment",x="PCoA1 [14.8%]",y="PCoA2 [7.8%]")+geom_point(size=2)+scale_shape_manual(values = c(0,1,2,4))+scale_color_manual(values =c("#8569D5","#00CDCD","orange","#DA5724","#508578","#CD9BCD","#AD6F3B","#673770","#D14285","#652926","#C84248","#CBD588","#D3D3ED","#BAFFE2","#000099","grey"))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=10,face="bold",family = "Arial"),legend.text = (element_text(size=8,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,2,2,1),"cm"))
dev.off()
```

## BRE

```{r}
BRE<-subset_samples(Seed_13021_up,Compartment=="Bulk"|Compartment=="Rhizosphere"|Compartment=="Endosphere")
BRE #21355 taxa and 134 samples 

BRE_up<-filter_taxa(BRE,function(x) sum(x)>1,prune = TRUE)
BRE_up #19681 taxa and 134 samples

r_BRE_up<-transform_sample_counts(BRE_up,function(x) x/sum(x))

## Read depth impact - turned to be insignificant
set.seed(1013) 
adonis2(t(otu_table(r_BRE_up))~Read_depth,data=data.frame(sample_data(r_BRE_up)),permutations=999,by="margin")

## Compartment impacts -- need to subset based on collection time and strata set to plot
set.seed(1013) 
perm<-how(nperm=999,plots = Plots(strata = paste(data.frame(sample_data(r_BRE_up))$Time,data.frame(sample_data(r_BRE_up))$Plot,sep = "_")),within=Within(type='free'))
adonis2(t(otu_table(r_BRE_up))~Compartment,data=data.frame(sample_data(r_BRE_up)),permutations=perm,by="margin")

## Treatment impact

# When I set Time and compartment combination as strata, it means that I will randomize samples within 1wk, 3wk and 4wk but not between them.
set.seed(1013) 
perm<-how(nperm=999,plots = Plots(strata = paste(data.frame(sample_data(r_BRE_up))$Compartment,data.frame(sample_data(r_BRE_up))$Time,sep = "_")),within=Within(type='free'))
adonis2(t(otu_table(r_BRE_up))~Treatment,data=data.frame(sample_data(r_BRE_up)),permutations=perm,by="margin")

# Do not constrains the permutation matrix, treatment impact is nonsignificant
set.seed(1013) 
adonis2(t(otu_table(r_BRE_up))~Treatment,data=data.frame(sample_data(r_BRE_up)),permutations=999,by="margin")


## Plot impact

# When I set Time as strata, it means that I will randomize samples within 1wk, 3wk and 4wk but not between them.

set.seed(1013) 
perm<-how(nperm=999,plots = Plots(strata = paste(data.frame(sample_data(r_BRE_up))$Compartment,data.frame(sample_data(r_BRE_up))$Time,sep = "_")),within=Within(type='free'))
adonis2(t(otu_table(r_BRE_up))~Plot,data=data.frame(sample_data(r_BRE_up)),permutations=perm,by="margin")

## Time impacts -- I should subset the data to specific compartment
set.seed(1013) 
perm<-how(nperm=999,plots = Plots(strata = paste(data.frame(sample_data(r_BRE_up))$Compartment,data.frame(sample_data(r_BRE_up))$Plot,sep = "_" )),
          within=Within(type='free'))
adonis2(t(otu_table(r_BRE_up))~Time,data=data.frame(sample_data(r_BRE_up)),permutations=perm,by="margin")

## Compartment vs Treatment vs Time impacts vs Read_depth

set.seed(1013) 
adonis(t(otu_table(r_BRE_up))~Compartment+Treatment+Time+Read_depth,data=data.frame(sample_data(r_BRE_up)),permutations=999,by="margin")

## Compartment vs Plot vs Time impacts  vs Read_depth

set.seed(1013) 
adonis(t(otu_table(r_BRE_up))~Compartment+Plot+Time+Read_depth,data=data.frame(sample_data(r_BRE_up)),permutations=999,by="margin")

## PCoA plot

# Compartment vs Time

r_BRE_up_bray<-vegdist(t(otu_table(r_BRE_up)),method = "bray",binary = FALSE)
r_BRE_up_PCoA<-ordinate(r_BRE_up,method = "PCoA",r_BRE_up_bray)
  

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_BRE_up Compartment vs Time.tiff', units="in", width=7, height=5, res=300)

plot_ordination(r_BRE_up,r_BRE_up_PCoA,type="samples",color="Compartment",shape = 'Time')+labs(title="PCoA r_BRE_up Compartment vs Time",x="PCoA1 [39.7%]",y="PCoA2 [14.3%]")+geom_point(size=2)+scale_shape_manual(values = c(0,1,2,4))+scale_color_manual(values = c("#CD9BCD","#C84248","orange","#00CDCD","#ABAD29"))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()


## Compartment vs Treatment

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_BRE Compartment vs Treatment.tiff', units="in", width=6.5, height=4.5, res=300)

plot_ordination(r_BRE_up,r_BRE_up_PCoA,type="samples",shape="Treatment",color="Compartment")+labs(title="PCoA norm_BRE Compartment vs Treatment",x="PCoA1 [39.7%]",y="PCoA2 [14.3%]")+geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+scale_color_manual(values = c("#CD9BCD","#C84248","orange","#00CDCD","#ABAD29"))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

## Treatment vs Time

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_BRE_up Treatmen vs Time.tiff', units="in", width=6.5, height=4.5, res=300)

plot_ordination(r_BRE_up,r_BRE_up_PCoA,type="samples",shape="Treatment",color="Time")+labs(title="PCoA norm_BRE Treatment vs Time",x="PCoA1 [39.7%]",y="PCoA2 [14.3%]")+geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+scale_color_manual(values = c("#CD9BCD","#C84248","orange","#00CDCD"))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

## Plot impacts

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_BRE_up between plots.tiff', units="in", width=6.5, height=4.5, res=300)

plot_ordination(r_BRE_up,r_BRE_up_PCoA,type="samples",color="Plot")+labs(title="PCoA norm_BRE between plots",x="PCoA1 [39.7%]",y="PCoA2 [14.3%]")+geom_point(size=2)+scale_color_manual(values = c("#8569D5","#00CDCD","orange","#DA5724","#508578","#CD9BCD","#AD6F3B","#673770","#D14285","#652926","#C84248","#CBD588","#D3D3ED","#BAFFE2","#000099","grey"))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

```

## Time based subset - Subset dataset to different development stage and see how compartment impacts will change. This is based on my assumption that at different development stage, the compartment impacts may differs

```{r}
# >>>> week 1 >>>>

wk1<-subset_samples(Seed_13021_up,Time=="1wk")
wk1 # 21355 taxa and 44 samples
wk1_up<-filter_taxa(wk1,function(x) sum(x)>1,prune = TRUE)
wk1_up #11316 taxa and 44 samples

r_wk1_up<-transform_sample_counts(wk1_up,function(x) x/sum(x))
r_wk1_up

# Read depth barplot

Seed_meta_wk1<-data.frame(sample_data(wk1_up))
tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/wk1_Read_depth_barplot.tiff",units = 'in',width = 16,height = 8,res = 300,bg='White',compression = 'lzw')

ggplot(data=Seed_meta_wk1,aes(x=Sample_ID,y=Read_depth))+geom_bar(width=0.7,stat = "identity") +xlab("Sample_ID")+ylab("Sequencing depth") +
  scale_y_continuous(labels = scales::comma)+theme(panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),text=element_text(family="Arial"), plot.title = element_text(family='Arial', hjust = .5,vjust = 3,size = 24,face="bold"), axis.text.x = element_text(family="Arial",size=8,angle=90,vjust = .6),axis.text.y = element_text(family="Arial",size=14),axis.title.x = element_text(family="Arial",size=20,face="bold"),axis.title.y = element_text(family="Arial",size=20,face="bold"),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

# reorder the Sample_ID based on Compartment and time
Seed_meta_wk1_up<-Seed_meta_wk1[with(Seed_meta_wk1,order(Compartment)),]
Seed_meta_wk1_up$Sample_ID<-factor(Seed_meta_wk1_up$Sample_ID,levels = as.character(Seed_meta_wk1_up$Sample_ID))

tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Read_depth_barplot reordered based on Compartment and time.tiff",units = 'in',width = 16,height = 8,res = 300,bg='White',compression = 'lzw')

ggplot(data=Seed_meta_wk1_up,aes(x=Sample_ID,y=Read_depth))+geom_bar(width=0.7,stat = "identity") +xlab("Sample_ID")+ylab("Sequencing depth") +
  scale_y_continuous(labels = scales::comma)+theme(panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),text=element_text(family="Arial"), plot.title = element_text(family='Arial', hjust = .5,vjust = 3,size = 24,face="bold"), axis.text.x = element_text(family="Arial",size=8,angle=90,vjust = .6),axis.text.y = element_text(family="Arial",size=14),axis.title.x = element_text(family="Arial",size=20,face="bold"),axis.title.y = element_text(family="Arial",size=20,face="bold"),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()


# PERMANOVA

# --- Read_depth --- insignificant
set.seed(1013) 
adonis2(t(otu_table(r_wk1_up ))~Read_depth,data=data.frame(sample_data(r_wk1_up )),permutations=999,by="margin")

# ----- Plot impacts-----insignificant
set.seed(1013) 
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_wk1_up))$Compartment),
          within=Within(type='free'))
adonis2(t(otu_table(r_wk1_up))~Plot,data=data.frame(sample_data(r_wk1_up )),permutations=perm,by="margin")

#  ---Treatment ---- # insignificant impact
set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_wk1_up))$Compartment),
          within=Within(type='free'))
adonis2(t(otu_table(r_wk1_up))~Treatment,data=data.frame(sample_data(r_wk1_up)),permutations=perm,by="margin")

#  ---Compartment ---- # p=0.001 and R2=59.85%
set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_wk1_up))$Plot),
          within=Within(type='free'))
adonis2(t(otu_table(r_wk1_up ))~Compartment,data=data.frame(sample_data(r_wk1_up )),permutations=perm,by="margin")

#            Df SumOfSqs      R2      F Pr(>F)    
#Compartment  2   5.9584 0.59882 30.599  0.001 ***
#Residual    41   3.9919 0.40118                  
#Total       43   9.9503 1.00000

# --- Compartment vs Read_depth --

set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_wk1_up))$Plot),
          within=Within(type='free'))
adonis2(t(otu_table(r_wk1_up ))~Compartment+Read_depth,data=data.frame(sample_data(r_wk1_up )),permutations=perm,by="margin")
#             Df SumOfSqs      R2       F Pr(>F)    
#Compartment  2   5.9501 0.59848 30.6182  0.001 ***
#Read_depth   1   0.1051 0.01057  1.0812  0.447    
#Residual    40   3.8866 0.39093                   
#Total       43   9.9419 1.00000

# ---- Read_Depth vs Compartment vs Treatment
set.seed(1013)
tidy(adonis2(t(otu_table(r_wk1_up ))~Compartment+Read_depth+Treatment,data=data.frame(sample_data(r_wk1_up )),permutations=999,by="margin"))


# ---- Read_Depth vs Compartment vs Plot
set.seed(1013)
tidy(adonis2(t(otu_table(r_wk1_up ))~Compartment+Read_depth+Plot,data=data.frame(sample_data(r_wk1_up )),permutations=999,by="margin"))


# PCoA plot

r_wk1_bray<-vegdist(t(otu_table(r_wk1_up)),method="bray",binary = FALSE)
r_wk1_PCoA<-ordinate(r_wk1_up,method = "PCoA", r_wk1_bray)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_wk1 Compartment vs Treatment.tiff', units="in", width=7, height=5, res=300)

plot_ordination(r_wk1_up,r_wk1_PCoA,type="samples",color="Compartment",shape = 'Treatment')+labs(title="r_wk1_Time",x="PCoA1 [40.1%]",y="PCoA2 [24.0%]")+geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+scale_color_manual(values = c("#C84248","orange","#00CDCD"))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

# >>>> week 3 compartment impacts >>>>

wk3<-subset_samples(Seed_13021_up,Time=="3wk")
wk3 #21355 taxa and 45 samples
wk3_up<-filter_taxa(wk3,function(x) sum(x)>1,prune = TRUE)
wk3_up #11403 taxa and 45 samples

r_wk3_up<-transform_sample_counts(wk3_up,function(x) x/sum(x))

wk3_up2<-prune_samples(sample_names(wk3_up)[which(sample_names(wk3_up)!="105E_3w_ML")],wk3_up)
r_wk3_up2<-transform_sample_counts(wk3_up2,function(x) x/sum(x))

# Read depth barplot

Seed_meta_wk3<-data.frame(sample_data(wk3_up))
tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/wk3_Read_depth_barplot.tiff",units = 'in',width = 16,height = 8,res = 300,bg='White',compression = 'lzw')

ggplot(data=Seed_meta_wk3,aes(x=Sample_ID,y=Read_depth))+geom_bar(width=0.7,stat = "identity") +xlab("Sample_ID")+ylab("Sequencing depth") +
  scale_y_continuous(labels = scales::comma)+theme(panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),text=element_text(family="Arial"), plot.title = element_text(family='Arial', hjust = .5,vjust = 3,size = 24,face="bold"), axis.text.x = element_text(family="Arial",size=8,angle=90,vjust = .6),axis.text.y = element_text(family="Arial",size=14),axis.title.x = element_text(family="Arial",size=20,face="bold"),axis.title.y = element_text(family="Arial",size=20,face="bold"),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

# reorder the Sample_ID based on Compartment and time
Seed_meta_wk3_up<-Seed_meta_wk3[with(Seed_meta_wk3,order(Compartment)),]
Seed_meta_wk3_up$Sample_ID<-factor(Seed_meta_wk3_up$Sample_ID,levels = as.character(Seed_meta_wk3_up$Sample_ID))

tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Read_depth_barplot reordered based on Compartment and time.tiff",units = 'in',width = 16,height = 8,res = 300,bg='White',compression = 'lzw')

ggplot(data=Seed_meta_wk3_up,aes(x=Sample_ID,y=Read_depth))+geom_bar(width=0.7,stat = "identity") +xlab("Sample_ID")+ylab("Sequencing depth") +
  scale_y_continuous(labels = scales::comma)+theme(panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),text=element_text(family="Arial"), plot.title = element_text(family='Arial', hjust = .5,vjust = 3,size = 24,face="bold"), axis.text.x = element_text(family="Arial",size=8,angle=90,vjust = .6),axis.text.y = element_text(family="Arial",size=14),axis.title.x = element_text(family="Arial",size=20,face="bold"),axis.title.y = element_text(family="Arial",size=20,face="bold"),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

# PERMANOVA

# --- Read_depth --- insignificant
set.seed(1013) 
adonis2(t(otu_table(r_wk3_up))~Read_depth,data=data.frame(sample_data(r_wk3_up)),permutations=999,by="margin")

set.seed(1013) 
adonis2(t(otu_table(r_wk3_up2))~Read_depth,data=data.frame(sample_data(r_wk3_up2)),permutations=999,by="margin")

# ----- Plot impacts-----significant
set.seed(1013) 
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_wk3_up))$Compartment),
          within=Within(type='free'))
adonis2(t(otu_table(r_wk3_up))~Plot,data=data.frame(sample_data(r_wk3_up )),permutations=perm,by="margin")

#  ---Treatment ---- # pvalue= 0.022 *

set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_wk3_up))$Compartment),
          within=Within(type='free'))
adonis2(t(otu_table(r_wk3_up ))~Treatment,data=data.frame(sample_data(r_wk3_up)),permutations=perm,by="margin")

#  ---Compartment ---- # p=0.001 and R2=61.28%
set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_wk3_up))$Plot),
          within=Within(type='free'))
adonis2(t(otu_table(r_wk3_up ))~Compartment,data=data.frame(sample_data(r_wk3_up )),permutations=perm,by="margin")

# --- Compartment vs Read_depth --

set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_wk3_up))$Plot),
          within=Within(type='free'))
adonis2(t(otu_table(r_wk3_up ))~Compartment+Read_depth,data=data.frame(sample_data(r_wk3_up )),permutations=perm,by="margin")

set.seed(1013)
adonis2(t(otu_table(r_wk3_up ))~Compartment:Read_depth,data=data.frame(sample_data(r_wk3_up )),permutations=999,by="margin")

set.seed(1013)
adonis2(t(otu_table(r_wk3_up ))~Compartment+Read_depth,data=data.frame(sample_data(r_wk3_up )),permutations=999,by="margin")

set.seed(1013)
adonis2(t(otu_table(r_wk3_up ))~Compartment/Read_depth,data=data.frame(sample_data(r_wk3_up )),permutations=999,by="margin")

# ---- Read_Depth vs Compartment vs Treatment
set.seed(1013)
tidy(adonis2(t(otu_table(r_wk3_up ))~Compartment+Read_depth+Treatment,data=data.frame(sample_data(r_wk3_up )),permutations=999,by="margin"))

# ----  Compartment vs plot vs read depth
set.seed(1013)
tidy(adonis2(t(otu_table(r_wk3_up ))~Compartment+Read_depth+Plot,data=data.frame(sample_data(r_wk3_up )),permutations=999,by="margin"))


# PCoA plot

# --- Compartment vs Treatment ----

r_wk3_bray<-vegdist(t(otu_table(r_wk3_up)),method="bray",binary = FALSE)
r_wk3_PCoA<-ordinate(r_wk3_up,method = "PCoA", r_wk3_bray)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_wk3 Compartment vs Treatment.tiff', units="in", width=7, height=5, res=300)

plot_ordination(r_wk3_up,r_wk3_PCoA,type="samples",color="Compartment",shape = 'Treatment')+labs(title="r_wk3_Time",x="PCoA1 [61.7%]",y="PCoA2 [10.3%]")+geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+scale_color_manual(values = c("#C84248","orange","#00CDCD"))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

# ----Compartment vs Read_depth

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_wk3 Compartment vs Read_depth.tiff', units="in", width=7, height=5, res=300)

plot_ordination(r_wk3_up,r_wk3_PCoA,type="samples",color="Read_depth",shape ='Compartment')+labs(title="r_wk3_Time",x="PCoA1 [61.7%]",y="PCoA2 [10.3%]")+geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+scale_color_gradient(low = "#4c9b58",high = "#d13436")+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()


# >>>> week 4 compartment impacts >>>>

wk4<-subset_samples(Seed_13021_up,Time=="4wk")
wk4 #21355 taxa and 45 samples
wk4_up<-filter_taxa(wk4,function(x) sum(x)>1,prune = TRUE)
wk4_up #11513 taxa and 45 samples
r_wk4_up<-transform_sample_counts(wk4_up,function(x) x/sum(x))

# Read depth barplot

Seed_meta_wk4<-data.frame(sample_data(wk4_up))
tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/wk4_Read_depth_barplot.tiff",units = 'in',width = 16,height = 8,res = 300,bg='White',compression = 'lzw')

ggplot(data=Seed_meta_wk4,aes(x=Sample_ID,y=Read_depth))+geom_bar(width=0.7,stat = "identity") +xlab("Sample_ID")+ylab("Sequencing depth") +
  scale_y_continuous(labels = scales::comma)+theme(panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),text=element_text(family="Arial"), plot.title = element_text(family='Arial', hjust = .5,vjust = 3,size = 24,face="bold"), axis.text.x = element_text(family="Arial",size=8,angle=90,vjust = .6),axis.text.y = element_text(family="Arial",size=14),axis.title.x = element_text(family="Arial",size=20,face="bold"),axis.title.y = element_text(family="Arial",size=20,face="bold"),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

# reorder the Sample_ID based on Compartment and time
Seed_meta_wk4_up<-Seed_meta_wk4[with(Seed_meta_wk4,order(Compartment)),]
Seed_meta_wk4_up$Sample_ID<-factor(Seed_meta_wk4_up$Sample_ID,levels = as.character(Seed_meta_wk4_up$Sample_ID))

tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Read_depth_barplot reordered based on Compartment and time.tiff",units = 'in',width = 16,height = 8,res = 300,bg='White',compression = 'lzw')

ggplot(data=Seed_meta_wk4_up,aes(x=Sample_ID,y=Read_depth))+geom_bar(width=0.7,stat = "identity") +xlab("Sample_ID")+ylab("Sequencing depth") +
  scale_y_continuous(labels = scales::comma)+theme(panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),text=element_text(family="Arial"), plot.title = element_text(family='Arial', hjust = .5,vjust = 3,size = 24,face="bold"), axis.text.x = element_text(family="Arial",size=8,angle=90,vjust = .6),axis.text.y = element_text(family="Arial",size=14),axis.title.x = element_text(family="Arial",size=20,face="bold"),axis.title.y = element_text(family="Arial",size=20,face="bold"),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

# PERMANOVA

# --- Read_depth --- insignificant
set.seed(1013) 
adonis2(t(otu_table(r_wk4_up))~Read_depth,data=data.frame(sample_data(r_wk4_up )),permutations=999,by="margin")

# ----- Plot impacts-----insignificant
set.seed(1013) 
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_wk4_up))$Compartment),
          within=Within(type='free'))
adonis2(t(otu_table(r_wk4_up))~Plot,data=data.frame(sample_data(r_wk4_up )),permutations=perm,by="margin")

#  ---Treatment ---- # pvalue= 0.003 **

set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_wk4_up))$Compartment),
          within=Within(type='free'))
adonis2(t(otu_table(r_wk4_up ))~Treatment,data=data.frame(sample_data(r_wk4_up)),permutations=perm,by="margin")

#  ---Compartment ---- # p=0.001 and R2=65.85%

set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_wk4_up))$Plot),
          within=Within(type='free'))
adonis2(t(otu_table(r_wk4_up ))~Compartment,data=data.frame(sample_data(r_wk4_up )),permutations=perm,by="margin")

# --- Compartment vs Read_depth --

set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_wk4_up))$Plot),
          within=Within(type='free'))
adonis2(t(otu_table(r_wk4_up ))~Compartment+Read_depth,data=data.frame(sample_data(r_wk4_up )),permutations=perm,by="margin")

# ---- Read_Depth vs Compartment vs Treatment
set.seed(1013)
tidy(adonis2(t(otu_table(r_wk4_up ))~Compartment+Read_depth+Treatment,data=data.frame(sample_data(r_wk4_up )),permutations=999,by="margin"))


# ---- Read_Depth vs Compartment vs Plot
set.seed(1013)
tidy(adonis2(t(otu_table(r_wk4_up ))~Compartment+Read_depth+Plot,data=data.frame(sample_data(r_wk4_up )),permutations=999,by="margin"))

# PCoA plot

r_wk4_bray<-vegdist(t(otu_table(r_wk4_up)),method="bray",binary = FALSE)
r_wk4_PCoA<-ordinate(r_wk4_up,method = "PCoA", r_wk4_bray)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_wk4 Compartment vs Treatment.tiff', units="in", width=7, height=5, res=300)

plot_ordination(r_wk4_up,r_wk4_PCoA,type="samples",color="Compartment",shape = 'Treatment')+labs(title="r_wk4_Time",x="PCoA1 [54.9%]",y="PCoA2 [12.8%]")+geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+scale_color_manual(values = c("#C84248","orange","#00CDCD"))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()
```


## Subset dataset to different compartment and Evaluate the treatment and Time impacts


```{r}
## >>>>  Bulk soil >>>>>>

Bulk<-subset_samples(Seed_13021_up,Compartment=="Bulk")
Bulk # 21355 taxa and 45 samples
## remove all singletons
Bulk_up<-filter_taxa(Bulk,function(x) sum(x)>1,prune = TRUE)
Bulk_up # 14071 taxa and 45 samples
r_Bulk_up<-transform_sample_counts(Bulk_up,function(x) x/sum(x))

## PERMANOVA

## >>>>>> Plot >>>>> significant p=0.001 and explain 48.17% of the variantions
set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_Bulk_up))$Time),
          within=Within(type='free'))
adonis2(t(otu_table(r_Bulk_up))~Plot,data=data.frame(sample_data(r_Bulk_up)),permutations=perm,by="margin")

## >>>> Read_depth >>>>> 0.19
set.seed(1013)
adonis2(t(otu_table(r_Bulk_up))~Read_depth,data=data.frame(sample_data(r_Bulk_up)),permutations=999,by="margin")

## >>>> Time >>>> significant and could explain 11.99% of the variation
set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_Bulk_up))$Plot),
          within=Within(type='free'))
adonis2(t(otu_table(r_Bulk_up))~Time,data=data.frame(sample_data(r_Bulk_up)),permutations=perm,by="margin")

## >>>> Treatment >>>> barely significant p=0.038 and R2=6.20% 
set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_Bulk_up))$Time),
          within=Within(type='free'))
adonis2(t(otu_table(r_Bulk_up))~Treatment,data=data.frame(sample_data(r_Bulk_up)),permutations=999,by="margin")


# ---- Treatment: Time --- include individual impact and interactions impact. Could be infered based on freedom of variables.

set.seed(1013)
adonis2(t(otu_table(r_Bulk_up))~Time:Treatment,data=data.frame(sample_data(r_Bulk_up)),permutations=999,by="margin") 

#adonis2(formula = t(otu_table(r_Bulk_up)) ~ Time:Treatment, data = data.frame(sample_data(r_Bulk_up)), permutations = 999, by = "margin")
#               Df SumOfSqs      R2      F Pr(>F)   
#Time:Treatment  8   0.9388 0.25141 1.4693  0.002 **
#Residual       35   2.7952 0.74859                 
#Total          43   3.7340 1.00000

## >>>>>> Treatment*Time interaction impacts >>>>> insignificant

set.seed(1013)
adonis2(t(otu_table(r_Bulk_up))~Time+Treatment,data=data.frame(sample_data(r_Bulk_up)),permutations=999,by="margin") 

#          Df SumOfSqs      R2      F Pr(>F)    
#Time       2   0.5144 0.11990 2.9313  0.001 ***
#Treatment  2   0.2661 0.06201 1.5160  0.019 *  
#Residual  40   3.5098 0.81809                  
#Total     44   4.2903 1.00000


## >>>>>> Read_depth vs Time vs Treatment

set.seed(1013)
tidy(adonis2(t(otu_table(r_Bulk_up))~Read_depth+Treatment+Time,data=data.frame(sample_data(r_Bulk_up)),permutations=999,by="margin")) # Time is significant, p=0.001, and R2= 12.45. Treatment is significant, R2=0.12 and p=0.001; 

#adonis2(formula = t(otu_table(r_Bulk_up)) ~ Read_depth + Treatment + Time, data = data.frame(sample_data(r_Bulk_up)), permutations = 999, by = "margin")
#           Df SumOfSqs      R2      F Pr(>F)    
#Read_depth  1   0.0728 0.01950 0.9457  0.499    
#Treatment   2   0.2151 0.05762 1.3968  0.067 .  
#Time        2   0.4540 0.12159 2.9478  0.001 ***
#Residual   38   2.9264 0.78373                  
#Total      43   3.7340 1.00000

## >>>>>> Read_depth vs Time vs Plot

set.seed(1013)
adonis2(t(otu_table(r_Bulk_up))~Read_depth+Plot+Time,data=data.frame(sample_data(r_Bulk_up)),permutations=999,by="margin")
#tidy(adonis2(t(otu_table(r_Bulk_up))~Read_depth+Plot+Time,data=data.frame(sample_data(r_Bulk_up)),permutations=999,by="margin"))


#PCoA plot - Treatment vs Time

Bulk_up_bray<-vegdist(t(otu_table(r_Bulk_up)),method="bray",binary = FALSE)
Bulk_up_PCoA<-ordinate(r_Bulk_up,method = "PCoA",Bulk_up_bray)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Bulk_up Time vs Treatment.tiff', units="in", width=7, height=5, res=300)

plot_ordination(r_Bulk_up,Bulk_up_PCoA,type="samples",shape="Time",color="Treatment")+labs(title="Bulk_up_Time vs Treatment",x="PCoA1 [17.6%]",y="PCoA2 [11.7%]")+geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

## Plot difference
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Bulk_up between plots difference.tiff', units="in", width=7, height=5, res=300)

plot_ordination(r_Bulk_up,Bulk_up_PCoA,type="samples",color="Plot",shape = 'Time')+labs(title="PCoA Bulk_up between plots difference",x="PCoA1 [17.6%]",y="PCoA2 [11.7%]")+geom_point(size=2)+scale_color_manual(values = c("#8569D5","#00CDCD","orange","#DA5724","#508578","#CD9BCD","#AD6F3B","#673770","#D14285","#652926","#C84248","#CBD588","#D3D3ED","#BAFFE2","#000099","grey"))+scale_shape_manual(values = c(1,2,4))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

# >>>>> CAP analysis >>>

set.seed(1013)
cap_Bulk_up<-capscale(t(otu_table(r_Bulk_up))~Time+Condition(Plot),data=data.frame(sample_data(r_Bulk_up)),distance = 'bray',dfun = vegdist)
#summary(cap_Bulk_up)
anova(cap_Bulk_up) 

#capscale(formula = t(otu_table(r_Bulk_up)) ~ Time + Condition(Plot), data = data.frame(sample_data(r_Bulk_up)), distance = "bray", dfun = vegdist)
#         Df SumOfSqs      F Pr(>F)    
#Model     2  0.45865 4.5257  0.001 ***
#Residual 27  1.36812

Bulk_up_cap<-ordinate(Bulk_up,'CAP','bray',~Time+Treatment)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/CAP_Bulk_up.tiff', units="in", width=7, height=5, res=300)
plot_ordination(Bulk_up,Bulk_up_cap,type="samples",color="Treatment",shape = "Time")+
  geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+ ggtitle("CAP plot of soybean rhizosphere and bulk microbiome \n ~Soil_type+Treatment") +theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

      # >>> Bulk_1wk >>>>
      
      Bulk_1wk<-subset_samples(Bulk,Time=="1wk")
      Bulk_1wk_up<-filter_taxa(Bulk_1wk,function(x) sum(x)>1,prune = TRUE)
      Bulk_1wk_up #10647 taxa and 15 samples
      
      # PERMANOVA test for significance 
      
      r_Bulk_1wk_up<-transform_sample_counts(Bulk_1wk_up,function(x) x/sum(x))
      set.seed(1013)
      adonis2(t(otu_table(r_Bulk_1wk_up))~Treatment,data=data.frame(sample_data(r_Bulk_1wk_up)),permutations=999,by="margin") # p value =0.785
      
      
      #>>> PCoA >>>>>
      
      r_Bulk_1wk_up_bray<-vegdist(t(otu_table(r_Bulk_1wk_up)),method="bray",binary = FALSE)
      r_Bulk_1wk_up_PCoA<-ordinate(r_Bulk_1wk_up,method = "PCoA",r_Bulk_1wk_up_bray)
      tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_Bulk_1wk_13021_up_Treatment.tiff', units="in", width=7, height=5, res=300)
      
      plot_ordination(r_Bulk_1wk_up,r_Bulk_1wk_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="PCoA r_Bulk_1wk_up_Treatment")+
        geom_point(size=2)+scale_shape_manual(values = c(15,16,17))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+ ggtitle("Bulk_1wk_up_Treatment") +theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
      dev.off()
      
      # >>> Bulk_3wk_up >>>> 
      
      Bulk_3wk<-subset_samples(Bulk,Time=="3wk")
      Bulk_3wk # 28114 OTUs across 15 samples
      Bulk_3wk_up<-filter_taxa(Bulk_3wk,function(x) sum(x)>1,prune = TRUE)
      Bulk_3wk_up # 10389 taxa and 15 samples
      
      # PERMANOVA test
      r_Bulk_3wk_up<-transform_sample_counts(Bulk_3wk_up,function(x) x/sum(x))
      set.seed(1013)
      adonis2(t(otu_table(r_Bulk_3wk_up))~Treatment,data=data.frame(sample_data(r_Bulk_3wk_up)),permutations=999,by="margin") #0.473
      
      # >>>>> PCoA plot
      
      r_Bulk_3wk_up_bray<-vegdist(t(otu_table(r_Bulk_3wk_up)),method="bray",binary = FALSE)
      r_Bulk_3wk_up_PCoA<-ordinate(r_Bulk_3wk_up,method = "PCoA", r_Bulk_3wk_up_bray)
      tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_Bulk_3wk_up_Treatment.tiff', units="in", width=7, height=5, res=300)
      
      plot_ordination(r_Bulk_3wk_up,r_Bulk_3wk_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="PCoA r_Bulk_3wk_up_Treatment")+geom_point(size=2)+scale_shape_manual(values = c(15,16,17))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
      dev.off()
      
      ## >>>>>> Bulk_4wk_up >>>>
      
      Bulk_4wk<-subset_samples(Bulk,Time=="4wk")
      Bulk_4wk # 28114 OTUs across 15 samples
      Bulk_4wk_up<-filter_taxa(Bulk_4wk,function(x) sum(x)>1,prune = TRUE)
      Bulk_4wk_up # 10335 taxa and 14 samples
      
      # PERMANOVA test
      r_Bulk_4wk_up<-transform_sample_counts(Bulk_4wk_up,function(x) x/sum(x))
      
      #>>> Treatment impact --insignificant
      set.seed(1013)
      adonis2(t(otu_table(r_Bulk_4wk_up))~Treatment,data=data.frame(sample_data(r_Bulk_4wk_up)),permutations=999,by="margin") # p=0.213
      
      
      # >>>>> PCoA plot
      
      r_Bulk_4wk_up_bray<-vegdist(t(otu_table(r_Bulk_4wk_up)),method="bray",binary = FALSE)
      r_Bulk_4wk_up_PCoA<-ordinate(r_Bulk_4wk_up,method = "PCoA", r_Bulk_4wk_up_bray)
      tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_Bulk_4wk_up_Treatment.tiff', units="in", width=7, height=5, res=300)
      
      plot_ordination(r_Bulk_4wk_up,r_Bulk_4wk_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="PCoA r_Bulk_4wk_up_Treatment")+
        geom_point(size=2)+scale_shape_manual(values = c(15,16,17))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
      dev.off()
```

## >>>>  Rhisphere >>>>>>

```{r}

Rhi<-subset_samples(Seed_13021_up,Compartment=="Rhizosphere")
Rhi # 21355 taxa and 44 samples

## remove all singletons
Rhi_up<-filter_taxa(Rhi,function(x) sum(x)>1,prune = TRUE)
Rhi_up # 10465 taxa and 44 samples
r_Rhi_up<-transform_sample_counts(Rhi_up,function(x) x/sum(x))

## PERMANOVA

# ----- Plot ----- by margin
set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_Rhi_up))$Time),
          within=Within(type='free'))
adonis2(t(otu_table(r_Rhi_up))~Plot,data=data.frame(sample_data(r_Rhi_up)),permutations=perm,by="margin")

#         Df SumOfSqs      R2      F Pr(>F)    
#Plot     14   2.1208 0.30325 0.9016  0.001 ***
#Residual 29   4.8728 0.69675                  
#Total    43   6.9936 1.00000

# ----- Plot ----- by term
set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_Rhi_up))$Time),
          within=Within(type='free'))
adonis2(t(otu_table(r_Rhi_up))~Plot,data=data.frame(sample_data(r_Rhi_up)),permutations=perm,by="term")

#         Df SumOfSqs      R2      F Pr(>F)    
#Plot     14   2.1208 0.30325 0.9016  0.001 ***
#Residual 29   4.8728 0.69675                  
#Total    43   6.9936 1.00000

# ----- Read_depth ----insignificant

set.seed(1013)
adonis2(t(otu_table(r_Rhi_up))~Read_depth,data=data.frame(sample_data(r_Rhi_up)),permutations=999,by="margin")


# ------Time ------ significant and R2=42.69
set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_Rhi_up))$Plot),
          within=Within(type='free'))
adonis2(t(otu_table(r_Rhi_up))~Time,data=data.frame(sample_data(r_Rhi_up)),permutations=perm,by="margin")

#         Df SumOfSqs      R2      F Pr(>F)    
#Time      2   2.8993 0.41457 14.517  0.001 ***
#Residual 41   4.0943 0.58543                  
#Total    43   6.9936 1.00000

# -----Treatment------significant
set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_Rhi_up))$Time),
          within=Within(type='free'))
adonis2(t(otu_table(r_Rhi_up))~Treatment,data=data.frame(sample_data(r_Rhi_up)),permutations=perm,by="margin")

#          Df SumOfSqs      R2     F Pr(>F)   
#Treatment  2   0.3209 0.04589 0.986  0.008 **
#Residual  41   6.6727 0.95411                
#Total     43   6.9936 1.00000


# ----- Treatment and Time impacts
set.seed(1013)
adonis2(t(otu_table(r_Rhi_up))~Treatment+Time,data=data.frame(sample_data(r_Rhi_up)),permutations=999,by="margin")
#          Df SumOfSqs      R2       F Pr(>F)    
#Treatment  2   0.2958 0.04459  1.6537  0.103    
#Time       2   2.8356 0.42742 15.8522  0.001 ***
#Residual  39   3.4881 0.52577                   
#Total     43   6.6342 1.00000

# --- Treatment*Time interactions impact
set.seed(1013)
adonis2(t(otu_table(r_Rhi_up))~Treatment*Time,data=data.frame(sample_data(r_Rhi_up)),permutations=999,by="margin") 

#               Df SumOfSqs      R2      F Pr(>F)
#Treatment:Time  4   0.2973 0.04251 0.7451  0.776
#Residual       35   3.4912 0.49919              
#Total          43   6.9936 1.00000

# ----- Read_depth vs Time vs Treatment---
set.seed(1013)
tidy(adonis2(t(otu_table(r_Rhi_up))~Read_depth+Treatment+Time,data=data.frame(sample_data(r_Rhi_up)),permutations=999,by="margin")) 

#term          df SumOfSqs     R2 statistic p.value
#  <chr>      <dbl>    <dbl>  <dbl>     <dbl>   <dbl>
#1 Read_depth     1    0.124 0.0186      1.40   0.17 
#2 Treatment      2    0.244 0.0368      1.38   0.179
#3 Time           2    2.69  0.406      15.2    0.001
#4 Residual      38    3.36  0.507      NA     NA    
#5 Total         43    6.63  1          NA     NA 

# ----- Read_depth vs Time vs Plot---
set.seed(1013)
tidy(adonis2(t(otu_table(r_Rhi_up))~Read_depth+Plot+Time,data=data.frame(sample_data(r_Rhi_up)),permutations=999,by="margin")) 

#term          df SumOfSqs      R2 statistic p.value
#  <chr>      <dbl>    <dbl>   <dbl>     <dbl>   <dbl>
#1 Read_depth     1   0.0565 0.00807     0.739   0.592
#2 Plot          14   1.92   0.275       1.80    0.004
#3 Time           2   2.61   0.373      17.1     0.001
#4 Residual      26   1.99   0.284      NA      NA    
#5 Total         43   6.99   1          NA      NA 

# >>>> PCoA >>>

# >>> Treatment_vs_Time

dim(otu_table(r_Rhi_up)) # 13696 OTUs across 44 samples
Rhi_up_bray<-vegdist(t(otu_table(r_Rhi_up)),method="bray",binary = FALSE)
Rhi_up_PCoA<-ordinate(Rhi_up,method = "PCoA", Rhi_up_bray)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Rhi_up_Treatment_vs_Time.tiff', units="in", width=7, height=5, res=300)

plot_ordination(Rhi_up,Rhi_up_PCoA,type="samples",shape="Time",color="Treatment")+labs(title="PCoA Rhi_up_Treatment_vs_Time",x="PCoA1 [43.0%]",y="PCoA2 [10.8%]")+
  geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()


tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Rhi_up_Treatment_vs_Time with labels.tiff', units="in", width=7, height=5, res=300)

plot_ordination(Rhi_up,Rhi_up_PCoA,type="samples",shape="Time",color="Treatment",label="Sample_ID")+labs(title="PCoA Rhi_up_Treatment_vs_Time",x="PCoA1 [43.0%]",y="PCoA2 [10.8%]")+
  geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

## >>>> Plot impacts >>>>

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Rhi_up_between Plot.tiff', units="in", width=7, height=5, res=300)

plot_ordination(Rhi_up,Rhi_up_PCoA,type="samples",color="Plot")+labs(title="PCoA Rhi_up_between Plot",x="PCoA1 [43.0%]",y="PCoA2 [10.8%]")+geom_point(size=2)+scale_color_manual(values = c("#8569D5","#00CDCD","orange","#DA5724","#508578","#CD9BCD","#AD6F3B","#673770","#D14285","#652926","#C84248","#CBD588","#D3D3ED","#BAFFE2","#000099","grey"))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()


        # >>> Rhi_1wk >>>>
        
        Rhi_1wk<-subset_samples(Rhi,Time=="1wk")
        Rhi_1wk_up<-filter_taxa(Rhi_1wk,function(x) sum(x)>1,prune = TRUE)
        Rhi_1wk_up #6334 taxa and 14 samples
        
        # PERMANOVA test for significance 
        
        r_Rhi_1wk_up<-transform_sample_counts(Rhi_1wk_up,function(x) x/sum(x))
        set.seed(1013)
        
        # >>>>  Read_depth >>>  insignificant
        set.seed(1013)
        adonis2(t(otu_table(r_Rhi_1wk_up))~Read_depth,data=data.frame(sample_data(r_Rhi_1wk_up)),permutations=999,by="margin") # p value =0.431
        
        # >>> Treatment >>>> insignificant
        set.seed(1013)
        adonis2(t(otu_table(r_Rhi_1wk_up))~Treatment,data=data.frame(sample_data(r_Rhi_1wk_up)),permutations=999,by="margin")
        
      
        #>>> PCoA >>>>>
        
        r_Rhi_1wk_up_bray<-vegdist(t(otu_table(r_Rhi_1wk_up)),method="bray",binary = FALSE)
        r_Rhi_1wk_up_PCoA<-ordinate(r_Rhi_1wk_up,method = "PCoA", r_Rhi_1wk_up_bray)
        tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_Rhi_1wk_up_Treatment.tiff', units="in", width=7, height=5, res=300)
        
        plot_ordination(r_Rhi_1wk_up,r_Rhi_1wk_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="PCoA r_Rhi_1wk_up_Treatment")+
          geom_point(size=2)+scale_shape_manual(values = c(15,16,17))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
        dev.off()
        
        # >>> Rhi_3wk_up >>>>
        
        Rhi_3wk<-subset_samples(Rhi,Time=="3wk")
        Rhi_3wk # 27378 taxa and 15 samples
        Rhi_3wk_up<-filter_taxa(Rhi_3wk,function(x) sum(x)>1,prune = TRUE)
        Rhi_3wk_up # 8660 taxa and 15 samples
        
        # PERMANOVA test
        r_Rhi_3wk_up<-transform_sample_counts(Rhi_3wk_up,function(x) x/sum(x))
        
        # Treatment 
        set.seed(1013)
        adonis2(t(otu_table(r_Rhi_3wk_up))~Treatment,data=data.frame(sample_data(r_Rhi_3wk_up)),permutations=999,by="margin") # p=0.277
        
        # Read_Depth 
        set.seed(1013)
        adonis2(t(otu_table(r_Rhi_3wk_up))~Read_depth,data=data.frame(sample_data(r_Rhi_3wk_up)),permutations=999,by="margin") #  0.461
       
       
        # >>>>> PCoA plot
        
        r_Rhi_3wk_up_bray<-vegdist(t(otu_table(r_Rhi_3wk_up)),method="bray",binary = FALSE)
        r_Rhi_3wk_up_PCoA<-ordinate(r_Rhi_3wk_up,method = "PCoA", r_Rhi_3wk_up_bray)
        tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_Rhi_3wk_up_Treatment.tiff', units="in", width=7, height=5, res=300)
        
        plot_ordination(r_Rhi_3wk_up,r_Rhi_3wk_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="PCoA r_Rhi_3wk_up_Treatment")+
          geom_point(size=2)+scale_shape_manual(values = c(15,16,17))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
        dev.off()
        
        ## >>>>>> Rhi_4wk_up >>>>
        
        Rhi_4wk<-subset_samples(Rhi,Time=="4wk")
        Rhi_4wk # 27378 taxa and 15 samples
        Rhi_4wk_up<-filter_taxa(Rhi_4wk,function(x) sum(x)>1,prune = TRUE)
        Rhi_4wk_up # 8131 taxa and 15 samples
        
        # PERMANOVA test
        r_Rhi_4wk_up<-transform_sample_counts(Rhi_4wk_up,function(x) x/sum(x))
        
        # Treatment
        set.seed(1013)
        adonis2(t(otu_table(r_Rhi_4wk_up))~Treatment,data=data.frame(sample_data(r_Rhi_4wk_up)),permutations=999,by="margin") # p=0.181
        
        # Read_depth
        set.seed(1013)
        adonis2(t(otu_table(r_Rhi_4wk_up))~Read_depth,data=data.frame(sample_data(r_Rhi_4wk_up)),permutations=999,by="margin") # p=0.155
        
        # >>>>> PCoA plot
        
        r_Rhi_4wk_up_bray<-vegdist(t(otu_table(r_Rhi_4wk_up)),method="bray",binary = FALSE)
        r_Rhi_4wk_up_PCoA<-ordinate(r_Rhi_4wk_up,method = "PCoA", r_Rhi_4wk_up_bray)
        tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_Rhi_4wk_up_Treatment.tiff', units="in", width=7, height=5, res=300)
        
        plot_ordination(r_Rhi_4wk_up,r_Rhi_4wk_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="PCoA r_Rhi_4wk_up_Treatment")+geom_point(size=2)+scale_shape_manual(values = c(15,16,17))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
        dev.off()
```

## >>>>  Endosphere >>>>>>

```{r}
Endo<-subset_samples(Seed_13021_up,Compartment=="Endosphere")
Endo #21355 taxa and 45 samples

## remove all singletons
Endo_up<-filter_taxa(Endo,function(x) sum(x)>1,prune = TRUE)
Endo_up # 8715 taxa and 44 samples
r_Endo_up<-transform_sample_counts(Endo_up,function(x) x/sum(x))

## PERMANOVA

# ------ Plot ----- insignificant
set.seed(1013)
perm<-how(nperm=999,
          plots = Plots(strata = data.frame(sample_data(r_Endo_up))$Time),
          within=Within(type='free'))
adonis2(t(otu_table(r_Endo_up))~Plot,data=data.frame(sample_data(r_Endo_up)),permutations=perm,by="margin")

#         Df SumOfSqs      R2      F Pr(>F)    
#Plot     14   1.1034 0.17259 0.4321  0.001 ***
#Residual 29   5.2901 0.82741                  
#Total    43   6.3935 1.00000

# --- Read_depth --- insignificant pvalue=0.386

set.seed(1013)
adonis2(t(otu_table(r_Endo_up))~Read_depth,data=data.frame(sample_data(r_Endo_up)),permutations=999,by="margin")


# ----- Time --- significant and R2=60.62%
set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_Endo_up))$Plot),
          within=Within(type='free'))
adonis2(t(otu_table(r_Endo_up))~Time,data=data.frame(sample_data(r_Endo_up)),permutations=perm,by="margin")

#         Df SumOfSqs      R2      F Pr(>F)    
#Time      2   3.8939 0.60904 31.936  0.001 ***
#Residual 41   2.4996 0.39096                  
#Total    43   6.3935 1.00000

# ------- Treatment -------- insignificant

set.seed(1013)
perm<-how(nperm=999,plots = Plots(strata = data.frame(sample_data(r_Endo_up))$Time),
          within=Within(type='free'))
adonis2(t(otu_table(r_Endo_up))~Treatment,data=data.frame(sample_data(r_Endo_up)),permutations=perm,by="margin")

#adonis2(formula = t(otu_table(r_Endo_up)) ~ Treatment, data = data.frame(sample_data(r_Endo_up)), permutations = 999, by = "margin")
#          Df SumOfSqs     R2      F Pr(>F)
#Treatment  2   0.1656 0.0261 0.5494  0.762
#Residual  41   6.1809 0.9739              
#Total     43   6.3465 1.0000

## --- Treatment + Time impacts

set.seed(1013)
adonis2(t(otu_table(r_Endo_up))~Treatment+Time,data=data.frame(sample_data(r_Endo_up)),permutations=999,by="margin") 

#          Df SumOfSqs      R2       F Pr(>F)    
#Treatment  2   0.1557 0.02436  1.2958  0.231    
#Time       2   3.8826 0.60728 32.3023  0.001 ***
#Residual  39   2.3438 0.36660                   
#Total     43   6.3935 1.00000

# --- interaction between treatment and Time ---
set.seed(1013)
adonis2(t(otu_table(r_Endo_up))~Treatment*Time,data=data.frame(sample_data(r_Endo_up)),permutations=999,by="margin")

#               Df SumOfSqs      R2      F Pr(>F)
#Treatment:Time  4   0.1725 0.02698 0.6952  0.726
#Residual       35   2.1713 0.33961              
#Total          43   6.3935 1.00000 

# ----- Read_depth vs Time vs Treatment

set.seed(1013)
tidy(adonis2(t(otu_table(r_Endo_up))~Read_depth+Time+Treatment,data=data.frame(sample_data(r_Endo_up)),permutations=999,by="margin"))

# ----- Read_depth vs Time vs Plot ---
set.seed(1013)
tidy(adonis2(t(otu_table(r_Endo_up))~Read_depth+Time+Plot,data=data.frame(sample_data(r_Endo_up)),permutations=999,by="margin"))

#term          df SumOfSqs     R2 statistic p.value
#  <chr>      <dbl>    <dbl>  <dbl>     <dbl>   <dbl>
#1 Read_depth     1    0.127 0.0199      2.53   0.075
#2 Time           2    3.75  0.586      37.3    0.001
#3 Plot          14    1.14  0.178       1.62   0.046
#4 Residual      26    1.31  0.204      NA     NA    
#5 Total         43    6.39  1          NA     NA 

# >>>> PCoA >>>

# >>> Treatment_vs_Time

dim(otu_table(r_Endo_up)) # 8686 OTUs across 44 samples
Endo_up_bray<-vegdist(t(otu_table(r_Endo_up)),method="bray",binary = FALSE)
Endo_up_PCoA<-ordinate(Endo_up,method = "PCoA", Endo_up_bray)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Endo_up_Time_vs_Treatment.tiff', units="in", width=7, height=5, res=300)

plot_ordination(Endo_up,Endo_up_PCoA,type="samples",shape="Time",color="Treatment",label = "Sample_ID")+labs(title="PCoA Endo_up_Time_vs_Treatment",x="PCoA1 [59.9%]",y="PCoA2 [8.5%]")+
  geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+scale_color_manual(values =c('#fce75d','#5d9ffc','#fc5df7' ))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()


# Plots difference

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Endo_up between plots.tiff', units="in", width=7, height=5, res=300)

plot_ordination(Endo_up,Endo_up_PCoA,type="samples",color="Plot",label ="Sample_ID")+labs(title="Endo_up_between plots",x="PCoA1 [59.9%]",y="PCoA2 [8.5%]")+geom_point(size=2)+scale_color_manual(values=c("#8569D5","#00CDCD","orange","#DA5724","#508578","#CD9BCD","#AD6F3B","#673770","#D14285","#652926","#C84248","#CBD588","#D3D3ED","#BAFFE2","#000099","grey"))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

    # >>> Endo_1wk >>>>
    
    Endo_1wk<-subset_samples(Endo,Time=="1wk")
    Endo_1wk_up<-filter_taxa(Endo_1wk,function(x) sum(x)>1,prune = TRUE)
    Endo_1wk_up #5951 OTUs across 15 samples
    
    # PERMANOVA test for significance 
    
    # >>>>> Treatment is not significant
    
    r_Endo_1wk_up<-transform_sample_counts(Endo_1wk_up,function(x) x/sum(x))
    set.seed(1013)
    adonis2(t(otu_table(r_Endo_1wk_up))~Treatment,data=data.frame(sample_data(r_Endo_1wk_up)),permutations=999,by="margin") # p value =0.607
    
    # Read depth
     r_Endo_1wk_up<-transform_sample_counts(Endo_1wk_up,function(x) x/sum(x))
    set.seed(1013)
    adonis2(t(otu_table(r_Endo_1wk_up))~Read_depth,data=data.frame(sample_data(r_Endo_1wk_up)),permutations=999,by="margin") # p value =0.459
    
    #>>> PCoA >>>>>
    
    r_Endo_1wk_up_bray<-vegdist(t(otu_table(r_Endo_1wk_up)),method="bray",binary = FALSE)
    r_Endo_1wk_up_PCoA<-ordinate(r_Endo_1wk_up,method = "PCoA", r_Endo_1wk_up_bray)
    tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_Endo_1wk_up_Treatment.tiff', units="in", width=7, height=5, res=300)
    
    plot_ordination(r_Endo_1wk_up,r_Endo_1wk_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="PCoA r_Endo_1wk_up_Treatment")+
      geom_point(size=2)+scale_shape_manual(values = c(15,16,17))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
    dev.off()
    
    # >>> Endo_3wk_up >>>>
    
    Endo_3wk<-subset_samples(Endo,Time=="3wk")
    Endo_3wk # 27378 taxa and 15 sampless
    Endo_3wk_up<-filter_taxa(Endo_3wk,function(x) sum(x)>1,prune = TRUE)
    Endo_3wk_up # 3682 taxa and 15 samples
    
    # PERMANOVA test
    
    # Treatment - insignificant
    r_Endo_3wk_up<-transform_sample_counts(Endo_3wk_up,function(x) x/sum(x))
    set.seed(1013)
    adonis2(t(otu_table(r_Endo_3wk_up))~Treatment,data=data.frame(sample_data(r_Endo_3wk_up)),permutations=999,by="margin") # p=0.509
    
    # Read_Depth
     r_Endo_3wk_up<-transform_sample_counts(Endo_3wk_up,function(x) x/sum(x))
    set.seed(1013)
    adonis2(t(otu_table(r_Endo_3wk_up))~Read_depth,data=data.frame(sample_data(r_Endo_3wk_up)),permutations=999,by="margin") # p=0.509
    
    # >>>>> PCoA plot
    
    r_Endo_3wk_up_bray<-vegdist(t(otu_table(r_Endo_3wk_up)),method="bray",binary = FALSE)
    r_Endo_3wk_up_PCoA<-ordinate(r_Endo_3wk_up,method = "PCoA", r_Endo_3wk_up_bray)
    tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_Endo_3wk_up_Treatment.tiff', units="in", width=7, height=5, res=300)
    
    plot_ordination(r_Endo_3wk_up,r_Endo_3wk_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="PCoA r_Endo_3wk_up_Treatment")+geom_point(size=2)+scale_shape_manual(values = c(15,16,17))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
    dev.off()
    
    ## >>>>>> Endo_4wk_up >>>>
    
    Endo_4wk<-subset_samples(Endo,Time=="4wk")
    Endo_4wk # 28114 OTUs across 15 samples
    Endo_4wk_up<-filter_taxa(Endo_4wk,function(x) sum(x)>1,prune = TRUE)
    Endo_4wk_up # 3899 OTUs across 14 samples
    
    # PERMANOVA test
    r_Endo_4wk_up<-transform_sample_counts(Endo_4wk_up,function(x) x/sum(x))
    set.seed(1013)
    adonis2(t(otu_table(r_Endo_4wk_up))~Treatment,data=data.frame(sample_data(r_Endo_4wk_up)),permutations=999,by="margin") # p=0.82
  
    # >>>>> PCoA plot
    
    r_Endo_4wk_up_bray<-vegdist(t(otu_table(r_Endo_4wk_up)),method="bray",binary = FALSE)
    r_Endo_4wk_up_PCoA<-ordinate(r_Endo_4wk_up,method = "PCoA", r_Endo_4wk_up_bray)
    tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_Endo_4wk_up_Treatment.tiff', units="in", width=7, height=5, res=300)
    
    plot_ordination(r_Endo_4wk_up,r_Endo_4wk_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="PCoA r_Endo_4wk_up_Treatment")+
      geom_point(size=2)+scale_shape_manual(values = c(15,16,17))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

```


## >>> Endo2_up >>> after remove 105E_1wk_ML

```{r}
## >>>>  Endosphere >>>>>>

Endo<-subset_samples(Seed_13021_up,Compartment=="Endosphere")
Endo # 28114 OTUs across 44 samples

Endo2<-subset_samples(Endo,Sample_ID!= "105E_1w_ML")
Endo2 #28114 taxa and 43 samples

## remove all singletons
Endo2_up<-filter_taxa(Endo2,function(x) sum(x)>1,prune = TRUE)
Endo2_up #  8641 taxa and 43 samples
r_Endo2_up<-transform_sample_counts(Endo2_up,function(x) x/sum(x))


## PERMANOVA

# --plot --
set.seed(1013)
perm<-how(nperm=999,
          plots = Plots(strata = data.frame(sample_data(r_Endo2_up))$Time),
          within=Within(type='free'))
adonis2(t(otu_table(r_Endo2_up))~Plot,data=data.frame(sample_data(r_Endo2_up)),permutations=perm,by="margin")

#         Df SumOfSqs      R2      F Pr(>F)    
#Plot     14   1.0623 0.17304 0.4334  0.001 ***
#Residual 29   5.0765 0.82696                  
#Total    43   6.1387 1.00000

## >>>>>> Treatment and Time impacts

set.seed(1013)
adonis2(t(otu_table(r_Endo2_up))~Treatment+Time,data=data.frame(sample_data(r_Endo2_up)),permutations=999,by="margin") 

#          Df SumOfSqs      R2       F Pr(>F)    
#Treatment  2   0.1542 0.02512  1.4576  0.181    
#Time       2   3.9072 0.63649 36.9347  0.001 ***
#Residual  39   2.0628 0.33604                   
#Total     43   6.1387 1.00000


# >>>>> Treatment and time interactions >>>

set.seed(1013)
adonis2(t(otu_table(Endo2_up))~Treatment*Time,data=data.frame(sample_data(Endo2_up)),permutations=999,by="margin") 

#               Df SumOfSqs      R2      F Pr(>F)
#Treatment:Time  4   0.1677 0.02735 0.7745  0.608
#Residual       35   1.8949 0.30904              
#Total          43   6.1315 1.00000


# ----- Read_depth vs Time vs Treatment

set.seed(1013)
tidy(adonis2(t(otu_table(r_Endo2_up))~Read_depth+Time+Treatment,data=data.frame(sample_data(r_Endo2_up)),permutations=999,by="margin"))

# ----- Read_depth vs Time vs Plot ---
set.seed(1013)
tidy(adonis2(t(otu_table(r_Endo2_up))~Read_depth+Time+Plot,data=data.frame(sample_data(r_Endo2_up)),permutations=999,by="margin"))


# >>>> PCoA >>>>

# >>> Time vs Treatment

dim(otu_table(Endo2_up)) # 8641 OTUs across 43 samples
Endo2_up_bray<-vegdist(t(otu_table(Endo2_up)),method="bray",binary = FALSE)
Endo2_up_PCoA<-ordinate(Endo2_up,method = "PCoA", Endo2_up_bray)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Endo2_up_Time_vs_Treatment.tiff', units="in", width=7, height=5, res=300)

plot_ordination(Endo2_up,Endo2_up_PCoA,type="samples",shape="Time",color="Treatment")+labs(title="PCoA Endo2_up_Time_vs_Treatment",x="PCoA1 [63%]",y="PCoA2 [6.1%]")+
  geom_point(size=2)+scale_shape_manual(values = c(8,0,1))+scale_color_manual(values =c('#fce75d','#5d9ffc','#fc5df7' ))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()


    # >>> Endo2_1wk >>>>
    
    Endo2_1wk<-subset_samples(Endo2,Time=="1wk")
    Endo2_1wk_up<-filter_taxa(Endo2_1wk,function(x) sum(x)>1,prune = TRUE)
    Endo2_1wk_up #5879 taxa and 14 samples
    
    # PERMANOVA test for significance 
    
    r_Endo2_1wk_up<-transform_sample_counts(Endo2_1wk_up,function(x) x/sum(x))
    
    set.seed(1013)
    adonis2(t(otu_table(r_Endo2_1wk_up))~Treatment,data=data.frame(sample_data(r_Endo2_1wk_up)),permutations=999,by="margin") # p value = 0.337  # >>>>> Treatment is not significant
    
    
    #>>> PCoA >>>>>
    
    Endo2_1wk_up_bray<-vegdist(t(otu_table(Endo2_1wk_up)),method="bray",binary = FALSE)
    Endo2_1wk_up_PCoA<-ordinate(Endo2_1wk_up,method = "PCoA", Endo2_1wk_up_bray)
    tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Endo2_1wk_up_Treatment.tiff', units="in", width=7, height=5, res=300)
    
    plot_ordination(Endo2_1wk_up,Endo2_1wk_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="PCoA Endo2_1wk_up_Treatment")+geom_point(size=2)+scale_shape_manual(values = c(15,16,17))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
    dev.off()
    
    # >>> Endo2_3wk_up >>>>
    
    Endo2_3wk<-subset_samples(Endo2,Time=="3wk")
    Endo2_3wk # 28114 OTUs across 15 samples
    Endo2_3wk_up<-filter_taxa(Endo2_3wk,function(x) sum(x)>1,prune = TRUE)
    Endo2_3wk_up # 3644 OTUs across 15 samples
    
    # PERMANOVA test
    r_Endo2_3wk_up<-transform_sample_counts(Endo2_3wk_up,function(x) x/sum(x))
    set.seed(1013)
    adonis2(t(otu_table(r_Endo2_3wk_up))~Treatment,data=data.frame(sample_data(r_Endo2_3wk_up)),permutations=999,by="margin") # p=0.509
    
    # >>>>> PCoA plot
    
    r_Endo2_3wk_up_bray<-vegdist(t(otu_table(r_Endo2_3wk_up)),method="bray",binary = FALSE)
    r_Endo2_3wk_up_PCoA<-ordinate(r_Endo2_3wk_up,method = "PCoA", r_Endo2_3wk_up_bray)
    tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Endo2_3wk_up_Treatment.tiff', units="in", width=7, height=5, res=300)
    
    plot_ordination(r_Endo2_3wk_up,r_Endo2_3wk_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="PCoA Endo2_3wk_up_Treatment")+geom_point(size=2)+scale_shape_manual(values = c(15,16,17))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
    dev.off()
    
    ## >>>>>> Endo2_4wk_up >>>>
    
    Endo2_4wk<-subset_samples(Endo2,Time=="4wk")
    Endo2_4wk # 28114 OTUs across 15 samples
    Endo2_4wk_up<-filter_taxa(Endo2_4wk,function(x) sum(x)>1,prune = TRUE)
    Endo2_4wk_up # 3899 OTUs across 14 samples
    
    # PERMANOVA test
    r_Endo2_4wk_up<-transform_sample_counts(Endo2_4wk_up,function(x) x/sum(x))
    set.seed(1013)
    adonis2(t(otu_table(r_Endo2_4wk_up))~Treatment,data=data.frame(sample_data(r_Endo2_4wk_up)),permutations=999,by="margin")  #p value = 0.82
    
    r_Endo2_4wk_up_bray<-vegdist(t(otu_table(r_Endo2_4wk_up)),method="bray",binary = FALSE)
    r_Endo2_4wk_up_PCoA<-ordinate(r_Endo2_4wk_up,method = "PCoA", r_Endo2_4wk_up_bray)
    tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA Endo2_4wk_up_Treatment.tiff', units="in", width=7, height=5, res=300)
    
    plot_ordination(r_Endo2_4wk_up,r_Endo2_4wk_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="PCoA Endo2_4wk_up_Treatment")+
      geom_point(size=2)+scale_shape_manual(values = c(15,16,17))+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7'))+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
    dev.off()
```

## >>> Seed microbiome >>>

```{r}
## SEED

SEED<-subset_samples(Seed_phyloseq_up,Compartment=="SEED")
SEED #144821 taxa and 15 samples
SEED_up<-filter_taxa(SEED,function(x) sum(x)>1,prune = TRUE)
SEED_up # 1030 taxa and 15 samples
r_SEED_up<-transform_sample_counts(SEED_up,function(x) x/sum(x))

## Read depth barplot
SEED_meta<-data.frame(sample_data(r_SEED_up))

tiff("/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/SEED_Read_depth_barplot.tiff",units = 'in',width = 16,height = 8,res = 300,bg='White',compression = 'lzw')
ggplot(data=SEED_meta,aes(x=Sample_ID,y=Read_depth))+geom_bar(width=0.7,stat = "identity") +xlab("Sample_ID")+ylab("Sequencing depth") +
  scale_y_continuous(labels = scales::comma)+theme(panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),text=element_text(family="Arial"), plot.title = element_text(family='Arial', hjust = .5,vjust = 3,size = 24,face="bold"), axis.text.x = element_text(family="Arial",size=8,angle=90,vjust = .6),axis.text.y = element_text(family="Arial",size=14),axis.title.x = element_text(family="Arial",size=20,face="bold"),axis.title.y = element_text(family="Arial",size=20,face="bold"),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

# PERMANOVA test of treatment impacts
set.seed(1013)
adonis2(t(otu_table(r_SEED_up))~Treatment,data=data.frame(sample_data(r_SEED_up)),permutations=999,by="margin") 

#          Df SumOfSqs      R2      F Pr(>F)    
#Treatment  2   1.7254 0.47214 5.3668  0.001 ***
#Residual  12   1.9289 0.52786                  
#Total     14   3.6543 1.00000

# PCoA plot
r_SEED_up_bray<-vegdist(t(otu_table(r_SEED_up)),method="bray",binary = FALSE)
r_SEED_up_PCoA<-ordinate(r_SEED_up,method = "PCoA", r_SEED_up_bray)
tiff('/Users/fangliu/Documents/2018_fugicide_project/SEED_treatment/16S/SEED_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_SEED_up_Treatment.tiff', units="in", width=7, height=5, res=300)

plot_ordination(r_SEED_up,r_SEED_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="r_SEED_up_weighted Bray-Curtis PCoA plot",x="PCoA1 [32.9%]",y="PCoA2 [17.4%]")+
  geom_point(size=2)+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7')) +theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

## Stacked barplot of SEED samples

r_SEED_up #1030 taxa and 15 samples
sample_sums(SEED_up)

SEED_phylum<-tax_glom(r_SEED_up,taxrank = rank_names(r_SEED_up)[2],NArm = TRUE)
SEED_phylum #20 taxa and 15 samples
r_SEED_phylum<-transform_sample_counts(SEED_phylum,function(x) x/sum(x))

filter_r_SEED_phylum<- filter_taxa(r_SEED_phylum, function(x) sum(x>0.0001)>0.2*length(x),prune = TRUE)
filter_r_SEED_phylum # got 9 phyla across 159 sampels left
sample_sums(filter_r_SEED_phylum)
tax_table(filter_r_SEED_phylum)[,2]
Others<-1-sample_sums(filter_r_SEED_phylum)
phylum_SEED<-rbind(otu_table(filter_r_SEED_phylum),Others)
phylum_SEED<-as.data.frame(phylum_SEED)
phylum_SEED[,1:10]
rownames(phylum_SEED)<-c(tax_table(filter_r_SEED_phylum)[,2],"Others")
phylum_SEED
#phylum_rarefied_ST<-data.frame(Treat=as.factor(sample_data(r_phylum_SEED_13021)$Treatment),t(phylum_rarefied_ST))
head(phylum_SEED)
#phylum_SEED<-phylum_SEED%>% group_by(Sample_ID) %>%
#  summarise_all(funs(mean))
phylum_SEED_t<-as.data.frame (t(phylum_SEED))


phylum_SEED_t %>%
  mutate(dominant=phylum_SEED_t%>%
           select(c(Proteobacteria,Acidobacteria,Bacteroidetes,Actinobacteria,Planctomycetes))%>%
           rowSums()) %>% select(dominant)

rowSums(phylum_SEED_t[,1:10])
phylum_SEED_t$Sample_ID<-rownames(phylum_SEED_t)
identical(phylum_SEED_t$Sample_ID,as.character(sample_data(SEED_up)$Sample_ID))
phylum_SEED_t$Treatment<-sample_data(SEED_up)$Treatment
phylum_SEED_t_melt<-melt(phylum_SEED_t,id.vars = c("Sample_ID","Treatment"))
dim(phylum_SEED_t_melt)
head(phylum_SEED_t_melt)
colnames(phylum_SEED_t_melt)<-c('Sample_ID','Treatment','Phylum','Relative_abundance')
phylum_SEED_t_melt

tiff('/Users/fangliu/Documents/2018_fugicide_project/SEED_treatment/16S/SEED_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Phylum level stacked barplot for only SEED samples.tiff', units="in", width=5, height=5, res=300)

phylum_SEED_t_melt$Phylum<-factor(phylum_SEED_t_melt$Phylum,levels=phylum_labels[match(c(as.character(unique(phylum_SEED_t_melt$Phylum))),phylum_labels)])

ggplot(phylum_SEED_t_melt, aes(x =Sample_ID , y = Relative_abundance,fill=Phylum))+geom_bar(stat='identity',size=0.1)+labs(x="Treatment",y="Relative Abundance")+scale_fill_manual(values=phylum_colors[match(c(as.character(unique(phylum_SEED_t_melt$Phylum))),phylum_labels)])+theme(panel.background = element_rect(fill='white',color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",hjust = 0.5,vjust = 3)),axis.text.x = element_text(size = 10,family="Arial"),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=10,family="Arial",angle = 90),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## phylum stacked barplot across treatment - SEED

filter_r_SEED_phylum<- filter_taxa(r_SEED_phylum, function(x) sum(x>0.0001)>0.2*length(x),prune = TRUE)
filter_r_SEED_phylum # got 9 phyla across 159 sampels left
sample_sums(filter_r_SEED_phylum)
tax_table(filter_r_SEED_phylum)[,2]
Others<-1-sample_sums(filter_r_SEED_phylum)
phylum_SEED<-rbind(otu_table(filter_r_SEED_phylum),Others)
phylum_SEED<-as.data.frame(phylum_SEED)
phylum_SEED[,1:10]
rownames(phylum_SEED)<-c(tax_table(filter_r_SEED_phylum)[,2],"Others")
phylum_SEED
identical(as.character(sample_data(filter_r_SEED_phylum)$Sample_ID),colnames(phylum_SEED))

phylum_SEED<-data.frame(Treat=as.factor(sample_data(filter_r_SEED_phylum)$Treatment),t(phylum_SEED))

head(phylum_SEED)
phylum_SEED_grouped<-phylum_SEED%>% group_by(Treat) %>%
  summarise_all(funs(mean))

phylum_SEED_grouped %>%
  mutate(dominant=phylum_SEED_grouped%>%
           select(c(Proteobacteria,Bacteroidetes,Cyanobacteria))%>%
           rowSums()) %>% select(dominant)

rowSums(phylum_SEED_grouped[,2:11])

phylum_SEED_grouped_melt<-melt(phylum_SEED_grouped,id.vars = c("Treat"))
dim(phylum_SEED_grouped_melt)
head(phylum_SEED_grouped_melt)
colnames(phylum_SEED_grouped_melt)<-c('Treat','Phylum','Relative_abundance')
phylum_SEED_grouped_melt

tiff('/Users/fangliu/Documents/2018_fugicide_project/SEED_treatment/16S/SEED_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Phylum level stacked barplot for only SEED samples.tiff', units="in", width=15, height=5, res=300)

phylum_SEED_grouped_melt$Phylum<-factor(phylum_SEED_grouped_melt$Phylum,levels=phylum_labels[match(c(as.character(unique(phylum_SEED_grouped_melt$Phylum))),phylum_labels)])

ggplot(phylum_SEED_grouped_melt, aes(x =Treat, y = Relative_abundance,fill=Phylum))+geom_bar(stat='identity',size=0.1)+labs(x="Treatment",y="Relative Abundance")+scale_fill_manual(values=phylum_colors[match(c(as.character(unique(phylum_SEED_t_melt$Phylum))),phylum_labels)])+theme(panel.background = element_rect(fill='white',color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",hjust = 0.5,vjust = 3)),axis.text.x = element_text(size = 12,family="Arial",angle = 90),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


#-- Summarized all SEED samples by mean

phylum_SEED_mean<-phylum_SEED %>%
  summarise_all(funs(mean))
phylum_SEED_mean$Treat<-"SEED"
phylum_SEED_mean_melt<-melt(phylum_SEED_mean,id.vars = "Treat")
colnames(phylum_SEED_mean_melt)<-c('Treat','Phylum','Relative_abundance')

## Combine non_seed grouped phylum melt file with that of SEED

phylum_group_Seed_melt[1:10,]
phylum_SEED_grouped_melt[1:10,]
phylum_SEED_mean_melt[1:10,]

combine_melt2<-rbind(phylum_group_Seed_melt,phylum_SEED_mean_melt)
combine_melt2$Compartment<-gsub(as.character(combine_melt2$Treat),pattern = "_[134]wk",replacement = "",perl = TRUE)
combine_melt2$Compartment<-factor(combine_melt2$Compartment,levels = c('Soil','Bulk','Rhi','Endo','SEED'))

combine_melt2_soil_and_seed<-combine_melt2[combine_melt2$Compartment=="Soil"|combine_melt2$Compartment=="SEED",]
combine_melt2_soil_and_seed$Relative_abundanceup<-combine_melt2_soil_and_seed$Relative_abundance

combine_melt2_bulk_rhi_endo<-combine_melt2[combine_melt2$Compartment=="Bulk"|combine_melt2$Compartment=="Rhi"|combine_melt2$Compartment=="Endo",]
combine_melt2_bulk_rhi_endo$Relative_abundanceup<-combine_melt2_bulk_rhi_endo$Relative_abundance/3
combine_melt2up<-rbind(combine_melt2_soil_and_seed,combine_melt2_bulk_rhi_endo)

## between Treats

tiff('/Users/fangliu/Documents/2018_fugicide_project/SEED_treatment/16S/SEED_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Phylum level stacked barplot combine all samples between Treats.tiff', units="in", width=7, height=5, res=300)

combine_melt2$Phylum<-factor(combine_melt2$Phylum,levels=phylum_labels[match(c(as.character(unique(combine_melt2$Phylum))),phylum_labels)])

ggplot(combine_melt2, aes(x =Treat, y = Relative_abundance,fill=Phylum))+geom_bar(stat='identity',size=0.1)+labs(x="Treatment",y="Relative Abundance")+scale_fill_manual(values=phylum_colors[match(c(as.character(unique(combine_melt2$Phylum))),phylum_labels)])+theme(panel.background = element_rect(fill='white',color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",hjust = 0.5,vjust = 3)),axis.text.x = element_text(size = 12,family="Arial",angle = 90),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## between compartment

tiff('/Users/fangliu/Documents/2018_fugicide_project/SEED_treatment/16S/SEED_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Phylum level stacked barplot combine all samples between compartment.tiff', units="in", width=7, height=5, res=300)

combine_melt2up$Phylum<-factor(combine_melt2up$Phylum,levels=phylum_labels[match(c(as.character(unique(combine_melt2up$Phylum))),phylum_labels)])

ggplot(combine_melt2up, aes(x =Compartment, y = Relative_abundanceup,fill=Phylum))+geom_bar(stat='identity',size=0.1)+labs(x="Treatment",y="Relative Abundance")+scale_fill_manual(values=phylum_colors[match(c(as.character(unique(combine_melt2up$Phylum))),phylum_labels)])+theme(panel.background = element_rect(fill='white',color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",hjust = 0.5,vjust = 3)),axis.text.x = element_text(size = 12,family="Arial",angle = 90),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## SEED_genus

sample_sums(SEED_up)
SEED_genus<-tax_glom(SEED_up,taxrank = rank_names(SEED_up)[6],NArm = TRUE)
SEED_genus
sort(taxa_sums(SEED_genus),decreasing = TRUE)[1:10]
SEED_genus_count<-data.frame(tax_table(SEED_genus),otu_table(SEED_genus))
#write.csv(SEED_genus_count,file = "SEED_genus_count.csv")

## SEED_family
SEED_family<-tax_glom(SEED_up,taxrank = rank_names(SEED_up)[5],NArm = TRUE)
SEED_family
sort(taxa_sums(SEED_family),decreasing = TRUE)[1:10]
SEED_family_count<-data.frame(tax_table(SEED_family)[,1:5],otu_table(SEED_family))
SEED_family_count[c(labels(sort(taxa_sums(SEED_family),decreasing = TRUE)[1:10])),1:5]
#write.csv(SEED_ unt,file = "SEED_family_count.csv")

## Top 10 family
# Enterobacteriaceae, Pseudomonadaceaem Sphingomonadaceae, Beijerinckiaceae, Moraxellaceae, Rhizobiaceae, Xanthobacteraceae,Weeksellaceae, Rhizobiales_Incertae_Sedis


## Combine SEED diversity with that of nonseed

nonseed_Diversity<-data.frame(estimate_richness(Seed_13021_up,split=TRUE,measures = NULL),sample_data(Seed_13021_up))
SEED_diversity<-data.frame(estimate_richness(SEED,split = TRUE,measures = NULL),sample_data(SEED))
identical(colnames(nonseed_Diversity),colnames(SEED_diversity))
Diversity_all<-rbind(nonseed_Diversity,SEED_diversity)
dim(Diversity_all)

# Diversity between compartments

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Shannon diversity between compartments including SEED.tiff', units="in", width=5.5, height=4.5, res=300)

boxplot(Shannon~Compartment, data=Diversity_all, main="Diversity between compartments across all samples") 

ggplot(Diversity_all,aes(x=Compartment,y=Shannon,fill=Compartment))+geom_boxplot(size=0.5)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Compartment))+scale_fill_manual(values = as.character(Treat_colors),name="Treat")+labs(title="Shannon Diversity",x="Treatment",y="Shannon Index")+ylim(2.5,7.5)+scale_x_discrete(labels=c('Soil'='Soil','Bulk'='Bulk','Rhizosphere'='Rhi','Endosphere'='Endo','Seed'='SEED')) +theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=11,family="Arial"),axis.text.x = element_text(family="Arial",size=11),
    axis.title.x=element_text(size = 13,face="bold",family="Arial",vjust = 1,margin = margin(t = 5, r = 0, b = 0, l = 0)),axis.title.y=element_text(size = 13,face="bold",family="Arial",vjust = 1,margin = margin(t = 0, r = 5, b = 0, l = 0)),legend.title = element_text(size=11,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Diversity_all anova between compartment

# First, test the normality of dependence
res<-resid(lm(Shannon~Compartment,data=Diversity_all))
shapiro.test(res) # p-value = 5.35e-13
plot(density(Diversity$Shannon))

# Pairwise comparison
kruskal.test(Shannon~Compartment, data=Diversity_all)
dunn.test::dunn.test(x = Diversity_all$Shannon,g = Diversity_all$Compartment,method ='bh',kw = TRUE,label = TRUE,wrap = TRUE,table = TRUE,alpha = 0.1)


# Diversity_all  between treats

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Shannon diversity between treats across all samples.tiff', units="in", width=5.5, height=4.5, res=300)

boxplot(Shannon~Treat, data=Diversity_all, main="Diversity between treats across all samples") 
Diversity_all$Treat<-factor(Diversity_all$Treat,levels=unique(Diversity_all$Treat)[c(10,1:3,7:9,4:6,11)])

ggplot(Diversity_all,aes(x=Treat,y=Shannon,fill=Treat))+geom_boxplot(size=0.5)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),'#ffee32'),name="Treat")+labs(title="Shannon Diversity",x="Treatment",y="Shannon Index")+ylim(2.5,7.5)+scale_x_discrete(labels=c('Seed'='SEED'))+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=11,family="Arial"),axis.text.x = element_text(family="Arial",size=11,angle = 90),axis.title.x=element_text(size = 13,face="bold",family="Arial",vjust = 1,margin = margin(t = 5, r = 0, b = 0, l = 0)),axis.title.y=element_text(size = 13,face="bold",family="Arial",vjust = 1,margin = margin(t = 0, r = 5, b = 0, l = 0)),legend.title = element_text(size=11,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

# Anova Diversity_all between treats

# First, test normality between samples
res<-resid(lm(Shannon~Treat,data=Diversity_all))
shapiro.test(res) # p-value = 5.35e-13
plot(density(Diversity_all$Shannon))

# Pairwise comparison
kruskal.test(Shannon~Treat, data=Diversity_all)
dunn.test::dunn.test(x = Diversity_all$Shannon,g = Diversity_all$Treat,method ='bh',kw = TRUE,label = TRUE,wrap = TRUE,table = TRUE,alpha = 0.1)
```


## SEED rarefied

```{r}
min(sample_sums(SEED))
rf_SEED_up<-rarefy_even_depth(SEED_up,sample.size = min(sample_sums(SEED_up)),rngseed = 1013,replace = FALSE,trimOTUs = TRUE)
r_rf_SEED_up<-transform_sample_counts(rf_SEED_up,function(x) x/sum(x))

# PERMANOVA test of treatment impacts

set.seed(1013)
adonis2(t(otu_table(r_rf_SEED_up))~Treatment,data=data.frame(sample_data(r_rf_SEED_up)),permutations=999,by="margin") 

#          Df SumOfSqs     R2      F Pr(>F)    
#Treatment  2   1.6823 0.4434 4.7798  0.001 ***
#Residual  12   2.1118 0.5566                  
#Total     14   3.7941 1.0000

## Read depth
set.seed(1013)
adonis2(t(otu_table(r_rf_SEED_up))~Read_depth,data=data.frame(sample_data(r_rf_SEED_up)),permutations=999,by="margin") 

#          Df SumOfSqs     R2      F Pr(>F)    
#Treatment  2   1.6823 0.4434 4.7798  0.001 ***
#Residual  12   2.1118 0.5566                  
#Total     14   3.7941 1.0000


## Read_depth vs Treatment impact
set.seed(1013)
adonis2(t(otu_table(r_rf_SEED_up))~Read_depth+Treatment,data=data.frame(sample_data(r_rf_SEED_up)),permutations=999,by="margin") 

# PCoA plot
r_rf_SEED_up_bray<-vegdist(t(otu_table(r_rf_SEED_up)),method="bray",binary = FALSE)
r_rf_SEED_up_PCoA<-ordinate(r_rf_SEED_up,method = "PCoA", r_rf_SEED_up_bray)
tiff('/Users/fangliu/Documents/2018_fugicide_project/SEED_treatment/16S/SEED_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/PCoA r_rf_SEED_up_Treatment.tiff', units="in", width=7, height=5, res=300)

plot_ordination(r_rf_SEED_up,r_rf_SEED_up_PCoA,type="samples",color="Treatment",label="Sample_ID")+labs(title="r_rf_SEED_up_weighted Bray-Curtis PCoA plot",x="PCoA1 [32.9%]",y="PCoA2 [17.4%]")+
  geom_point(size=2)+scale_color_manual(values = c('#fce75d','#5d9ffc','#fc5df7')) +theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

## Stacked barplot of SEED samples

r_rf_SEED_up # 432 taxa and 15 samples
sample_sums(r_rf_SEED_up)

rf_SEED_phylum<-tax_glom(r_rf_SEED_up,taxrank = rank_names(r_rf_SEED_up)[2],NArm = TRUE)
rf_SEED_phylum #17 taxa and 15 samples
r_rf_SEED_phylum<-transform_sample_counts(rf_SEED_phylum,function(x) x/sum(x))

filter_r_rf_SEED_phylum<- filter_taxa(r_rf_SEED_phylum, function(x) sum(x>0.0001)>0.2*length(x),prune = TRUE)
filter_r_rf_SEED_phylum # got 8 phyla across 159 sampels left
sample_sums(filter_r_rf_SEED_phylum)
tax_table(filter_r_rf_SEED_phylum)[,2]
Others<-1-sample_sums(filter_r_rf_SEED_phylum)
phylum_rf_SEED<-rbind(otu_table(filter_r_rf_SEED_phylum),Others)
phylum_rf_SEED<-as.data.frame(phylum_rf_SEED)
phylum_rf_SEED[,1:10]
rownames(phylum_rf_SEED)<-c(tax_table(filter_r_rf_SEED_phylum)[,2],"Others")
phylum_rf_SEED
#phylum_rarefied_ST<-data.frame(Treat=as.factor(sample_data(r_phylum_rf_SEED_13021)$Treatment),t(phylum_rarefied_ST))
head(phylum_rf_SEED)
#phylum_rf_SEED<-phylum_rf_SEED%>% group_by(Sample_ID) %>%
#  summarise_all(funs(mean))
phylum_rf_SEED_t<-as.data.frame (t(phylum_rf_SEED))


phylum_rf_SEED_t %>%
  mutate(dominant=phylum_rf_SEED_t%>%
           select(c(Proteobacteria,Acidobacteria,Bacteroidetes,Actinobacteria,Planctomycetes))%>%
           rowSums()) %>% select(dominant)

phylum_rf_SEED_t$Sample_ID<-rownames(phylum_rf_SEED_t)
identical(phylum_rf_SEED_t$Sample_ID,as.character(sample_data(rf_SEED_up)$Sample_ID))
phylum_rf_SEED_t$Treatment<-sample_data(rf_SEED_up)$Treatment
phylum_rf_SEED_t_melt<-melt(phylum_rf_SEED_t,id.vars = c("Sample_ID","Treatment"))
dim(phylum_rf_SEED_t_melt)
head(phylum_rf_SEED_t_melt)
colnames(phylum_rf_SEED_t_melt)<-c('Sample_ID','Treatment','Phylum','Relative_abundance')
phylum_rf_SEED_t_melt


tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Phylum level stacked barplot for only rf_SEED samples.tiff', units="in", width=15, height=5, res=300)

phylum_rf_SEED_t_melt$Phylum<-factor(phylum_rf_SEED_t_melt$Phylum,levels=phylum_labels[match(c(as.character(unique(phylum_rf_SEED_t_melt$Phylum))),phylum_labels)])

ggplot(phylum_rf_SEED_t_melt, aes(x =Sample_ID , y = Relative_abundance,fill=Phylum))+geom_bar(stat='identity',size=0.1)+labs(x="Treatment",y="Relative Abundance")+scale_fill_manual(values=phylum_colors[match(c(as.character(unique(phylum_rf_SEED_t_melt$Phylum))),phylum_labels)])+theme(panel.background = element_rect(fill='white',color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",hjust = 0.5,vjust = 3)),axis.text.x = element_text(size = 6,family="Arial"),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=6,family="Arial",angle = 90),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()
# -- top phylum
# Proteobacteria, Actinobacteria, Acidobacteria, Cyanobacteria and Firmicutes

## rf_SEED_genus

sample_sums(rf_SEED_up)
rf_SEED_genus<-tax_glom(rf_SEED_up,taxrank = rank_names(rf_SEED_up)[6],NArm = TRUE)
rf_SEED_genus
sort(taxa_sums(rf_SEED_genus),decreasing = TRUE)[1:10]
rf_SEED_genus_count<-data.frame(tax_table(rf_SEED_genus),otu_table(rf_SEED_genus))
#write.csv(rf_SEED_genus_count,file = "rf_SEED_genus_count.csv")

rf_SEED_genus_tax_top10<-data.frame(tax_table(rf_SEED_genus))[match(c(labels(sort(taxa_sums(rf_SEED_genus),decreasing = TRUE))[1:10]),taxa_names(rf_SEED_genus)),]
rf_SEED_genus_tax_top10
#labels(sort(taxa_sums(rf_SEED_genus),decreasing = TRUE)[1:10])

## rf_SEED_family
rf_SEED_family<-tax_glom(rf_SEED_up,taxrank = rank_names(rf_SEED_up)[5],NArm = TRUE)
rf_SEED_family
sort(taxa_sums(rf_SEED_family),decreasing = TRUE)[1:10]
rf_SEED_family_count<-data.frame(tax_table(rf_SEED_family)[,1:5],otu_table(rf_SEED_family))
rf_SEED_family_count[c(labels(sort(taxa_sums(rf_SEED_family),decreasing = TRUE)[1:10])),1:5]
#write.csv(rf_SEED_family_count,file = "rf_SEED_family_count.csv")

rf_SEED_family_tax_top10<-data.frame(tax_table(rf_SEED_family)[,1:5])[match(c(labels(sort(taxa_sums(rf_SEED_family),decreasing = TRUE))[1:10]),taxa_names(rf_SEED_family)),]
rf_SEED_family_tax_top10
sort(taxa_sums(rf_SEED_family),decreasing = TRUE)[1:10]

## Top 10 family
# Enterobacteriaceae, Pseudomonadaceaem Sphingomonadaceae, Beijerinckiaceae, Moraxellaceae, Rhizobiaceae, Xanthobacteraceae,Weeksellaceae, Rhizobiales_Incertae_Sedis
```

## Visualized bray-curtis disimilarity between samples across all treat

```{r}
str(r_Bulk_1wk_up_bray)
r_Bulk_1wk_up_bray_matrix<-data.matrix(r_Bulk_1wk_up_bray,rownames.force = NA)
#melt_Bulk_1wk_up_bray_matrix<-melt (Bulk_1wk_up_bray_matrix)
#head(melt_Bulk_1wk_up_bray_matrix)
#ggplot(data=melt_Bulk_1wk_up_bray_matrix,aes(x=Var1,y=Var2,fill=value))+geom_tile()
r_Bulk_3wk_up_bray_matrix<-data.matrix(r_Bulk_3wk_up_bray,rownames.force = NA)
r_Bulk_4wk_up_bray_matrix<-data.matrix(r_Bulk_4wk_up_bray,rownames.force = NA)

r_Rhi_1wk_up_bray_matrix<-data.matrix(r_Rhi_1wk_up_bray,rownames.force = NA)
r_Rhi_3wk_up_bray_matrix<-data.matrix(r_Rhi_3wk_up_bray,rownames.force = NA)
r_Rhi_4wk_up_bray_matrix<-data.matrix(r_Rhi_4wk_up_bray,rownames.force = NA)

r_Endo_1wk_up_bray_matrix<-data.matrix(r_Endo_1wk_up_bray,rownames.force = NA)
r_Endo_3wk_up_bray_matrix<-data.matrix(r_Endo_3wk_up_bray,rownames.force = NA)
r_Endo_4wk_up_bray_matrix<-data.matrix(r_Endo_4wk_up_bray,rownames.force = NA)

Seed_13021_up_bray_matrix<-data.matrix(Seed_13021_up_bray,rownames.force = NA)
r_BRE_up_bray_matrix<-data.matrix(r_BRE_up_bray,rownames.force = NA)
Bulk_up_bray_matrix<-data.matrix(Bulk_up_bray,rownames.force = NA)
Rhi_up_bray_matrix<-data.matrix(Rhi_up_bray,rownames.force = NA)
Endo_up_bray_matrix<-data.matrix(Endo_up_bray,rownames.force = NA)
Soil_up_bray_matrix<-data.matrix(Soil_up_bray,rownames.force = NA)
mean(c(Soil_up_bray_matrix)[(c(Soil_up_bray_matrix)!=0)]) # 0.41447

Seed_phyloseq_up
Seed_up_bray_matrix<-data.matrix(Seed_up_bray,rownames.force = NA)
dim(Seed_up_bray_matrix)

library(pheatmap)

## BER bray curtis distance heatmap 

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bray-curtis disdance based heatmap BRE.tiff', units="in", width=25, height=20, res=300)
pheatmap(r_BRE_up_bray_matrix,color = colorRampPalette(c("navy", "white", "firebrick3"))(100),annotation_col = data.frame(sample_data(r_BRE_up))[,c(3,2)],legend = TRUE,legend_labels = "Bray_curtis Distance",annotation_legend = TRUE,fontsize = 35,fontsize_row = 15,fontsize_col = 15,cellwidth = 7.5,cellheight = 7.5,border_color = NA,main = "Bacteria Dissimilarity heatmap between Compartments")
dev.off()

## Seed_13021 bray curtis distance heatmap

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bray-curtis disdance based heatmap 13021 rarefied samples.tiff', units="in", width=25, height=20, res=300)
pheatmap(Seed_13021_up_bray_matrix,color = colorRampPalette(c("navy", "white", "firebrick3"))(100),annotation_col = data.frame(sample_data(Seed_13021_up))[,c(3,2)],legend = TRUE,legend_labels = "Bray_curtis Distance",annotation_legend = TRUE,fontsize = 25,fontsize_row = 12,fontsize_col = 12,cellwidth = 8,cellheight = 8,border_color = NA,main = "Bacteria Dissimilarity heatmap between all non_seed samples")
dev.off()

## all sample bray curtis distance heatmap
annotation_colors<-list(
  Time=c(Presowing='#dba7a7',"1wk"='#91c989',"3wk"='#79b8b4',"4wk"='#b17bb8',SEED='#ABAD29'),
  Compartment=c(Soil='#CD9BCD',Bulk='#C84248',Rhizosphere="orange",Endosphere='#00CDCD',SEED='#ABAD29')
)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bray-curtis disdance based heatmap all samples.tiff', units="in", width=30, height=22, res=300)
test<-pheatmap(Seed_up_bray_matrix,color = colorRampPalette(c("navy", "white", "firebrick3"))(100),annotation_col = data.frame(sample_data(Seed_phyloseq_up))[,c(3,2)],annotation_colors = annotation_colors,legend = TRUE,legend_labels = "Bray_curtis Distance",annotation_legend = TRUE,fontsize = 25,fontsize_row = 12,fontsize_col = 12,cellwidth = 8,cellheight = 8,border_color = NA,main = "Bacteria Dissimilarity heatmap between all samples")
dev.off()

pheatmap(Bulk_up_bray_matrix)
pheatmap(Endo_up_bray_matrix)
pheatmap(Rhi_up_bray_matrix)
```

## Generate a Bray-curtis distance dataframe and export used to do correlation analysis with ITS data

```{r}
dim(Seed_13021_up_bray_matrix)
colnames(Seed_13021_up_bray_matrix)
Seed_13021_up_bray_matrix_melt<-melt(Seed_13021_up_bray_matrix)
head(Seed_13021_up_bray_matrix_melt)
colnames(Seed_13021_up_bray_matrix_melt)<-c("B_row","B_col","B_Bray_curtis_distance")
Seed_13021_up_bray_matrix_melt$pair<-paste(Seed_13021_up_bray_matrix_melt$B_row,Seed_13021_up_bray_matrix_melt$B_col,sep = "-")
dim(Seed_13021_up_bray_matrix_melt)
#write.csv(Seed_13021_up_bray_matrix_melt,file = "/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Bacteria_Seed_13021_up_bray_matrix_melt.csv")
```

## Calculate the correlation between bacteria and fungi beta diversity

```{r}
bacteria_beta_div<-read.csv(file="/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Bacteria_Seed_13021_up_bray_matrix_melt.csv",row.names = 1)
dim(bacteria_beta_div) #22201 rows and 4 columns

fungi_beta_div<-read.csv(file="/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Fungi rms_non_Seed_up_bray_matrix_melt.csv",row.names = 1)
dim(fungi_beta_div) # 22201 rows and 4 columns


bacteria_beta_div$pair[1:5]
fungi_beta_div$pair[1:5]
identical(bacteria_beta_div$pair,fungi_beta_div$pair)

beta_div_join<-join(bacteria_beta_div,fungi_beta_div,by="pair")
head(beta_div_join)
# Add compartment information
beta_div_join$Compartment<-rep("Cross_Compartment",dim(beta_div_join)[1])
beta_div_join$Compartment[grep("*S_.*-*S_*",beta_div_join$pair)]="Within_Soil"
beta_div_join$Compartment[grep("*B_.*-*B_*",beta_div_join$pair)]="Within_Bulk"
beta_div_join$Compartment[grep("*R_.*-*R_*",beta_div_join$pair)]="Within_Rhizosphere"
beta_div_join$Compartment[grep("*E_.*-*E_*",beta_div_join$pair)]="Within_Endosphere"
beta_div_join$Compartment[c(grep("*S_.*-*B_*",beta_div_join$pair),grep("*B_.*-*S_*",beta_div_join$pair))]="Soil_vs_Bulk"
beta_div_join$Compartment[c(grep("*S_.*-*R_*",beta_div_join$pair),grep("*R_.*-*S_*",beta_div_join$pair))]="Soil_vs_Rhizosphere"
beta_div_join$Compartment[c(grep("*S_.*-*E_*",beta_div_join$pair),grep("*E_.*-*S_*",beta_div_join$pair))]="Soil_vs_Endosphere"
beta_div_join$Compartment[c(grep("*B_.*-*R_*",beta_div_join$pair),grep("*R_.*-*B_*",beta_div_join$pair))]="Bulk_vs_Rhizosphere"
beta_div_join$Compartment[c(grep("*B_.*-*E_*",beta_div_join$pair),grep("*E_.*-*B_*",beta_div_join$pair))]="Bulk_vs_Endosphere"
beta_div_join$Compartment[c(grep("*R_.*-*E_*",beta_div_join$pair),grep("*E_.*-*R_*",beta_div_join$pair))]="Rhizosphere_vs_Endosphere"

#Add Time information
beta_div_join$Time<-rep("Cross_Weeks",dim(beta_div_join)[1])
beta_div_join$Time[grep("*S_.*-*S_*",beta_div_join$pair)]="Pre_sowing"
beta_div_join$Time[grep("*_1w_.*_1w*",beta_div_join$pair)]="Within_Week1"
beta_div_join$Time[grep("*_3w_.*_3w*",beta_div_join$pair)]="Within_Week3"
beta_div_join$Time[grep("*_4w_.*_4w*",beta_div_join$pair)]="Within_Week4"
beta_div_join$Time[c(grep("*S_.*_1w*",beta_div_join$pair),grep("*_1w.*S_*",beta_div_join$pair))]="Pre_sowing_vs_Week1"
beta_div_join$Time[c(grep("*S_.*_3w*",beta_div_join$pair),grep("*_3w.*S_*",beta_div_join$pair))]="Pre_sowing_vs_Week3"
beta_div_join$Time[c(grep("*S_.*_4w*",beta_div_join$pair),grep("*_4w.*S_*",beta_div_join$pair))]="Pre_sowing_vs_Week4"
beta_div_join$Time[c(grep("*_1w.*_3w*",beta_div_join$pair),grep("*_3w.*_1w*",beta_div_join$pair))]="Week1_vs_Week3"
beta_div_join$Time[c(grep("*_1w.*_4w*",beta_div_join$pair),grep("*_4w.*_1w*",beta_div_join$pair))]="Week1_vs_Week4"
beta_div_join$Time[c(grep("*_3w.*_4w*",beta_div_join$pair),grep("*_4w.*_3w*",beta_div_join$pair))]="Week3_vs_Week4"

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bray-curtis disdance scatter plot of Bulk.tiff', units="in", width=8, height=5, res=300)
ggplot(data=beta_div_join[c(beta_div_join$B_Bray_curtis_distance!=0&c(beta_div_join$Compartment=="Within_Bulk"|beta_div_join$Time=="Pre_sowing")),],aes(x=B_Bray_curtis_distance,y=F_Bray_curtis_distance,color=Time))+xlab("\n Bacteria Bray_Curtis distance")+ylab("Fungi Bray_Curtis distance \n")+labs(title="Beta diversity of Bulk Compartment")+geom_point(size=1)+scale_color_manual(values = c("grey","#C84284","#00CDCD","orange","#16ab05","#756568","#d02dfc"))+xlim(0,1)+ylim(0,1)+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bray-curtis disdance scatter plot Rhizosphere.tiff', units="in", width=8, height=5, res=300)
ggplot(data=beta_div_join[c(beta_div_join$B_Bray_curtis_distance!=0&c(beta_div_join$Compartment=="Within_Rhizosphere"|beta_div_join$Time=="Pre_sowing")),],aes(x=B_Bray_curtis_distance,y=F_Bray_curtis_distance,color=Time))+xlab("\n Bacteria Bray_Curtis distance")+ylab("Fungi Bray_Curtis distance \n")+labs(title="Beta diversity of Rhizosphere Compartment")+geom_point(size=1)+scale_color_manual(values = c("grey","#C84284","#00CDCD","orange","#16ab05","#756568","#d02dfc"))+xlim(0,1)+ylim(0,1)+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()


tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bray-curtis disdance scatter plot Endosphere.tiff', units="in", width=8, height=5, res=300)
ggplot(data=beta_div_join[c(beta_div_join$B_Bray_curtis_distance!=0&c(beta_div_join$Compartment=="Within_Endosphere"|beta_div_join$Time=="Pre_sowing")),],aes(x=B_Bray_curtis_distance,y=F_Bray_curtis_distance,color=Time))+xlab("\n Bacteria Bray_Curtis distance")+ylab("Fungi Bray_Curtis distance \n")+labs(title="Beta diversity of Endosphere Compartment")+geom_point(size=1)+scale_color_manual(values = c("grey","#C84284","#00CDCD","orange","#16ab05","#756568","#d02dfc"))+xlim(0,1)+ylim(0,1)+theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=13,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()


tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bray-curtis disdance scatter plot week1.tiff', units="in", width=8, height=5, res=300)
ggplot(data=beta_div_join[c(beta_div_join$B_Bray_curtis_distance!=0&c(beta_div_join$Time=="Within_Week1"|beta_div_join$Time=="Pre_sowing")),],aes(x=B_Bray_curtis_distance,y=F_Bray_curtis_distance,color=Compartment))+labs(title="Beta diversity of Week1 and pre-sowing samples")+geom_point(size=0.8)+scale_color_manual(values = c("#C84284","#00CDCD","orange","#16ab05","#756568","#d02dfc","grey"))+xlim(0,1)+ylim(0,1)+guides(colour = guide_legend(override.aes = list(size=2)))+xlab("\n Bacteria Bray_Curtis distance")+ylab("Fungi Bray_Curtis distance \n")+ theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=15,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bray-curtis disdance scatter plot week3.tiff', units="in", width=8, height=5, res=300)
ggplot(data=beta_div_join[c(beta_div_join$B_Bray_curtis_distance!=0&c(beta_div_join$Time=="Within_Week3"|beta_div_join$Time=="Pre_sowing")),],aes(x=B_Bray_curtis_distance,y=F_Bray_curtis_distance,color=Compartment))+labs(title="Beta diversity of Week3 and pre-sowing samples")+geom_point(size=0.8)+scale_color_manual(values = c("#C84284","#00CDCD","orange","#16ab05","#756568","#d02dfc","grey"))+xlim(0,1)+ylim(0,1)+guides(colour = guide_legend(override.aes = list(size=2)))+xlab("\n Bacteria Bray_Curtis distance")+ylab("Fungi Bray_Curtis distance \n")+ theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=15,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bray-curtis disdance scatter plot week4.tiff', units="in", width=8, height=5, res=300)
ggplot(data=beta_div_join[c(beta_div_join$B_Bray_curtis_distance!=0&c(beta_div_join$Time=="Within_Week4"|beta_div_join$Time=="Pre_sowing")),],aes(x=B_Bray_curtis_distance,y=F_Bray_curtis_distance,color=Compartment))+labs(title="Beta diversity of Week4 and pre-sowing samples")+geom_point(size=0.8)+scale_color_manual(values = c("#C84284","#00CDCD","orange","#16ab05","#756568","#d02dfc","grey"))+xlim(0,1)+ylim(0,1)+guides(colour = guide_legend(override.aes = list(size=2)))+xlab("\n Bacteria Bray_Curtis distance")+ylab("Fungi Bray_Curtis distance \n")+ theme(panel.background = element_rect(fill = "transparent"), plot.background = element_rect(fill = "transparent", color = NA),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=15,family = "Arial",face="bold",vjust=3,hjust = .5)),text=element_text(family = "Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text=element_text(size=13,family="Arial"),
    axis.title=element_text(size = 15,face="bold",family = "Arial"),legend.title = element_text(size=15,face="bold",family = "Arial"),legend.text = (element_text(size=10,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin=unit(c(1,1,1,1),"cm"))
dev.off()

##  -----Bray-curtis dopt plot ---------- Bacteria verus Fungi

## ---------Bacteria---------

# Bacteria between weeks and within weeks

dim(beta_div_join)
beta_div_join[1:10,]
bacteria_beta_div_up<-beta_div_join[,c(1,2,3,10,11)]
head(bacteria_beta_div_up)
bacteria_beta_div_up_within_compartment<-bacteria_beta_div_up[grep('Within',bacteria_beta_div_up$Compartment),]
head(bacteria_beta_div_up_within_compartment)
bacteria_beta_div_up_within_compartment<-bacteria_beta_div_up_within_compartment[bacteria_beta_div_up_within_compartment$B_Bray_curtis_distance!=0,]
bacteria_beta_div_up_within_compartment$Compartment<-factor(bacteria_beta_div_up_within_compartment$Compartment,levels = c('Within_Soil','Within_Bulk','Within_Rhizosphere','Within_Endosphere'))

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bacteria Bray-curtis diversity between samples - between weeks and within weeks.tiff', units="in", width=7, height=10, res=300)

ggplot(bacteria_beta_div_up_within_compartment,aes(x=Compartment,y=B_Bray_curtis_distance))+
geom_jitter(shape=16, position=position_jitter(width = 0.3,height = 0),aes(color=Time))+scale_color_manual(values = c(as.character(Treat_colors),'#dd4632'),name="Time Group")+labs(title="Between sample dissimilarity of bacteria",x="Compartment",y="Bray-Curtis dissimilarity")+scale_x_discrete(labels=c('Within_Bulk'='Bulk','Within_Endosphere'='Endo','Within_Rhizosphere'='Rhi','Within_Soil'='Soil'))+ylim(0,1)+scale_y_continuous(limits=c(0,1))+theme(legend.position="right",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=20,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=15,family="Arial"),axis.text.x = element_text(family="Arial",size=15),
    axis.title.x=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 10, r = 0, b = 0, l = 0)),axis.title.y=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 0, r = 10, b = 0, l = 0)),legend.key.size = unit(1.2,'line'),legend.title = element_text(size=18,face="bold",family = "Arial"),legend.text = (element_text(size=15,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bacteria Bray-curtis diversity between samples - between weeks and within weeks-2.tiff', units="in", width=10, height=10, res=300)

ggplot(bacteria_beta_div_up_within_compartment,aes(x=Compartment,y=B_Bray_curtis_distance))+
geom_jitter(shape=16, position=position_jitter(width = 0.3,height = 0),aes(color=Time))+scale_color_manual(values = c(as.character(Treat_colors),'#dd4632'),name="Time Group")+labs(title="Between sample dissimilarity of bacteria",x="Compartment",y="Bray-Curtis dissimilarity")+scale_x_discrete(labels=c('Within_Bulk'='Bulk','Within_Endosphere'='Endo','Within_Rhizosphere'='Rhi','Within_Soil'='Soil'))+ylim(0,1)+scale_y_continuous(limits=c(0,1))+theme(legend.position="right",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=20,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=15,family="Arial"),axis.text.x = element_text(family="Arial",size=15),
    axis.title.x=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 10, r = 0, b = 0, l = 0)),axis.title.y=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 0, r = 10, b = 0, l = 0)),legend.key.size = unit(1.2,'line'),legend.title = element_text(size=18,face="bold",family = "Arial"),legend.text = (element_text(size=15,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()
## Bacteria Only within weeks

bacteria_beta_div_up_within_compartment_withinwk<-bacteria_beta_div_up_within_compartment[-c(grep('vs',bacteria_beta_div_up_within_compartment$Time)),]

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bacteria Bray-curtis diversity between samples - Within weeks.tiff', units="in", width=7, height=10, res=300)

ggplot(bacteria_beta_div_up_within_compartment_withinwk,aes(x=Compartment,y=B_Bray_curtis_distance))+
geom_jitter(shape=16, position=position_jitter(width = 0.3,height = 0),aes(color=Time))+scale_color_manual(values = c(as.character(Treat_colors),'#dd4632'),name="Time Group")+labs(title="Between sample dissimilarity of bacteria",x="Compartment",y="Bray-Curtis dissimilarity")+scale_x_discrete(labels=c('Within_Bulk'='Bulk','Within_Endosphere'='Endo','Within_Rhizosphere'='Rhi','Within_Soil'='Soil'))+ylim(0,1)+scale_y_continuous(limits=c(0,1))+theme(legend.position="right",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=20,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=15,family="Arial"),axis.text.x = element_text(family="Arial",size=15),
    axis.title.x=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 10, r = 0, b = 0, l = 0)),axis.title.y=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 0, r = 10, b = 0, l = 0)),legend.key.size = unit(1.2,'line'),legend.title = element_text(size=18,face="bold",family = "Arial"),legend.text = (element_text(size=15,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Bacteria only between weeks

bacteria_beta_div_up_within_compartment_betweenwk<-bacteria_beta_div_up_within_compartment[grep('vs',bacteria_beta_div_up_within_compartment$Time),]

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bacteria Bray-curtis diversity between samples - Between weeks.tiff', units="in", width=7, height=10, res=300)

ggplot(bacteria_beta_div_up_within_compartment_betweenwk,aes(x=Compartment,y=B_Bray_curtis_distance))+
geom_jitter(shape=16, position=position_jitter(width = 0.3,height = 0),aes(color=Time))+scale_color_manual(values = c(as.character(Treat_colors),'#dd4632'),name="Time Group")+labs(title="Between sample dissimilarity of bacteria",x="Compartment",y="Bray-Curtis dissimilarity")+scale_x_discrete(labels=c('Within_Bulk'='Bulk','Within_Endosphere'='Endo','Within_Rhizosphere'='Rhi','Within_Soil'='Soil'))+ylim(0,1)+scale_y_continuous(limits=c(0,1))+theme(legend.position="right",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=20,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=15,family="Arial"),axis.text.x = element_text(family="Arial",size=15),
    axis.title.x=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 10, r = 0, b = 0, l = 0)),axis.title.y=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 0, r = 10, b = 0, l = 0)),legend.key.size = unit(1.2,'line'),legend.title = element_text(size=18,face="bold",family = "Arial"),legend.text = (element_text(size=15,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Bacteria beta diversity - compare between plots difference - Week1, Week3 and Week4

dim(beta_div_join)
beta_div_join[1:10,]
bacteria_beta_div_up2<-beta_div_join[,c(1,2,3,10,11)]
head(bacteria_beta_div_up2)
bacteria_beta_div_up2_within_Time1<-bacteria_beta_div_up2[grep('Within',bacteria_beta_div_up2$Time),]
bacteria_beta_div_up2_within_Time2<-bacteria_beta_div_up2[-c(grep('Week',bacteria_beta_div_up2$Time)),]
bacteria_beta_div_up2_within_Time<-rbind(bacteria_beta_div_up2_within_Time1,bacteria_beta_div_up2_within_Time2)
head(bacteria_beta_div_up2_within_Time)

bacteria_beta_div_up2_within_Time<-bacteria_beta_div_up2_within_Time[bacteria_beta_div_up2_within_Time$B_Bray_curtis_distance!=0,]
bacteria_beta_div_up2_within_Time$Time<-factor(bacteria_beta_div_up2_within_Time$Time,levels = c('Pre_sowing','Within_Week1','Within_Week3','Within_Week4','Within_Endosphere'))

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bacteria Bray-curtis diversity between samples - between compartment.tiff', units="in", width=9, height=10, res=300)

ggplot(bacteria_beta_div_up2_within_Time,aes(x=Time,y=B_Bray_curtis_distance))+
geom_jitter(shape=16, position=position_jitter(width = 0.3,height = 0),aes(color=Compartment))+scale_color_manual(values = c(as.character(Treat_colors),'#dd4632'),name="Compartment")+labs(title="Between sample dissimilarity of bacteria",x="Time",y="Bray-Curtis dissimilarity")+scale_x_discrete(labels=c('Within_Week1'='Week1','Within_Week3'='Week3','Within_Week4'='Week4','Pre_sowing'='Pre_sow'))+ylim(0,1)+scale_y_continuous(limits=c(0,1))+theme(legend.position="right",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=20,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=15,family="Arial"),axis.text.x = element_text(family="Arial",size=15),
    axis.title.x=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 10, r = 0, b = 0, l = 0)),axis.title.y=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 0, r = 10, b = 0, l = 0)),legend.key.size = unit(1.2,'line'),legend.title = element_text(size=18,face="bold",family = "Arial"),legend.text = (element_text(size=15,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


##  ----Fungi ------

## Fungi Between weeks and within weeks

dim(beta_div_join)
beta_div_join[1:10,]
Fungi_beta_div_up<-beta_div_join[,c(8,9,7,10,11)]
head(Fungi_beta_div_up)
Fungi_beta_div_up_within_Compartment<-Fungi_beta_div_up[grep('Within',Fungi_beta_div_up$Compartment),]
head(Fungi_beta_div_up_within_Compartment)
Fungi_beta_div_up_within_Compartment<-Fungi_beta_div_up_within_Compartment[Fungi_beta_div_up_within_Compartment$F_Bray_curtis_distance!=0,]
Fungi_beta_div_up_within_Compartment$Compartment<-factor(Fungi_beta_div_up_within_Compartment$Compartment,levels = c('Within_Soil','Within_Bulk','Within_Rhizosphere','Within_Endosphere'))

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Fungi Bray-curtis diversity between samples - between weeks and withinweeks.tiff', units="in", width=7, height=10, res=300)

ggplot(Fungi_beta_div_up_within_Compartment,aes(x=Compartment,y=F_Bray_curtis_distance))+
geom_jitter(shape=16, position=position_jitter(width = 0.3,height = 0),aes(color=Time))+scale_color_manual(values = c(as.character(Treat_colors),'#dd4632'),name="Time")+labs(title="Between sample dissimilarity of Fungi",x="Compartment",y="Bray-Curtis dissimilarity")+scale_x_discrete(labels=c('Within_Bulk'='Bulk','Within_Endosphere'='Endo','Within_Rhizosphere'='Rhi','Within_Soil'='Soil'))+ylim(0,1)+scale_y_continuous(limits=c(0,1))+theme(legend.position="right",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=20,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=15,family="Arial"),axis.text.x = element_text(family="Arial",size=15),
    axis.title.x=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 10, r = 0, b = 0, l = 0)),axis.title.y=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 0, r = 10, b = 0, l = 0)),legend.key.size = unit(1.2,'line'),legend.title = element_text(size=18,face="bold",family = "Arial"),legend.text = (element_text(size=15,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Fungi Bray-curtis diversity between samples - between weeks and withinweeks-2.tiff', units="in", width=10, height=10, res=300)

ggplot(Fungi_beta_div_up_within_Compartment,aes(x=Compartment,y=F_Bray_curtis_distance))+
geom_jitter(shape=16, position=position_jitter(width = 0.3,height = 0),aes(color=Time))+scale_color_manual(values = c(as.character(Treat_colors),'#dd4632'),name="Time")+labs(title="Between sample dissimilarity of Fungi",x="Compartment",y="Bray-Curtis dissimilarity")+scale_x_discrete(labels=c('Within_Bulk'='Bulk','Within_Endosphere'='Endo','Within_Rhizosphere'='Rhi','Within_Soil'='Soil'))+ylim(0,1)+scale_y_continuous(limits=c(0,1))+theme(legend.position="right",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=20,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=15,family="Arial"),axis.text.x = element_text(family="Arial",size=15),
    axis.title.x=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 10, r = 0, b = 0, l = 0)),axis.title.y=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 0, r = 10, b = 0, l = 0)),legend.key.size = unit(1.2,'line'),legend.title = element_text(size=18,face="bold",family = "Arial"),legend.text = (element_text(size=15,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Fungi only within weeks

Fungi_beta_div_up_within_Compartment_withinwk<-Fungi_beta_div_up_within_Compartment[-c(grep('vs',Fungi_beta_div_up_within_Compartment$Time)),]

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Fungi Bray-curtis diversity between samples - Within weeks.tiff', units="in", width=7, height=10, res=300)

ggplot(Fungi_beta_div_up_within_Compartment_withinwk,aes(x=Compartment,y=F_Bray_curtis_distance))+
geom_jitter(shape=16, position=position_jitter(width = 0.3,height = 0),aes(color=Time))+scale_color_manual(values = c(as.character(Treat_colors),'#dd4632'),name="Time Group")+labs(title="Between sample dissimilarity of Fungi",x="Compartment",y="Bray-Curtis dissimilarity")+scale_x_discrete(labels=c('Within_Bulk'='Bulk','Within_Endosphere'='Endo','Within_Rhizosphere'='Rhi','Within_Soil'='Soil'))+ylim(0,1)+scale_y_continuous(limits=c(0,1))+theme(legend.position="right",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=20,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=15,family="Arial"),axis.text.x = element_text(family="Arial",size=15),
    axis.title.x=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 10, r = 0, b = 0, l = 0)),axis.title.y=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 0, r = 10, b = 0, l = 0)),legend.key.size = unit(1.2,'line'),legend.title = element_text(size=18,face="bold",family = "Arial"),legend.text = (element_text(size=15,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


##Fungi only between weeks

Fungi_beta_div_up_within_Compartment_betweenwk<-Fungi_beta_div_up_within_Compartment[grep('vs',Fungi_beta_div_up_within_Compartment$Time),]

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Fungi Bray-curtis diversity between samples - Between weeks.tiff', units="in", width=7, height=10, res=300)

ggplot(Fungi_beta_div_up_within_Compartment_betweenwk,aes(x=Compartment,y=F_Bray_curtis_distance))+
geom_jitter(shape=16, position=position_jitter(width = 0.3,height = 0),aes(color=Time))+scale_color_manual(values = c(as.character(Treat_colors),'#dd4632'),name="Time Group")+labs(title="Between sample dissimilarity of Fungi",x="Compartment",y="Bray-Curtis dissimilarity")+scale_x_discrete(labels=c('Within_Bulk'='Bulk','Within_Endosphere'='Endo','Within_Rhizosphere'='Rhi','Within_Soil'='Soil'))+ylim(0,1)+scale_y_continuous(limits=c(0,1))+theme(legend.position="right",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=20,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=15,family="Arial"),axis.text.x = element_text(family="Arial",size=15),
    axis.title.x=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 10, r = 0, b = 0, l = 0)),axis.title.y=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 0, r = 10, b = 0, l = 0)),legend.key.size = unit(1.2,'line'),legend.title = element_text(size=18,face="bold",family = "Arial"),legend.text = (element_text(size=15,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()



## Fungi between compartment

Fungi_beta_div_up2<-beta_div_join[,c(8,9,7,10,11)]
head(Fungi_beta_div_up2)
Fungi_beta_div_up2_within_Time1<-Fungi_beta_div_up2[grep('Within',Fungi_beta_div_up2$Time),]
Fungi_beta_div_up2_within_Time2<-Fungi_beta_div_up2[-c(grep('Week',Fungi_beta_div_up2$Time)),]
Fungi_beta_div_up2_within_Time<-rbind(Fungi_beta_div_up2_within_Time1,Fungi_beta_div_up2_within_Time2)
head(Fungi_beta_div_up2_within_Time)

Fungi_beta_div_up2_within_Time<-Fungi_beta_div_up2_within_Time[Fungi_beta_div_up2_within_Time$F_Bray_curtis_distance!=0,]
Fungi_beta_div_up2_within_Time$Time<-factor(Fungi_beta_div_up2_within_Time$Time,levels = c('Pre_sowing','Within_Week1','Within_Week3','Within_Week4','Within_Endosphere'))

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Fungi Bray-curtis diversity between samples - between compartment.tiff', units="in", width=9, height=10, res=300)

ggplot(Fungi_beta_div_up2_within_Time,aes(x=Time,y=F_Bray_curtis_distance))+
geom_jitter(shape=16, position=position_jitter(width = 0.3,height = 0),aes(color=Compartment))+scale_color_manual(values = c(as.character(Treat_colors),'#dd4632'),name="Compartment")+labs(title="Between sample dissimilarity of Fungi",x="Time",y="Bray-Curtis dissimilarity")+scale_x_discrete(labels=c('Within_Week1'='Week1','Within_Week3'='Week3','Within_Week4'='Week4','Pre_sowing'='Pre_sow'))+ylim(0,1)+scale_y_continuous(limits=c(0,1))+theme(legend.position="right",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=20,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=15,family="Arial"),axis.text.x = element_text(family="Arial",size=15),
    axis.title.x=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 10, r = 0, b = 0, l = 0)),axis.title.y=element_text(size = 18,face="bold",family="Arial",vjust = 1,margin = margin(t = 0, r = 10, b = 0, l = 0)),legend.key.size = unit(1.2,'line'),legend.title = element_text(size=18,face="bold",family = "Arial"),legend.text = (element_text(size=15,family = "Arial")),legend.background = element_rect(fill="transparent"),legend.box.background = element_rect(color="transparent", fill = "transparent"),legend.key = element_rect(fill = "transparent", colour = NA),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

```

#Venn plot 
# First, I need to join r_Seed_13021_up with r_SEED_up phyloseq objectives

```{r}
r_SEED_up
r_Seed_13021_up

r_SEED_up_otu_table<-data.frame(otu_table(r_SEED_up)@.Data)
r_SEED_up_otu_table$OTUID<-taxa_names(r_SEED_up)

r_Seed_13021_up_otu_table<-data.frame(otu_table(r_Seed_13021_up)@.Data)
r_Seed_13021_up_otu_table$OTUID<-taxa_names(r_Seed_13021_up)

join_otu_table<-join(r_SEED_up_otu_table,r_Seed_13021_up_otu_table,by = "OTUID",type="full")
colnames(join_otu_table)<-sub("X","",colnames(join_otu_table),ignore.case=FALSE,perl=TRUE,fixed = FALSE, useBytes = FALSE)
#length(intersect(r_SEED_up_otu_table$OTUID,r_Seed_13021_up_otu_table$OTUID))

join_otu_table_up<-join_otu_table[,c(1:15,17:165)]
rownames(join_otu_table_up)<-join_otu_table$OTUID
join_otu_table_up[is.na(join_otu_table_up)]<-0
join_OTU_object<-otu_table(join_otu_table_up,taxa_are_rows = TRUE)
join_tax_object<-tax_table(Seed_phyloseq_up)
join_Seed_phyloseq<-merge_phyloseq(join_OTU_object,join_tax_object)
join_Seed_phyloseq_up<-merge_phyloseq(join_Seed_phyloseq,Seed_meta_up_phyloseq)
join_Seed_phyloseq_up


## join_genus

join_genus<-tax_glom(join_Seed_phyloseq_up,taxrank = rank_names(join_Seed_phyloseq_up)[6],NArm = TRUE)
join_genus
join_genus_df<-data.frame(otu_table(join_genus)@.Data)
write.csv(join_otu_df,file = "join_genus_df.csv")

# remove sample Seed_CM_4 due to its confusing taxonomy composition compared with other samples and biased sequencing depth.
join_genus_up1<-subset_samples(join_genus,Sample_ID!="Seed_CM_4")
join_genus_up1 #1128 taxa and 163 samples
join_genus_up<-filter_taxa(join_genus_up1,function(x) length(which(x>0))>=5,prune = TRUE)
join_genus_up # 956 taxa across 163 samples
# join_genus_Compartment

join_genus_Compartment<-merge_samples(join_genus_up,group = "Compartment",fun = mean)
join_genus_Compartment
Compartment_otu_genus_table<-data.frame(t(otu_table(join_genus_Compartment)))

# Compare between Rhi and Endo, Seed

Rhi<-filter(Compartment_otu_genus_table,Rhizosphere>0)
dim(Rhi)#901
Endo<-filter(Compartment_otu_genus_table,Endosphere>0)
dim(Endo)#840
Seed<-filter(Compartment_otu_genus_table,SEED>0)
dim(Seed) #175

Rhi_Endo<-filter(Compartment_otu_genus_table,Rhizosphere>0&Endosphere>0)
dim(Rhi_Endo) #806

Rhi_Seed<-filter(Compartment_otu_genus_table,Rhizosphere>0&SEED>0)
dim(Rhi_Seed) #161

Endo_Seed<-filter(Compartment_otu_genus_table,Endosphere>0&SEED>0)
dim(Endo_Seed) #166

Rhi_Endo_Seed<-filter(Compartment_otu_genus_table,Endosphere>0&SEED>0&Rhizosphere>0)
dim(Rhi_Endo_Seed) #159

# Subset to wk1, wk3 and wk4

## ---- week1 ----

wk1_join_genus_up<-subset_samples(join_genus_up,Time=="1wk"|Compartment=="SEED"|Compartment=="Soil")
wk1_join_genus_up_compartment<-merge_samples(wk1_join_genus_up,group = "Compartment",fun = mean)
wk1_join_genus_up_compartment_table<-data.frame(t(otu_table(wk1_join_genus_up_compartment)))

## Week1 Venn plot --- Rhi vs Endo vs Seed

wk1_Rhi<-filter(wk1_join_genus_up_compartment_table,Rhizosphere>0)
dim(wk1_Rhi) #761 
wk1_Endo<-filter(wk1_join_genus_up_compartment_table,Endosphere>0)
dim(wk1_Endo) #782
wk1_Seed<-filter(wk1_join_genus_up_compartment_table,SEED>0)
dim(wk1_Seed) #175

wk1_Rhi_Endo<-filter(wk1_join_genus_up_compartment_table,Rhizosphere>0&Endosphere>0)
dim(wk1_Rhi_Endo) #672

wk1_Rhi_Seed<-filter(wk1_join_genus_up_compartment_table,Rhizosphere>0&SEED>0)
dim(wk1_Rhi_Seed) #151

wk1_Endo_Seed<-filter(wk1_join_genus_up_compartment_table,Endosphere>0&SEED>0)
dim(wk1_Endo_Seed) #164

wk1_Rhi_Endo_Seed<-filter(wk1_join_genus_up_compartment_table,Rhizosphere>0&Endosphere>0&SEED>0)
dim(wk1_Rhi_Endo_Seed) #150


## Week1 - between Soil vs Bulk vs Rhi

wk1_Soil<-filter(wk1_join_genus_up_compartment_table,Soil>0)
dim(wk1_Soil) #842
wk1_Bulk<-filter(wk1_join_genus_up_compartment_table,Bulk>0)
dim(wk1_Bulk) #821
wk1_Rhi<-filter(wk1_join_genus_up_compartment_table,Rhizosphere>0)
dim(wk1_Rhi) #761
wk1_Soil_Bulk<-filter(wk1_join_genus_up_compartment_table,Soil>0&Bulk>0)
dim(wk1_Soil_Bulk) #780
wk1_Soil_Rhi<-filter(wk1_join_genus_up_compartment_table,Soil>0&Rhizosphere>0)
dim(wk1_Soil_Rhi) #721
wk1_Bulk_Rhi<-filter(wk1_join_genus_up_compartment_table,Bulk>0&Rhizosphere>0)
dim(wk1_Bulk_Rhi) #714
wk1_Soil_Bulk_Rhi<-filter(wk1_join_genus_up_compartment_table,Soil>0&Bulk>0&Rhizosphere>0)
dim(wk1_Soil_Bulk_Rhi) #693


## ---- week3 ----

wk3_join_genus_up<-subset_samples(join_genus_up,Time=="3wk"|Compartment=="SEED"|Compartment=="Soil")
wk3_join_genus_up_compartment<-merge_samples(wk3_join_genus_up,group = "Compartment",fun = mean)
wk3_join_genus_up_compartment_table<-data.frame(t(otu_table(wk3_join_genus_up_compartment)))

## Week3 Venn plot- Rhi vs Endo vs Seed

wk3_Rhi<-filter(wk3_join_genus_up_compartment_table,Rhizosphere>0)
dim(wk3_Rhi) #829
wk3_Endo<-filter(wk3_join_genus_up_compartment_table,Endosphere>0)
dim(wk3_Endo) #676
wk3_Seed<-filter(wk3_join_genus_up_compartment_table,SEED>0)
dim(wk3_Seed) #175
wk3_Rhi_Endo<-filter(wk3_join_genus_up_compartment_table,Rhizosphere>0&Endosphere>0)
dim(wk3_Rhi_Endo) #633
wk3_Rhi_Seed<-filter(wk3_join_genus_up_compartment_table,Rhizosphere>0&SEED>0)
dim(wk3_Rhi_Seed) #148
wk3_Endo_Seed<-filter(wk3_join_genus_up_compartment_table,Endosphere>0&SEED>0)
dim(wk3_Endo_Seed) #152
wk3_Rhi_Endo_Seed<-filter(wk3_join_genus_up_compartment_table,Rhizosphere>0&Endosphere>0&SEED>0)
dim(wk3_Rhi_Endo_Seed) #142


## Week3 - between Soil vs Bulk vs Rhi

wk3_Soil<-filter(wk3_join_genus_up_compartment_table,Soil>0)
dim(wk3_Soil) #842
wk3_Bulk<-filter(wk3_join_genus_up_compartment_table,Bulk>0)
dim(wk3_Bulk) #863
wk3_Rhi<-filter(wk3_join_genus_up_compartment_table,Rhizosphere>0)
dim(wk3_Rhi) #829
wk3_Soil_Bulk<-filter(wk3_join_genus_up_compartment_table,Soil>0&Bulk>0)
dim(wk3_Soil_Bulk) #794
wk3_Soil_Rhi<-filter(wk3_join_genus_up_compartment_table,Soil>0&Rhizosphere>0)
dim(wk3_Soil_Rhi) #762
wk3_Bulk_Rhi<-filter(wk3_join_genus_up_compartment_table,Bulk>0&Rhizosphere>0)
dim(wk3_Bulk_Rhi) #772
wk3_Soil_Bulk_Rhi<-filter(wk3_join_genus_up_compartment_table,Soil>0&Bulk>0&Rhizosphere>0)
dim(wk3_Soil_Bulk_Rhi) #731

## ---- Week4 ----

wk4_join_genus_up<-subset_samples(join_genus_up,Time=="4wk"|Compartment=="SEED"|Compartment=="Soil")
wk4_join_genus_up_compartment<-merge_samples(wk4_join_genus_up,group = "Compartment",fun = mean)
wk4_join_genus_up_compartment_table<-data.frame(t(otu_table(wk4_join_genus_up_compartment)))

## Week4 Venn plot - Rhi vs Endo vs Seed

wk4_Rhi<-filter(wk4_join_genus_up_compartment_table,Rhizosphere>0)
dim(wk4_Rhi) #809
wk4_Endo<-filter(wk4_join_genus_up_compartment_table,Endosphere>0)
dim(wk4_Endo) #685
wk4_Seed<-filter(wk4_join_genus_up_compartment_table,SEED>0)
dim(wk4_Seed) #175
wk4_Rhi_Endo<-filter(wk4_join_genus_up_compartment_table,Rhizosphere>0&Endosphere>0)
dim(wk4_Rhi_Endo) #639
wk4_Rhi_Seed<-filter(wk4_join_genus_up_compartment_table,Rhizosphere>0&SEED>0)
dim(wk4_Rhi_Seed) #149
wk4_Endo_Seed<-filter(wk4_join_genus_up_compartment_table,Endosphere>0&SEED>0)
dim(wk4_Endo_Seed) #152
wk4_Rhi_Endo_Seed<-filter(wk4_join_genus_up_compartment_table,Rhizosphere>0&Endosphere>0&SEED>0)
dim(wk4_Rhi_Endo_Seed) #144

## Week4 - between Soil vs Bulk vs Rhi

wk4_Soil<-filter(wk4_join_genus_up_compartment_table,Soil>0)
dim(wk4_Soil) #842
wk4_Bulk<-filter(wk4_join_genus_up_compartment_table,Bulk>0)
dim(wk4_Bulk) #852
wk4_Rhi<-filter(wk4_join_genus_up_compartment_table,Rhizosphere>0)
dim(wk4_Rhi) #809
wk4_Soil_Bulk<-filter(wk4_join_genus_up_compartment_table,Soil>0&Bulk>0)
dim(wk4_Soil_Bulk) #788
wk4_Soil_Rhi<-filter(wk4_join_genus_up_compartment_table,Soil>0&Rhizosphere>0)
dim(wk4_Soil_Rhi) #748
wk4_Bulk_Rhi<-filter(wk4_join_genus_up_compartment_table,Bulk>0&Rhizosphere>0)
dim(wk4_Bulk_Rhi) #763
wk4_Soil_Bulk_Rhi<-filter(wk4_join_genus_up_compartment_table,Soil>0&Bulk>0&Rhizosphere>0)
dim(wk4_Soil_Bulk_Rhi) #721

## Bulk - wk1 vs wk3 vs wk4
Bulk_join_genus_up<-subset_samples(join_genus_up,Compartment=="Bulk")
time_Bulk_join_genus_up<-merge_samples(Bulk_join_genus_up,group = "Time",fun = mean)
time_Bulk_join_genus_up_table<-data.frame(t(otu_table(time_Bulk_join_genus_up)))

Bulk_wk1<-filter(time_Bulk_join_genus_up_table,X1wk>0)
dim(Bulk_wk1) #821
Bulk_wk3<-filter(time_Bulk_join_genus_up_table,X3wk>0)
dim(Bulk_wk3) #863
Bulk_wk4<-filter(time_Bulk_join_genus_up_table,X4wk>0)
dim(Bulk_wk4) #852
Bulk_wk1_wk3<-filter(time_Bulk_join_genus_up_table,X1wk>0&X3wk>0)
dim(Bulk_wk1_wk3) #778
Bulk_wk1_wk4<-filter(time_Bulk_join_genus_up_table,X1wk>0&X4wk>0)
dim(Bulk_wk1_wk4) #773
Bulk_wk3_wk4<-filter(time_Bulk_join_genus_up_table,X3wk>0&X4wk>0)
dim(Bulk_wk3_wk4) #810
Bulk_wk1_wk3_wk4<-filter(time_Bulk_join_genus_up_table,X1wk>0&X3wk>0&X4wk>0)
dim(Bulk_wk1_wk3_wk4)  # 747

## Rhizosphere - wk1 vs wk3 vs wk4

Rhi_join_genus_up<-subset_samples(join_genus_up,Compartment=="Rhizosphere")
time_Rhi_join_genus_up<-merge_samples(Rhi_join_genus_up,group = "Time",fun = mean)
time_Rhi_join_genus_up_table<-data.frame(t(otu_table(time_Rhi_join_genus_up)))

Rhi_wk1<-filter(time_Rhi_join_genus_up_table,X1wk>0)
dim(Rhi_wk1) #761

Rhi_wk3<-filter(time_Rhi_join_genus_up_table,X3wk>0)
dim(Rhi_wk3) #829
Rhi_wk4<-filter(time_Rhi_join_genus_up_table,X4wk>0)
dim(Rhi_wk4) #809
Rhi_wk1_wk3<-filter(time_Rhi_join_genus_up_table,X1wk>0&X3wk>0)
dim(Rhi_wk1_wk3) #717
Rhi_wk1_wk4<-filter(time_Rhi_join_genus_up_table,X1wk>0&X4wk>0)
dim(Rhi_wk1_wk4) #703
Rhi_wk3_wk4<-filter(time_Rhi_join_genus_up_table,X3wk>0&X4wk>0)
dim(Rhi_wk3_wk4) #765
Rhi_wk1_wk3_wk4<-filter(time_Rhi_join_genus_up_table,X1wk>0&X3wk>0&X4wk>0)
dim(Rhi_wk1_wk3_wk4)  #687

## Endosphere - wk1 vs wk3 vs wk4

Endo_join_genus_up<-subset_samples(join_genus_up,Compartment=="Endosphere")
time_Endo_join_genus_up<-merge_samples(Endo_join_genus_up,group = "Time",fun = mean)
time_Endo_join_genus_up_table<-data.frame(t(otu_table(time_Endo_join_genus_up)))

Endo_wk1<-filter(time_Endo_join_genus_up_table,X1wk>0)
dim(Endo_wk1) #782
Endo_wk3<-filter(time_Endo_join_genus_up_table,X3wk>0)
dim(Endo_wk3) #676
Endo_wk4<-filter(time_Endo_join_genus_up_table,X4wk>0)
dim(Endo_wk4) #685
Endo_wk1_wk3<-filter(time_Endo_join_genus_up_table,X1wk>0&X3wk>0)
dim(Endo_wk1_wk3) #644
Endo_wk1_wk4<-filter(time_Endo_join_genus_up_table,X1wk>0&X4wk>0)
dim(Endo_wk1_wk4) #636
Endo_wk3_wk4<-filter(time_Endo_join_genus_up_table,X3wk>0&X4wk>0)
dim(Endo_wk3_wk4) #609
Endo_wk1_wk3_wk4<-filter(time_Endo_join_genus_up_table,X1wk>0&X3wk>0&X4wk>0)
dim(Endo_wk1_wk3_wk4)  #586
```

## Plot Venn -- absolute value

```{r}
library(VennDiagram)

## --- Week1 between compartments --- Rhi vs Endo vs Seed -- 881

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between Rhi vs Endo vs Seed- Week1.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 761,
area2 = 782,
area3 = 175,
n12 = 672,
n13 = 151,
n23 = 164,
n123 = 150,
category = c("Rhi", "Endo", "SEED"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()

## --- Week3 between compartments ---Rhi vs Endo vs Seed --- 889

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between Rhi vs Endo vs Seed- Week3.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 829,
area2 = 676,
area3 = 175,
n12 = 633,
n13 = 148,
n23 = 152,
n123 = 142,
category = c("Rhi", "Endo", "SEED"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()


## --- Week4 between compartments ---Rhi vs Endo vs Seed --873

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between Rhi vs Endo vs Seed- Week4.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 809,
area2 = 685,
area3 = 175,
n12 = 639,
n13 = 149,
n23 = 152,
n123 = 144,
category = c("Rhi", "Endo", "SEED"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()


## --- Week1 between compartments --- Soil vs Bulk vs Rhizosphere -- 902

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between Soil vs Bulk vs Rhizosphere - Week1.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 842,
area2 = 821,
area3 = 761,
n12 = 780,
n13 = 721,
n23 = 714,
n123 = 693,
category = c("Soil", "Bulk", "Rhizosphere"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()


## --- Week3 between compartments --- Soil vs Bulk vs Rhizosphere -937

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between Soil vs Bulk vs Rhizosphere - Week3.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 842,
area2 = 863,
area3 = 829,
n12 = 794,
n13 = 762,
n23 = 772,
n123 = 731,
category = c("Soil", "Bulk", "Rhizosphere"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()


## --- Week4 between compartments --- Soil vs Bulk vs Rhizosphere --925

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between Soil vs Bulk vs Rhizosphere - Week4.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 842,
area2 = 852,
area3 = 809,
n12 = 788,
n13 = 748,
n23 = 763,
n123 = 721,
category = c("Soil", "Bulk", "Rhizosphere"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()


##  ---- Between time points - Bulk ----922

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between development stages- Bulk.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 821,
area2 = 863,
area3 = 852,
n12 = 778,
n13 = 773,
n23 = 810,
n123 = 747,
category = c("Week1", "Week3", "Week4"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()

##  ---- Between time points - Rhizosphere ----901

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between development stages- Rhizosphere.tiff', units="in", width=7, height=7, res=300)

venn.plot <- draw.triple.venn(
area1 = 761,
area2 = 829,
area3 = 809,
n12 = 717,
n13 = 703,
n23 = 765,
n123 = 687,
category = c("Week1", "Week3", "Week4"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()

##  ---- Between time points - Endosphere ----836

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between development stages- Endosphere.tiff', units="in", width=7, height=7, res=300)

venn.plot <- draw.triple.venn(
area1 = 782,
area2 = 672,
area3 = 685,
n12 = 644,
n13 = 636,
n23 = 609,
n123 = 586,
category = c("Week1", "Week3", "Week4"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()

```

## Plot Venn -- percentage

```{r}
library(VennDiagram)

## --- Week1 between compartments --- Rhi vs Endo vs Seed -- 881

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between Rhi vs Endo vs Seed- Week1 percentage.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 761,
area2 = 782,
area3 = 175,
n12 = 672,
n13 = 151,
n23 = 164,
n123 = 150,
category = c("Rhi", "Endo", "SEED"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
print.mode = "percent",
sigdigs = 2,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()

## --- Week3 between compartments ---Rhi vs Endo vs Seed --- 889

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between Rhi vs Endo vs Seed- Week3 percentage.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 829,
area2 = 676,
area3 = 175,
n12 = 633,
n13 = 148,
n23 = 152,
n123 = 142,
category = c("Rhi", "Endo", "SEED"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
print.mode = 'percent',
sigdigs = 2,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()


## --- Week4 between compartments ---Rhi vs Endo vs Seed --873

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between Rhi vs Endo vs Seed- Week4 percentage.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 809,
area2 = 685,
area3 = 175,
n12 = 639,
n13 = 149,
n23 = 152,
n123 = 144,
category = c("Rhi", "Endo", "SEED"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
print.mode = 'percent',
sigdigs = 2,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()


## --- Week1 between compartments --- Soil vs Bulk vs Rhizosphere -- 902

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between Soil vs Bulk vs Rhizosphere - Week1 percentage.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 842,
area2 = 821,
area3 = 761,
n12 = 780,
n13 = 721,
n23 = 714,
n123 = 693,
category = c("Soil", "Bulk", "Rhizosphere"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
print.mode = 'percent',
sigdigs = 2,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()


## --- Week3 between compartments --- Soil vs Bulk vs Rhizosphere -937

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between Soil vs Bulk vs Rhizosphere - Week3 percentage.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 842,
area2 = 863,
area3 = 829,
n12 = 794,
n13 = 762,
n23 = 772,
n123 = 731,
category = c("Soil", "Bulk", "Rhizosphere"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
print.mode = 'percent',
sigdigs = 2,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()


## --- Week4 between compartments --- Soil vs Bulk vs Rhizosphere --925

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between Soil vs Bulk vs Rhizosphere - Week4 percentage.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 842,
area2 = 852,
area3 = 809,
n12 = 788,
n13 = 748,
n23 = 763,
n123 = 721,
category = c("Soil", "Bulk", "Rhizosphere"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
print.mode = 'percent',
sigdigs = 2,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()


##  ---- Between time points - Bulk ----922

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between development stages- Bulk percentage.tiff', units="in", width=7, height=7, res=300)
venn.plot <- draw.triple.venn(
area1 = 821,
area2 = 863,
area3 = 852,
n12 = 778,
n13 = 773,
n23 = 810,
n123 = 747,
category = c("Week1", "Week3", "Week4"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
print.mode = 'percent',
sigdigs = 2,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()

##  ---- Between time points - Rhizosphere ----901

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between development stages- Rhizosphere percentage.tiff', units="in", width=7, height=7, res=300)

venn.plot <- draw.triple.venn(
area1 = 761,
area2 = 829,
area3 = 809,
n12 = 717,
n13 = 703,
n23 = 765,
n123 = 687,
category = c("Week1", "Week3", "Week4"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
print.mode = 'percent',
sigdigs = 2,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()

##  ---- Between time points - Endosphere ----836

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Venn_plot/Venn plot of bacteria genus between development stages- Endosphere percentage.tiff', units="in", width=7, height=7, res=300)

venn.plot <- draw.triple.venn(
area1 = 782,
area2 = 672,
area3 = 685,
n12 = 644,
n13 = 636,
n23 = 609,
n123 = 586,
category = c("Week1", "Week3", "Week4"),
fill = c('#838B8B','#48af9c',"#9c5546"),
lty = "dashed",
cex = 2,
fontface = rep("bold",7),
cat.fontface = rep("bold",3),
cat.default.pos="outer",
cat.cex = 2.2,
cat.dist = c(0.07,0.07,0.05),
margin=0.1,
print.mode = 'percent',
sigdigs = 2,
cat.col = c('#838B8B','#48af9c',"#9c5546"),
fontfamily = 'Arial',
cat.fontfamily = "Arial"
);
dev.off()
```

## I need to figure out what genera are unique to Endo-wk1, which not only contribute to the high diversity for Endo_1wk but also its species richness

```{r}
str(wk1_join_genus_up_compartment_table)
Endo_wk1_unique<-wk1_join_genus_up_compartment_table[which(wk1_join_genus_up_compartment_table$Rhizosphere==0&wk1_join_genus_up_compartment_table$Soil==0&wk1_join_genus_up_compartment_table$Endosphere!=0),]
Endo_wk1_unique$OTUID<-rownames(Endo_wk1_unique)
Endo_wk1_unique_sort_count_table<-
  Endo_wk1_unique %>%
  arrange(desc(Endosphere))

Endo_wk1_unique_sort_tax<-data.frame(tax_table(wk1_join_genus_up_compartment)@.Data)[match(Endo_wk1_unique_sort$OTUID,rownames(tax_table(wk1_join_genus_up_compartment))),]
head(Endo_wk1_unique_sort_tax,10)
Endo_wk1_unique_sort_tax$OTUID<-rownames(Endo_wk1_unique_sort_tax)

Endo_wk1_unique_genus_join<-join(Endo_wk1_unique_sort_count_table,Endo_wk1_unique_sort_tax,by="OTUID")

## dominate seed bacteria genus list

str(Compartment_otu_genus_table)
head(Compartment_otu_genus_table)
Compartment_otu_genus_table$OTUID<-rownames(Compartment_otu_genus_table)
Compartment_otu_genus_table_seed_sorted<-Compartment_otu_genus_table%>%
  arrange(desc(Seed))

Compartment_genus_taxonomy_table_seed_sorted<-data.frame(tax_table(join_genus_Compartment)@.Data)[match(Compartment_otu_genus_table_seed_sorted$OTUID,rownames(tax_table(join_genus_Compartment))),]
Compartment_genus_taxonomy_table_seed_sorted[1:20,]
Compartment_genus_taxonomy_table_seed_sorted$OTUID<-rownames(Compartment_genus_taxonomy_table_seed_sorted)

Compartment_genus_join_table_seed_sorted<-join(Compartment_otu_genus_table_seed_sorted,Compartment_genus_taxonomy_table_seed_sorted,by="OTUID")
dim(Compartment_genus_join_table_seed_sorted)
head(Compartment_genus_join_table_seed_sorted)
```

## barplot for specific taxa based on differential abundance analysis - Lefse
# here the input file are rarefied and singleton removed phylotype based shared and cons.taxonomy file

```{r}
seed_genus<-import_mothur(mothur_shared_file = '/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Seed.trim.contigs.good.unique.good.filter.unique.precluster.pick.nr_v132.wang.pick.subsample.tx.1.pick.shared',mothur_constaxonomy_file = '/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/Seed.trim.contigs.good.unique.good.filter.unique.precluster.pick.nr_v132.wang.pick.subsample.tx.1.pick.1.cons.taxonomy')
seed_genus<-prune_samples(seed_genus,samples = sample_names(seed_genus)[1:149])
colnames(tax_table(seed_genus))<-c('Kingdom','Phylum','Class','Order','Family','Genus')
length(sample_names(seed_genus))
rownames(seed_meta_up)[1:5]
seed_genus_up<-merge_phyloseq(seed_genus,Seed_meta_up_phyloseq)
r_seed_genus_up<-transform_sample_counts(seed_genus_up,function(x) x/sum(x))
r_seed_genus_up

## merge_based on treat factor
r_seed_genus_up_treat<-merge_samples(r_seed_genus_up,group = "Treat",fun = mean)
r_seed_genus_up_treat<-transform_sample_counts(r_seed_genus_up_treat,function(x) x/sum(x))

Bulk_1wk_top20<-data.frame(Bulk_1wk_top20=labels(sort(otu_table(r_seed_genus_up_treat)@.Data[1,],decreasing = TRUE)[1:20]))
Bulk_1wk_top20$Genus_ID<-Bulk_1wk_top20$Bulk_1wk_top20
sum(sort(otu_table(r_seed_genus_up_treat)@.Data[1,],decreasing = TRUE)[1:20]) # 47%

Bulk_3wk_top20<-data.frame(Bulk_3wk_top20=labels(sort(otu_table(r_seed_genus_up_treat)@.Data[2,],decreasing = TRUE)[1:20]))
Bulk_3wk_top20$Genus_ID<-Bulk_3wk_top20$Bulk_3wk_top20
sum(sort(otu_table(r_seed_genus_up_treat)@.Data[2,],decreasing = TRUE)[1:20]) # 46%

Bulk_4wk_top20<-data.frame(Bulk_4wk_top20=labels(sort(otu_table(r_seed_genus_up_treat)@.Data[3,],decreasing = TRUE)[1:20]))
Bulk_4wk_top20$Genus_ID<-Bulk_4wk_top20$Bulk_4wk_top20
sum(sort(otu_table(r_seed_genus_up_treat)@.Data[3,],decreasing = TRUE)[1:20]) # 45%

Endo_1wk_top20<-data.frame(Endo_1wk_top20=labels(sort(otu_table(r_seed_genus_up_treat)@.Data[4,],decreasing = TRUE)[1:20]))
Endo_1wk_top20$Genus_ID<-Endo_1wk_top20$Endo_1wk_top20
sum(sort(otu_table(r_seed_genus_up_treat)@.Data[4,],decreasing = TRUE)[1:20]) # 60%
tax_table(r_seed_genus_up_treat)@.Data[match(Endo_1wk_top20$Genus_ID,taxa_names(r_seed_genus_up_treat)),]

Endo_3wk_top20<-data.frame(Endo_3wk_top20=labels(sort(otu_table(r_seed_genus_up_treat)@.Data[5,],decreasing = TRUE)[1:20]))
Endo_3wk_top20$Genus_ID<-Endo_3wk_top20$Endo_3wk_top20
sum(sort(otu_table(r_seed_genus_up_treat)@.Data[5,],decreasing = TRUE)[1:20]) # 77%
tax_table(r_seed_genus_up_treat)@.Data[match(Endo_3wk_top20$Genus_ID,taxa_names(r_seed_genus_up_treat)),]

Endo_4wk_top20<-data.frame(Endo_4wk_top20=labels(sort(otu_table(r_seed_genus_up_treat)@.Data[6,],decreasing = TRUE)[1:20]))
Endo_4wk_top20$Genus_ID<-Endo_4wk_top20$Endo_4wk_top20
sum(sort(otu_table(r_seed_genus_up_treat)@.Data[6,],decreasing = TRUE)[1:20]) # 74%
tax_table(r_seed_genus_up_treat)@.Data[match(Endo_4wk_top20$Genus_ID,taxa_names(r_seed_genus_up_treat)),]


Rhi_1wk_top20<-data.frame(Rhi_1wk_top20=labels(sort(otu_table(r_seed_genus_up_treat)@.Data[7,],decreasing = TRUE)[1:20]))
Rhi_1wk_top20$Genus_ID<-Rhi_1wk_top20$Rhi_1wk_top20
sum(sort(otu_table(r_seed_genus_up_treat)@.Data[7,],decreasing = TRUE)[1:20]) # 67%
tax_table(r_seed_genus_up_treat)@.Data[match(Rhi_1wk_top20$Genus_ID,taxa_names(r_seed_genus_up_treat)),]

Rhi_3wk_top20<-data.frame(Rhi_3wk_top20=labels(sort(otu_table(r_seed_genus_up_treat)@.Data[8,],decreasing = TRUE)[1:20]))
Rhi_3wk_top20$Genus_ID<-Rhi_3wk_top20$Rhi_3wk_top20
sum(sort(otu_table(r_seed_genus_up_treat)@.Data[8,],decreasing = TRUE)[1:20]) # 37%
tax_table(r_seed_genus_up_treat)@.Data[match(Rhi_3wk_top20$Genus_ID,taxa_names(r_seed_genus_up_treat)),]

Rhi_4wk_top20<-data.frame(Rhi_4wk_top20=labels(sort(otu_table(r_seed_genus_up_treat)@.Data[9,],decreasing = TRUE)[1:20]))
Rhi_4wk_top20$Genus_ID<-Rhi_4wk_top20$Rhi_4wk_top20
sum(sort(otu_table(r_seed_genus_up_treat)@.Data[9,],decreasing = TRUE)[1:20]) # 40%
tax_table(r_seed_genus_up_treat)@.Data[match(Rhi_4wk_top20$Genus_ID,taxa_names(r_seed_genus_up_treat)),]


Soil_top20<-data.frame(Soil_top20=labels(sort(otu_table(r_seed_genus_up_treat)@.Data[10,],decreasing = TRUE)[1:20]))
Soil_top20$Genus_ID<-Soil_top20$Soil_top20
sum(sort(otu_table(r_seed_genus_up_treat)@.Data[10,],decreasing = TRUE)[1:20]) # 47%

#intersect_top20<-Reduce(intersect,list(Bulk_1wk_top20,Bulk_3wk_top20,Bulk_4wk_top20,Endo_1wk_top20,Endo_3wk_top20,Endo_4wk_top20,Rhi_1wk_top20,Rhi_3wk_top20,Rhi_4wk_top20,Soil_top20))

all_top20<-data.frame(all_top20=unique(Bulk_1wk_top20$Bulk_1wk_top20,Bulk_3wk_top20$Bulk_3wk_top20,Bulk_4wk_top20$Bulk_4wk_top20,Endo_1wk_top20$Endo_1wk_top20,Endo_3wk_top20$Endo_3wk_top20,Endo_4wk_top20$Endo_4wk_top20,Rhi_1wk_top20$Rhi_1wk_top20,Rhi_3wk_top20$Rhi_3wk_top20,Rhi_4wk_top20$Rhi_4wk_top20,Soil_top20$Soil_top20))
all_top20$Genus_ID<-all_top20$all_top20

## join all_top20 with individual top20 to see how much they overlap with each other to descide to plot which genus abundance barplot

join_all_top20<- join_all(dfs=list(all_top20,Bulk_1wk_top20,Bulk_3wk_top20,Bulk_4wk_top20,Endo_1wk_top20,Endo_3wk_top20,Endo_4wk_top20,Rhi_1wk_top20,Rhi_3wk_top20,Rhi_4wk_top20,Soil_top20),type = "full",by="Genus_ID")

Genus_table<-data.frame(tax_table(r_seed_genus_up_treat)@.Data,Genus_ID=taxa_names(r_seed_genus_up_treat))
join_all_top20_up<-join(join_all_top20,Genus_table,by="Genus_ID")

#write.csv(join_all_top20_up,file = "/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/join_all_top20.csv")

## SEED_genus
SEED_genus
r_SEED_genus<-transform_sample_counts(SEED_genus,function(x) x/sum(x))

# top 10 genus
data.frame(tax_table(r_SEED_genus)@.Data)$Genus[match(labels(sort(taxa_sums(r_SEED_genus),decreasing = TRUE)[1:10]),taxa_names(r_SEED_genus))]

sort(taxa_sums(r_SEED_genus),decreasing = TRUE)[1:10]/15

top10_SEED_genus_and_relative_abundance<-data.frame(data.frame(tax_table(r_SEED_genus)@.Data)[match(labels(sort(taxa_sums(r_SEED_genus),decreasing = TRUE)[1:10]),taxa_names(r_SEED_genus)),],relative_abundance=sort(taxa_sums(r_SEED_genus),decreasing = TRUE)[1:10]/15)

#write.csv(top10_SEED_genus_and_relative_abundance,'/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/top10_SEED_genus_and_relative_abundance.csv')

## Bradyrhizobium non_SEED genus

Bradyrhizobium_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Bradyrhizobium",]
Bradyrhizobium_genus_ab_up<-data.frame(Bradyrhizobium=Bradyrhizobium_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Bradyrhizobium_genus_ab_up)<-c('Bradyrhizobium','Treat')
head(Bradyrhizobium_genus_ab_up)

## Bradyrhizobium SEED
SEED_genus
r_SEED_genus<-transform_sample_counts(SEED_genus,function(x) x/sum(x))
Bradyrhizobium_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Bradyrhizobium",]
Bradyrhizobium_genus_ab_SEED_up<-data.frame(Bradyrhizobium=Bradyrhizobium_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Bradyrhizobium_genus_ab_SEED_up)<-c('Bradyrhizobium','Treat')
head(Bradyrhizobium_genus_ab_SEED_up)

Bradyrhizobium_genus_all<-rbind(Bradyrhizobium_genus_ab_up,Bradyrhizobium_genus_ab_SEED_up)

#Bradyrhizobium_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bradyrhizobium genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Bradyrhizobium_genus_all$Treat<-factor(Bradyrhizobium_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Bradyrhizobium_genus_all,aes(x=Treat,y=Bradyrhizobium,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Bradyrhizobium Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Syntrophus non_SEED genus

Syntrophus_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Syntrophus",]
Syntrophus_genus_ab_up<-data.frame(Syntrophus=Syntrophus_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Syntrophus_genus_ab_up)<-c('Syntrophus','Treat')
head(Syntrophus_genus_ab_up)

## Syntrophus SEED
SEED_genus
r_SEED_genus<-transform_sample_counts(SEED_genus,function(x) x/sum(x))
Syntrophus_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Syntrophus",]
Syntrophus_genus_ab_SEED_up<-data.frame(Syntrophus=Syntrophus_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Syntrophus_genus_ab_SEED_up)<-c('Syntrophus','Treat')
head(Syntrophus_genus_ab_SEED_up)

Syntrophus_genus_all<-rbind(Syntrophus_genus_ab_up,Syntrophus_genus_ab_SEED_up)

#Syntrophus_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Syntrophus genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Syntrophus_genus_all$Treat<-factor(Syntrophus_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Syntrophus_genus_all,aes(x=Treat,y=Syntrophus,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Syntrophus Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Streptomyces non_SEED genus
Streptomyces_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Streptomyces",]
Streptomyces_genus_ab_up<-data.frame(Streptomyces=Streptomyces_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Streptomyces_genus_ab_up)<-c('Streptomyces','Treat')
head(Streptomyces_genus_ab_up)

## Streptomyces SEED
SEED_genus
r_SEED_genus<-transform_sample_counts(SEED_genus,function(x) x/sum(x))
Streptomyces_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Streptomyces",]
Streptomyces_genus_ab_SEED_up<-data.frame(Streptomyces=Streptomyces_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Streptomyces_genus_ab_SEED_up)<-c('Streptomyces','Treat')
head(Streptomyces_genus_ab_SEED_up)

Streptomyces_genus_all<-rbind(Streptomyces_genus_ab_up,Streptomyces_genus_ab_SEED_up)

#Streptomyces_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Streptomyces genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Streptomyces_genus_all$Treat<-factor(Streptomyces_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Streptomyces_genus_all,aes(x=Treat,y=Streptomyces,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Streptomyces Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Novosphingobium non_SEED genus

Novosphingobium_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Novosphingobium",]
Novosphingobium_genus_ab_up<-data.frame(Novosphingobium=Novosphingobium_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Novosphingobium_genus_ab_up)<-c('Novosphingobium','Treat')
head(Novosphingobium_genus_ab_up)

## Novosphingobium SEED

SEED_genus
r_SEED_genus<-transform_sample_counts(SEED_genus,function(x) x/sum(x))
Novosphingobium_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Novosphingobium",]
Novosphingobium_genus_ab_SEED_up<-data.frame(Novosphingobium=Novosphingobium_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Novosphingobium_genus_ab_SEED_up)<-c('Novosphingobium','Treat')
head(Novosphingobium_genus_ab_SEED_up)

Novosphingobium_genus_all<-rbind(Novosphingobium_genus_ab_up,Novosphingobium_genus_ab_SEED_up)

#Novosphingobium_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Novosphingobium genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Novosphingobium_genus_all$Treat<-factor(Novosphingobium_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Novosphingobium_genus_all,aes(x=Treat,y=Novosphingobium,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Novosphingobium Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Pantoea non_SEED genus

Pantoea_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Pantoea",]
Pantoea_genus_ab_up<-data.frame(Pantoea=Pantoea_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Pantoea_genus_ab_up)<-c('Pantoea','Treat')
head(Pantoea_genus_ab_up)

## Pantoea SEED

SEED_genus
r_SEED_genus<-transform_sample_counts(SEED_genus,function(x) x/sum(x))
Pantoea_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Pantoea",]
Pantoea_genus_ab_SEED_up<-data.frame(Pantoea=Pantoea_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Pantoea_genus_ab_SEED_up)<-c('Pantoea','Treat')
head(Pantoea_genus_ab_SEED_up)

Pantoea_genus_all<-rbind(Pantoea_genus_ab_up,Pantoea_genus_ab_SEED_up)

#Pantoea_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Pantoea genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Pantoea_genus_all$Treat<-factor(Pantoea_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Pantoea_genus_all,aes(x=Treat,y=Pantoea,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Pantoea Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Methylobacterium non_SEED genus

Methylobacterium_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Methylobacterium",]
Methylobacterium_genus_ab_up<-data.frame(Methylobacterium=Methylobacterium_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Methylobacterium_genus_ab_up)<-c('Methylobacterium','Treat')
head(Methylobacterium_genus_ab_up)

## Methylobacterium SEED

SEED_genus
r_SEED_genus<-transform_sample_counts(SEED_genus,function(x) x/sum(x))
Methylobacterium_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Methylobacterium",]
Methylobacterium_genus_ab_SEED_up<-data.frame(Methylobacterium=Methylobacterium_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Methylobacterium_genus_ab_SEED_up)<-c('Methylobacterium','Treat')
head(Methylobacterium_genus_ab_SEED_up)

Methylobacterium_genus_all<-rbind(Methylobacterium_genus_ab_up,Methylobacterium_genus_ab_SEED_up)

#Methylobacterium_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Methylobacterium genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Methylobacterium_genus_all$Treat<-factor(Methylobacterium_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Methylobacterium_genus_all,aes(x=Treat,y=Methylobacterium,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Methylobacterium Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Pseudomonas non_SEED genus

Pseudomonas_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Pseudomonas",]
Pseudomonas_genus_ab_up<-data.frame(Pseudomonas=Pseudomonas_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Pseudomonas_genus_ab_up)<-c('Pseudomonas','Treat')
head(Pseudomonas_genus_ab_up)

## Pseudomonas SEED

SEED_genus
r_SEED_genus<-transform_sample_counts(SEED_genus,function(x) x/sum(x))
Pseudomonas_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Pseudomonas",]
Pseudomonas_genus_ab_SEED_up<-data.frame(Pseudomonas=Pseudomonas_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Pseudomonas_genus_ab_SEED_up)<-c('Pseudomonas','Treat')
head(Pseudomonas_genus_ab_SEED_up)

Pseudomonas_genus_all<-rbind(Pseudomonas_genus_ab_up,Pseudomonas_genus_ab_SEED_up)

#Pseudomonas_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Pseudomonas genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Pseudomonas_genus_all$Treat<-factor(Pseudomonas_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Pseudomonas_genus_all,aes(x=Treat,y=Pseudomonas,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Pseudomonas Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Bacillus non_SEED genus

Bacillus_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Bacillus",]
Bacillus_genus_ab_up<-data.frame(Bacillus=Bacillus_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Bacillus_genus_ab_up)<-c('Bacillus','Treat')
head(Bacillus_genus_ab_up)

## Bacillus SEED

SEED_genus
r_SEED_genus<-transform_sample_counts(SEED_genus,function(x) x/sum(x))
Bacillus_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Bacillus",]
Bacillus_genus_ab_SEED_up<-data.frame(Bacillus=Bacillus_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Bacillus_genus_ab_SEED_up)<-c('Bacillus','Treat')
head(Bacillus_genus_ab_SEED_up)

Bacillus_genus_all<-rbind(Bacillus_genus_ab_up,Bacillus_genus_ab_SEED_up)

#Bacillus_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bacillus genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Bacillus_genus_all$Treat<-factor(Bacillus_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Bacillus_genus_all,aes(x=Treat,y=Bacillus,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Bacillus Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Acinetobacter non_SEED genus

Acinetobacter_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Acinetobacter",]
Acinetobacter_genus_ab_up<-data.frame(Acinetobacter=Acinetobacter_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Acinetobacter_genus_ab_up)<-c('Acinetobacter','Treat')
head(Acinetobacter_genus_ab_up)

## Acinetobacter SEED

SEED_genus
r_SEED_genus<-transform_sample_counts(SEED_genus,function(x) x/sum(x))
Acinetobacter_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Acinetobacter",]
Acinetobacter_genus_ab_SEED_up<-data.frame(Acinetobacter=Acinetobacter_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Acinetobacter_genus_ab_SEED_up)<-c('Acinetobacter','Treat')
head(Acinetobacter_genus_ab_SEED_up)

Acinetobacter_genus_all<-rbind(Acinetobacter_genus_ab_up,Acinetobacter_genus_ab_SEED_up)

#Acinetobacter_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Acinetobacter genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Acinetobacter_genus_all$Treat<-factor(Acinetobacter_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Acinetobacter_genus_all,aes(x=Treat,y=Acinetobacter,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Acinetobacter Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## RB41 non_SEED genus

RB41_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="RB41",]
RB41_genus_ab_up<-data.frame(RB41=RB41_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(RB41_genus_ab_up)<-c('RB41','Treat')
head(RB41_genus_ab_up)

## RB41 SEED

SEED_genus
r_SEED_genus<-transform_sample_counts(SEED_genus,function(x) x/sum(x))
RB41_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="RB41",]
RB41_genus_ab_SEED_up<-data.frame(RB41=RB41_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(RB41_genus_ab_SEED_up)<-c('RB41','Treat')
head(RB41_genus_ab_SEED_up)

RB41_genus_all<-rbind(RB41_genus_ab_up,RB41_genus_ab_SEED_up)

#RB41_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/RB41 genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  RB41_genus_all$Treat<-factor(RB41_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(RB41_genus_all,aes(x=Treat,y=RB41,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="RB41 Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Sphaerisporangium non_SEED genus

Sphaerisporangium_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Sphaerisporangium",]
Sphaerisporangium_genus_ab_up<-data.frame(Sphaerisporangium=Sphaerisporangium_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Sphaerisporangium_genus_ab_up)<-c('Sphaerisporangium','Treat')
head(Sphaerisporangium_genus_ab_up)

## Sphaerisporangium SEED

SEED_genus
r_SEED_genus<-transform_sample_counts(SEED_genus,function(x) x/sum(x))
Sphaerisporangium_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Sphaerisporangium",]
Sphaerisporangium_genus_ab_SEED_up<-data.frame(Sphaerisporangium=Sphaerisporangium_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Sphaerisporangium_genus_ab_SEED_up)<-c('Sphaerisporangium','Treat')
head(Sphaerisporangium_genus_ab_SEED_up)

Sphaerisporangium_genus_all<-rbind(Sphaerisporangium_genus_ab_up,Sphaerisporangium_genus_ab_SEED_up)

#Sphaerisporangium_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Sphaerisporangium genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Sphaerisporangium_genus_all$Treat<-factor(Sphaerisporangium_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Sphaerisporangium_genus_all,aes(x=Treat,y=Sphaerisporangium,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Sphaerisporangium Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Novosphingobium non_SEED genus

Novosphingobium_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Novosphingobium",]
Novosphingobium_genus_ab_up<-data.frame(Novosphingobium=Novosphingobium_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Novosphingobium_genus_ab_up)<-c('Novosphingobium','Treat')
head(Novosphingobium_genus_ab_up)

## Novosphingobium SEED

Novosphingobium_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Novosphingobium",]
Novosphingobium_genus_ab_SEED_up<-data.frame(Novosphingobium=Novosphingobium_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Novosphingobium_genus_ab_SEED_up)<-c('Novosphingobium','Treat')
head(Novosphingobium_genus_ab_SEED_up)

Novosphingobium_genus_all<-rbind(Novosphingobium_genus_ab_up,Novosphingobium_genus_ab_SEED_up)

#Novosphingobium_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Novosphingobium genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Novosphingobium_genus_all$Treat<-factor(Novosphingobium_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Novosphingobium_genus_all,aes(x=Treat,y=Novosphingobium,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Novosphingobium Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Enterobacter non_SEED genus

Enterobacter_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Enterobacter",]
Enterobacter_genus_ab_up<-data.frame(Enterobacter=Enterobacter_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Enterobacter_genus_ab_up)<-c('Enterobacter','Treat')
head(Enterobacter_genus_ab_up)

## Enterobacter SEED

Enterobacter_genus_ab_SEED_up<-data.frame(Enterobacter=0,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Enterobacter_genus_ab_SEED_up)<-c('Enterobacter','Treat')
head(Enterobacter_genus_ab_SEED_up)

Enterobacter_genus_all<-rbind(Enterobacter_genus_ab_up,Enterobacter_genus_ab_SEED_up)

#Enterobacter_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Enterobacter genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Enterobacter_genus_all$Treat<-factor(Enterobacter_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Enterobacter_genus_all,aes(x=Treat,y=Enterobacter,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Enterobacter Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Enterobacteriaceae_unclassified non_SEED genus

Enterobacteriaceae_unclassified_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Enterobacteriaceae_unclassified",]
Enterobacteriaceae_unclassified_genus_ab_up<-data.frame(Enterobacteriaceae_unclassified=Enterobacteriaceae_unclassified_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Enterobacteriaceae_unclassified_genus_ab_up)<-c('Enterobacteriaceae_unclassified','Treat')
head(Enterobacteriaceae_unclassified_genus_ab_up)

## Enterobacteriaceae_unclassified SEED

Enterobacteriaceae_unclassified_genus_ab_SEED_up<-data.frame(Enterobacteriaceae_unclassified=0,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Enterobacteriaceae_unclassified_genus_ab_SEED_up)<-c('Enterobacteriaceae_unclassified','Treat')
head(Enterobacteriaceae_unclassified_genus_ab_SEED_up)

Enterobacteriaceae_unclassified_genus_all<-rbind(Enterobacteriaceae_unclassified_genus_ab_up,Enterobacteriaceae_unclassified_genus_ab_SEED_up)

#Enterobacteriaceae_unclassified_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Enterobacteriaceae_unclassified genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Enterobacteriaceae_unclassified_genus_all$Treat<-factor(Enterobacteriaceae_unclassified_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Enterobacteriaceae_unclassified_genus_all,aes(x=Treat,y=Enterobacteriaceae_unclassified,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Enterobacteriaceae_unclassified Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Ralstonia non_SEED genus

Ralstonia_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Ralstonia",]
Ralstonia_genus_ab_up<-data.frame(Ralstonia=Ralstonia_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Ralstonia_genus_ab_up)<-c('Ralstonia','Treat')
head(Ralstonia_genus_ab_up)

## Ralstonia SEED

Ralstonia_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Ralstonia",]
Ralstonia_genus_ab_SEED_up<-data.frame(Ralstonia=Ralstonia_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Ralstonia_genus_ab_SEED_up)<-c('Ralstonia','Treat')
head(Ralstonia_genus_ab_SEED_up)

Ralstonia_genus_all<-rbind(Ralstonia_genus_ab_up,Ralstonia_genus_ab_SEED_up)

# Ralstonia_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Ralstonia genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Ralstonia_genus_all$Treat<-factor(Ralstonia_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Ralstonia_genus_all,aes(x=Treat,y=Ralstonia,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Ralstonia Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Sphingomonas non_SEED genus

Sphingomonas_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Sphingomonas",]
Sphingomonas_genus_ab_up<-data.frame(Sphingomonas=Sphingomonas_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Sphingomonas_genus_ab_up)<-c('Sphingomonas','Treat')
head(Sphingomonas_genus_ab_up)

## Sphingomonas SEED

Sphingomonas_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Sphingomonas",]
Sphingomonas_genus_ab_SEED_up<-data.frame(Sphingomonas=Sphingomonas_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Sphingomonas_genus_ab_SEED_up)<-c('Sphingomonas','Treat')
head(Sphingomonas_genus_ab_SEED_up)

Sphingomonas_genus_all<-rbind(Sphingomonas_genus_ab_up,Sphingomonas_genus_ab_SEED_up)

# Sphingomonas_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Sphingomonas genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Sphingomonas_genus_all$Treat<-factor(Sphingomonas_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Sphingomonas_genus_all,aes(x=Treat,y=Sphingomonas,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Sphingomonas Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Paenibacillus non_SEED genus

Paenibacillus_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Paenibacillus",]
Paenibacillus_genus_ab_up<-data.frame(Paenibacillus=Paenibacillus_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Paenibacillus_genus_ab_up)<-c('Paenibacillus','Treat')
head(Paenibacillus_genus_ab_up)

## Paenibacillus SEED

Paenibacillus_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Paenibacillus",]
Paenibacillus_genus_ab_SEED_up<-data.frame(Paenibacillus=Paenibacillus_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Paenibacillus_genus_ab_SEED_up)<-c('Paenibacillus','Treat')
head(Paenibacillus_genus_ab_SEED_up)

Paenibacillus_genus_all<-rbind(Paenibacillus_genus_ab_up,Paenibacillus_genus_ab_SEED_up)

#Paenibacillus_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Paenibacillus genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Paenibacillus_genus_all$Treat<-factor(Paenibacillus_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Paenibacillus_genus_all,aes(x=Treat,y=Paenibacillus,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Paenibacillus Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Subgroup_6_ge non_SEED genus

Subgroup_6_ge_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Subgroup_6_ge",]
Subgroup_6_ge_genus_ab_up<-data.frame(Subgroup_6_ge=Subgroup_6_ge_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Subgroup_6_ge_genus_ab_up)<-c('Subgroup_6_ge','Treat')
head(Subgroup_6_ge_genus_ab_up)

## Subgroup_6_ge SEED

Subgroup_6_ge_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Subgroup_6_ge",]
Subgroup_6_ge_genus_ab_SEED_up<-data.frame(Subgroup_6_ge=Subgroup_6_ge_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Subgroup_6_ge_genus_ab_SEED_up)<-c('Subgroup_6_ge','Treat')
head(Subgroup_6_ge_genus_ab_SEED_up)

Subgroup_6_ge_genus_all<-rbind(Subgroup_6_ge_genus_ab_up,Subgroup_6_ge_genus_ab_SEED_up)

#Subgroup_6_ge_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Subgroup_6_ge genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Subgroup_6_ge_genus_all$Treat<-factor(Subgroup_6_ge_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Subgroup_6_ge_genus_all,aes(x=Treat,y=Subgroup_6_ge,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Subgroup_6_ge Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Candidatus_Udaeobacter non_SEED genus

Candidatus_Udaeobacter_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Candidatus_Udaeobacter",]
Candidatus_Udaeobacter_genus_ab_up<-data.frame(Candidatus_Udaeobacter=Candidatus_Udaeobacter_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Candidatus_Udaeobacter_genus_ab_up)<-c('Candidatus_Udaeobacter','Treat')
head(Candidatus_Udaeobacter_genus_ab_up)

## Candidatus_Udaeobacter SEED

Candidatus_Udaeobacter_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Candidatus_Udaeobacter",]
Candidatus_Udaeobacter_genus_ab_SEED_up<-data.frame(Candidatus_Udaeobacter=Candidatus_Udaeobacter_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Candidatus_Udaeobacter_genus_ab_SEED_up)<-c('Candidatus_Udaeobacter','Treat')
head(Candidatus_Udaeobacter_genus_ab_SEED_up)

Candidatus_Udaeobacter_genus_all<-rbind(Candidatus_Udaeobacter_genus_ab_up,Candidatus_Udaeobacter_genus_ab_SEED_up)

#Candidatus_Udaeobacter_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Candidatus_Udaeobacter genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Candidatus_Udaeobacter_genus_all$Treat<-factor(Candidatus_Udaeobacter_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Candidatus_Udaeobacter_genus_all,aes(x=Treat,y=Candidatus_Udaeobacter,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Candidatus_Udaeobacter Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Bryobacter non_SEED genus

Bryobacter_genus_ab<-otu_table(r_seed_genus_up)@.Data[data.frame(tax_table(r_seed_genus_up)@.Data)$Genus=="Bryobacter",]
Bryobacter_genus_ab_up<-data.frame(Bryobacter=Bryobacter_genus_ab,Treat=sample_data(r_seed_genus_up)$Treat)
colnames(Bryobacter_genus_ab_up)<-c('Bryobacter','Treat')
head(Bryobacter_genus_ab_up)

## Bryobacter SEED

Bryobacter_genus_ab_SEED<-otu_table(r_SEED_genus)@.Data[data.frame(tax_table(r_SEED_genus)@.Data)$Genus=="Bryobacter",]
Bryobacter_genus_ab_SEED_up<-data.frame(Bryobacter=Bryobacter_genus_ab_SEED,Treat=sample_data(r_SEED_genus)$Treat)
colnames(Bryobacter_genus_ab_SEED_up)<-c('Bryobacter','Treat')
head(Bryobacter_genus_ab_SEED_up)

Bryobacter_genus_all<-rbind(Bryobacter_genus_ab_up,Bryobacter_genus_ab_SEED_up)

#Bryobacter_genus_ab
tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bryobacter genus abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")

  Bryobacter_genus_all$Treat<-factor(Bryobacter_genus_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Bryobacter_genus_all,aes(x=Treat,y=Bryobacter,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Bryobacter Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Family level barplot

r_seed_family_up<-tax_glom(r_seed_genus_up,taxrank = rank_names(r_seed_genus_up)[5],NArm = TRUE)
r_seed_family_up

r_SEED_genus
r_SEED_family<-tax_glom(r_SEED_genus,taxrank = rank_names(r_SEED_genus)[5],NArm = TRUE)

## Xanthobacteraceae non_SEED family

Xanthobacteraceae_family_ab<-otu_table(r_seed_family_up)@.Data[data.frame(tax_table(r_seed_family_up)@.Data[,1:5])$Family=="Xanthobacteraceae",]
Xanthobacteraceae_family_ab_up<-data.frame(Xanthobacteraceae=Xanthobacteraceae_family_ab,Treat=sample_data(r_seed_family_up)$Treat)
colnames(Xanthobacteraceae_family_ab_up)<-c('Xanthobacteraceae','Treat')
head(Xanthobacteraceae_family_ab_up)

## Xanthobacteraceae SEED family

Xanthobacteraceae_family_ab_SEED<-otu_table(r_SEED_family)@.Data[data.frame(tax_table(r_SEED_family)@.Data)[,1:5]$Family=="Xanthobacteraceae",]
Xanthobacteraceae_family_ab_SEED_up<-data.frame(Xanthobacteraceae=Xanthobacteraceae_family_ab_SEED,Treat=sample_data(r_SEED_family)$Treat)
colnames(Xanthobacteraceae_family_ab_SEED_up)<-c('Xanthobacteraceae','Treat')
head(Xanthobacteraceae_family_ab_SEED_up)

Xanthobacteraceae_family_all<-rbind(Xanthobacteraceae_family_ab_up,Xanthobacteraceae_family_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Xanthobacteraceae family abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Xanthobacteraceae_family_all$Treat<-factor(Xanthobacteraceae_family_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Xanthobacteraceae_family_all,aes(x=Treat,y=Xanthobacteraceae,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Xanthobacteraceae Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Sphingomonadaceae non_SEED family

Sphingomonadaceae_family_ab<-otu_table(r_seed_family_up)@.Data[data.frame(tax_table(r_seed_family_up)@.Data[,1:5])$Family=="Sphingomonadaceae",]
Sphingomonadaceae_family_ab_up<-data.frame(Sphingomonadaceae=Sphingomonadaceae_family_ab,Treat=sample_data(r_seed_family_up)$Treat)
colnames(Sphingomonadaceae_family_ab_up)<-c('Sphingomonadaceae','Treat')
head(Sphingomonadaceae_family_ab_up)

## Sphingomonadaceae SEED family

Sphingomonadaceae_family_ab_SEED<-otu_table(r_SEED_family)@.Data[data.frame(tax_table(r_SEED_family)@.Data)[,1:5]$Family=="Sphingomonadaceae",]
Sphingomonadaceae_family_ab_SEED_up<-data.frame(Sphingomonadaceae=Sphingomonadaceae_family_ab_SEED,Treat=sample_data(r_SEED_family)$Treat)
colnames(Sphingomonadaceae_family_ab_SEED_up)<-c('Sphingomonadaceae','Treat')
head(Sphingomonadaceae_family_ab_SEED_up)

Sphingomonadaceae_family_all<-rbind(Sphingomonadaceae_family_ab_up,Sphingomonadaceae_family_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Sphingomonadaceae family abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Sphingomonadaceae_family_all$Treat<-factor(Sphingomonadaceae_family_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Sphingomonadaceae_family_all,aes(x=Treat,y=Sphingomonadaceae,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Sphingomonadaceae Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

##Burkholderiaceae non_SEED family

Burkholderiaceae_family_ab<-otu_table(r_seed_family_up)@.Data[data.frame(tax_table(r_seed_family_up)@.Data[,1:5])$Family=="Burkholderiaceae",]
Burkholderiaceae_family_ab_up<-data.frame(Burkholderiaceae=Burkholderiaceae_family_ab,Treat=sample_data(r_seed_family_up)$Treat)
colnames(Burkholderiaceae_family_ab_up)<-c('Burkholderiaceae','Treat')
head(Burkholderiaceae_family_ab_up)

## Burkholderiaceae SEED family

Burkholderiaceae_family_ab_SEED<-otu_table(r_SEED_family)@.Data[data.frame(tax_table(r_SEED_family)@.Data)[,1:5]$Family=="Burkholderiaceae",]
Burkholderiaceae_family_ab_SEED_up<-data.frame(Burkholderiaceae=Burkholderiaceae_family_ab_SEED,Treat=sample_data(r_SEED_family)$Treat)
colnames(Burkholderiaceae_family_ab_SEED_up)<-c('Burkholderiaceae','Treat')
head(Burkholderiaceae_family_ab_SEED_up)

Burkholderiaceae_family_all<-rbind(Burkholderiaceae_family_ab_up,Burkholderiaceae_family_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Burkholderiaceae family abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Burkholderiaceae_family_all$Treat<-factor(Burkholderiaceae_family_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Burkholderiaceae_family_all,aes(x=Treat,y=Burkholderiaceae,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Burkholderiaceae Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


##Enterobacteriaceae non_SEED family

Enterobacteriaceae_family_ab<-otu_table(r_seed_family_up)@.Data[data.frame(tax_table(r_seed_family_up)@.Data[,1:5])$Family=="Enterobacteriaceae",]
Enterobacteriaceae_family_ab_up<-data.frame(Enterobacteriaceae=Enterobacteriaceae_family_ab,Treat=sample_data(r_seed_family_up)$Treat)
colnames(Enterobacteriaceae_family_ab_up)<-c('Enterobacteriaceae','Treat')
head(Enterobacteriaceae_family_ab_up)

## Enterobacteriaceae SEED family

Enterobacteriaceae_family_ab_SEED<-otu_table(r_SEED_family)@.Data[data.frame(tax_table(r_SEED_family)@.Data)[,1:5]$Family=="Enterobacteriaceae",]
Enterobacteriaceae_family_ab_SEED_up<-data.frame(Enterobacteriaceae=Enterobacteriaceae_family_ab_SEED,Treat=sample_data(r_SEED_family)$Treat)
colnames(Enterobacteriaceae_family_ab_SEED_up)<-c('Enterobacteriaceae','Treat')
head(Enterobacteriaceae_family_ab_SEED_up)

Enterobacteriaceae_family_all<-rbind(Enterobacteriaceae_family_ab_up,Enterobacteriaceae_family_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Enterobacteriaceae family abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Enterobacteriaceae_family_all$Treat<-factor(Enterobacteriaceae_family_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Enterobacteriaceae_family_all,aes(x=Treat,y=Enterobacteriaceae,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Enterobacteriaceae Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

##Bacillaceae non_SEED family

Bacillaceae_family_ab<-otu_table(r_seed_family_up)@.Data[data.frame(tax_table(r_seed_family_up)@.Data[,1:5])$Family=="Bacillaceae",]
Bacillaceae_family_ab_up<-data.frame(Bacillaceae=Bacillaceae_family_ab,Treat=sample_data(r_seed_family_up)$Treat)
colnames(Bacillaceae_family_ab_up)<-c('Bacillaceae','Treat')
head(Bacillaceae_family_ab_up)

## Bacillaceae SEED family

Bacillaceae_family_ab_SEED<-otu_table(r_SEED_family)@.Data[data.frame(tax_table(r_SEED_family)@.Data)[,1:5]$Family=="Bacillaceae",]
Bacillaceae_family_ab_SEED_up<-data.frame(Bacillaceae=Bacillaceae_family_ab_SEED,Treat=sample_data(r_SEED_family)$Treat)
colnames(Bacillaceae_family_ab_SEED_up)<-c('Bacillaceae','Treat')
head(Bacillaceae_family_ab_SEED_up)

Bacillaceae_family_all<-rbind(Bacillaceae_family_ab_up,Bacillaceae_family_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bacillaceae family abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Bacillaceae_family_all$Treat<-factor(Bacillaceae_family_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Bacillaceae_family_all,aes(x=Treat,y=Bacillaceae,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Bacillaceae Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

##Streptomycetaceae non_SEED family

Streptomycetaceae_family_ab<-otu_table(r_seed_family_up)@.Data[data.frame(tax_table(r_seed_family_up)@.Data[,1:5])$Family=="Streptomycetaceae",]
Streptomycetaceae_family_ab_up<-data.frame(Streptomycetaceae=Streptomycetaceae_family_ab,Treat=sample_data(r_seed_family_up)$Treat)
colnames(Streptomycetaceae_family_ab_up)<-c('Streptomycetaceae','Treat')
head(Streptomycetaceae_family_ab_up)

## Streptomycetaceae SEED family

Streptomycetaceae_family_ab_SEED<-otu_table(r_SEED_family)@.Data[data.frame(tax_table(r_SEED_family)@.Data)[,1:5]$Family=="Streptomycetaceae",]
Streptomycetaceae_family_ab_SEED_up<-data.frame(Streptomycetaceae=Streptomycetaceae_family_ab_SEED,Treat=sample_data(r_SEED_family)$Treat)
colnames(Streptomycetaceae_family_ab_SEED_up)<-c('Streptomycetaceae','Treat')
head(Streptomycetaceae_family_ab_SEED_up)

Streptomycetaceae_family_all<-rbind(Streptomycetaceae_family_ab_up,Streptomycetaceae_family_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Streptomycetaceae family abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Streptomycetaceae_family_all$Treat<-factor(Streptomycetaceae_family_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Streptomycetaceae_family_all,aes(x=Treat,y=Streptomycetaceae,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Streptomycetaceae Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


##Moraxellaceae non_SEED family

Moraxellaceae_family_ab<-otu_table(r_seed_family_up)@.Data[data.frame(tax_table(r_seed_family_up)@.Data[,1:5])$Family=="Moraxellaceae",]
Moraxellaceae_family_ab_up<-data.frame(Moraxellaceae=Moraxellaceae_family_ab,Treat=sample_data(r_seed_family_up)$Treat)
colnames(Moraxellaceae_family_ab_up)<-c('Moraxellaceae','Treat')
head(Moraxellaceae_family_ab_up)

## Moraxellaceae SEED family

Moraxellaceae_family_ab_SEED<-otu_table(r_SEED_family)@.Data[data.frame(tax_table(r_SEED_family)@.Data)[,1:5]$Family=="Moraxellaceae",]
Moraxellaceae_family_ab_SEED_up<-data.frame(Moraxellaceae=Moraxellaceae_family_ab_SEED,Treat=sample_data(r_SEED_family)$Treat)
colnames(Moraxellaceae_family_ab_SEED_up)<-c('Moraxellaceae','Treat')
head(Moraxellaceae_family_ab_SEED_up)

Moraxellaceae_family_all<-rbind(Moraxellaceae_family_ab_up,Moraxellaceae_family_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Moraxellaceae family abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Moraxellaceae_family_all$Treat<-factor(Moraxellaceae_family_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Moraxellaceae_family_all,aes(x=Treat,y=Moraxellaceae,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Moraxellaceae Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


##Pyrinomonadaceae non_SEED family

Pyrinomonadaceae_family_ab<-otu_table(r_seed_family_up)@.Data[data.frame(tax_table(r_seed_family_up)@.Data[,1:5])$Family=="Pyrinomonadaceae",]
Pyrinomonadaceae_family_ab_up<-data.frame(Pyrinomonadaceae=Pyrinomonadaceae_family_ab,Treat=sample_data(r_seed_family_up)$Treat)
colnames(Pyrinomonadaceae_family_ab_up)<-c('Pyrinomonadaceae','Treat')
head(Pyrinomonadaceae_family_ab_up)

## Pyrinomonadaceae SEED family

Pyrinomonadaceae_family_ab_SEED<-otu_table(r_SEED_family)@.Data[data.frame(tax_table(r_SEED_family)@.Data)[,1:5]$Family=="Pyrinomonadaceae",]
Pyrinomonadaceae_family_ab_SEED_up<-data.frame(Pyrinomonadaceae=Pyrinomonadaceae_family_ab_SEED,Treat=sample_data(r_SEED_family)$Treat)
colnames(Pyrinomonadaceae_family_ab_SEED_up)<-c('Pyrinomonadaceae','Treat')
head(Pyrinomonadaceae_family_ab_SEED_up)

Pyrinomonadaceae_family_all<-rbind(Pyrinomonadaceae_family_ab_up,Pyrinomonadaceae_family_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Pyrinomonadaceae family abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Pyrinomonadaceae_family_all$Treat<-factor(Pyrinomonadaceae_family_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Pyrinomonadaceae_family_all,aes(x=Treat,y=Pyrinomonadaceae,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Pyrinomonadaceae Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


##Chitinophagaceae non_SEED family

Chitinophagaceae_family_ab<-otu_table(r_seed_family_up)@.Data[data.frame(tax_table(r_seed_family_up)@.Data[,1:5])$Family=="Chitinophagaceae",]
Chitinophagaceae_family_ab_up<-data.frame(Chitinophagaceae=Chitinophagaceae_family_ab,Treat=sample_data(r_seed_family_up)$Treat)
colnames(Chitinophagaceae_family_ab_up)<-c('Chitinophagaceae','Treat')
head(Chitinophagaceae_family_ab_up)

## Chitinophagaceae SEED family

Chitinophagaceae_family_ab_SEED<-otu_table(r_SEED_family)@.Data[data.frame(tax_table(r_SEED_family)@.Data)[,1:5]$Family=="Chitinophagaceae",]
Chitinophagaceae_family_ab_SEED_up<-data.frame(Chitinophagaceae=Chitinophagaceae_family_ab_SEED,Treat=sample_data(r_SEED_family)$Treat)
colnames(Chitinophagaceae_family_ab_SEED_up)<-c('Chitinophagaceae','Treat')
head(Chitinophagaceae_family_ab_SEED_up)

Chitinophagaceae_family_all<-rbind(Chitinophagaceae_family_ab_up,Chitinophagaceae_family_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Chitinophagaceae family abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Chitinophagaceae_family_all$Treat<-factor(Chitinophagaceae_family_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Chitinophagaceae_family_all,aes(x=Treat,y=Chitinophagaceae,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Chitinophagaceae Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## order level
r_seed_order_up<-tax_glom(r_seed_genus_up,taxrank = rank_names(r_seed_genus_up)[4],NArm = TRUE)
r_SEED_order<-tax_glom(r_SEED_genus,taxrank = rank_names(r_SEED_genus)[4],NArm = TRUE)

## Enterobacteriales SEED order

Enterobacteriales_order_ab_SEED<-otu_table(r_SEED_order)@.Data[data.frame(tax_table(r_SEED_order)@.Data[,1:4])$Order=="Enterobacteriales",]
Enterobacteriales_order_ab_SEED_up<-data.frame(Enterobacteriales=Enterobacteriales_order_ab_SEED,Treat=sample_data(r_SEED_order)$Treat)
colnames(Enterobacteriales_order_ab_SEED_up)<-c('Enterobacteriales','Treat')
head(Enterobacteriales_order_ab_SEED_up)

## Enterobacteriales non_SEED order

Enterobacteriales_order_ab<-otu_table(r_seed_order_up)@.Data[data.frame(tax_table(r_seed_order_up)@.Data[,1:4])$Order=="Enterobacteriales",]
Enterobacteriales_order_ab_up<-data.frame(Enterobacteriales=Enterobacteriales_order_ab,Treat=sample_data(r_seed_order_up)$Treat)
colnames(Enterobacteriales_order_ab_up)<-c('Enterobacteriales','Treat')
head(Enterobacteriales_order_ab_up)

Enterobacteriales_order_all<-rbind(Enterobacteriales_order_ab_up,Enterobacteriales_order_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Enterobacteriales order abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Enterobacteriales_order_all$Treat<-factor(Enterobacteriales_order_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Enterobacteriales_order_all,aes(x=Treat,y=Enterobacteriales,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Enterobacteriales Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Pseudomonadales SEED order

Pseudomonadales_order_ab_SEED<-otu_table(r_SEED_order)@.Data[data.frame(tax_table(r_SEED_order)@.Data[,1:4])$Order=="Pseudomonadales",]
Pseudomonadales_order_ab_SEED_up<-data.frame(Pseudomonadales=Pseudomonadales_order_ab_SEED,Treat=sample_data(r_SEED_order)$Treat)
colnames(Pseudomonadales_order_ab_SEED_up)<-c('Pseudomonadales','Treat')
head(Pseudomonadales_order_ab_SEED_up)

## Pseudomonadales non_SEED order

Pseudomonadales_order_ab<-otu_table(r_seed_order_up)@.Data[data.frame(tax_table(r_seed_order_up)@.Data[,1:4])$Order=="Pseudomonadales",]
Pseudomonadales_order_ab_up<-data.frame(Pseudomonadales=Pseudomonadales_order_ab,Treat=sample_data(r_seed_order_up)$Treat)
colnames(Pseudomonadales_order_ab_up)<-c('Pseudomonadales','Treat')
head(Pseudomonadales_order_ab_up)

Pseudomonadales_order_all<-rbind(Pseudomonadales_order_ab_up,Pseudomonadales_order_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Pseudomonadales order abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Pseudomonadales_order_all$Treat<-factor(Pseudomonadales_order_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Pseudomonadales_order_all,aes(x=Treat,y=Pseudomonadales,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Pseudomonadales Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Rhizobiales SEED order

Rhizobiales_order_ab_SEED<-otu_table(r_SEED_order)@.Data[data.frame(tax_table(r_SEED_order)@.Data[,1:4])$Order=="Rhizobiales",]
Rhizobiales_order_ab_SEED_up<-data.frame(Rhizobiales=Rhizobiales_order_ab_SEED,Treat=sample_data(r_SEED_order)$Treat)
colnames(Rhizobiales_order_ab_SEED_up)<-c('Rhizobiales','Treat')
head(Rhizobiales_order_ab_SEED_up)

## Rhizobiales non_SEED order

Rhizobiales_order_ab<-otu_table(r_seed_order_up)@.Data[data.frame(tax_table(r_seed_order_up)@.Data[,1:4])$Order=="Rhizobiales",]
Rhizobiales_order_ab_up<-data.frame(Rhizobiales=Rhizobiales_order_ab,Treat=sample_data(r_seed_order_up)$Treat)
colnames(Rhizobiales_order_ab_up)<-c('Rhizobiales','Treat')
head(Rhizobiales_order_ab_up)

Rhizobiales_order_all<-rbind(Rhizobiales_order_ab_up,Rhizobiales_order_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Rhizobiales order abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Rhizobiales_order_all$Treat<-factor(Rhizobiales_order_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Rhizobiales_order_all,aes(x=Treat,y=Rhizobiales,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Rhizobiales Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Sphingomonadales SEED order

Sphingomonadales_order_ab_SEED<-otu_table(r_SEED_order)@.Data[data.frame(tax_table(r_SEED_order)@.Data[,1:4])$Order=="Sphingomonadales",]
Sphingomonadales_order_ab_SEED_up<-data.frame(Sphingomonadales=Sphingomonadales_order_ab_SEED,Treat=sample_data(r_SEED_order)$Treat)
colnames(Sphingomonadales_order_ab_SEED_up)<-c('Sphingomonadales','Treat')
head(Sphingomonadales_order_ab_SEED_up)

## Sphingomonadales non_SEED order

Sphingomonadales_order_ab<-otu_table(r_seed_order_up)@.Data[data.frame(tax_table(r_seed_order_up)@.Data[,1:4])$Order=="Sphingomonadales",]
Sphingomonadales_order_ab_up<-data.frame(Sphingomonadales=Sphingomonadales_order_ab,Treat=sample_data(r_seed_order_up)$Treat)
colnames(Sphingomonadales_order_ab_up)<-c('Sphingomonadales','Treat')
head(Sphingomonadales_order_ab_up)

Sphingomonadales_order_all<-rbind(Sphingomonadales_order_ab_up,Sphingomonadales_order_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Sphingomonadales order abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Sphingomonadales_order_all$Treat<-factor(Sphingomonadales_order_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Sphingomonadales_order_all,aes(x=Treat,y=Sphingomonadales,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, positcion=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Sphingomonadales Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Betaproteobacteriales SEED order

Betaproteobacteriales_order_ab_SEED<-otu_table(r_SEED_order)@.Data[data.frame(tax_table(r_SEED_order)@.Data[,1:4])$Order=="Betaproteobacteriales",]
Betaproteobacteriales_order_ab_SEED_up<-data.frame(Betaproteobacteriales=Betaproteobacteriales_order_ab_SEED,Treat=sample_data(r_SEED_order)$Treat)
colnames(Betaproteobacteriales_order_ab_SEED_up)<-c('Betaproteobacteriales','Treat')
head(Betaproteobacteriales_order_ab_SEED_up)

## Betaproteobacteriales non_SEED order

Betaproteobacteriales_order_ab<-otu_table(r_seed_order_up)@.Data[data.frame(tax_table(r_seed_order_up)@.Data[,1:4])$Order=="Betaproteobacteriales",]
Betaproteobacteriales_order_ab_up<-data.frame(Betaproteobacteriales=Betaproteobacteriales_order_ab,Treat=sample_data(r_seed_order_up)$Treat)
colnames(Betaproteobacteriales_order_ab_up)<-c('Betaproteobacteriales','Treat')
head(Betaproteobacteriales_order_ab_up)

Betaproteobacteriales_order_all<-rbind(Betaproteobacteriales_order_ab_up,Betaproteobacteriales_order_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Betaproteobacteriales order abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Betaproteobacteriales_order_all$Treat<-factor(Betaproteobacteriales_order_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Betaproteobacteriales_order_all,aes(x=Treat,y=Betaproteobacteriales,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Betaproteobacteriales Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Streptomycetales SEED order

Streptomycetales_order_ab_SEED<-otu_table(r_SEED_order)@.Data[data.frame(tax_table(r_SEED_order)@.Data[,1:4])$Order=="Streptomycetales",]
Streptomycetales_order_ab_SEED_up<-data.frame(Streptomycetales=Streptomycetales_order_ab_SEED,Treat=sample_data(r_SEED_order)$Treat)
colnames(Streptomycetales_order_ab_SEED_up)<-c('Streptomycetales','Treat')
head(Streptomycetales_order_ab_SEED_up)

## Streptomycetales non_SEED order

Streptomycetales_order_ab<-otu_table(r_seed_order_up)@.Data[data.frame(tax_table(r_seed_order_up)@.Data[,1:4])$Order=="Streptomycetales",]
Streptomycetales_order_ab_up<-data.frame(Streptomycetales=Streptomycetales_order_ab,Treat=sample_data(r_seed_order_up)$Treat)
colnames(Streptomycetales_order_ab_up)<-c('Streptomycetales','Treat')
head(Streptomycetales_order_ab_up)

Streptomycetales_order_all<-rbind(Streptomycetales_order_ab_up,Streptomycetales_order_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Streptomycetales order abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Streptomycetales_order_all$Treat<-factor(Streptomycetales_order_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Streptomycetales_order_all,aes(x=Treat,y=Streptomycetales,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Streptomycetales Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Solirubrobacterales SEED order

Solirubrobacterales_order_ab_SEED<-otu_table(r_SEED_order)@.Data[data.frame(tax_table(r_SEED_order)@.Data[,1:4])$Order=="Solirubrobacterales",]
Solirubrobacterales_order_ab_SEED_up<-data.frame(Solirubrobacterales=Solirubrobacterales_order_ab_SEED,Treat=sample_data(r_SEED_order)$Treat)
colnames(Solirubrobacterales_order_ab_SEED_up)<-c('Solirubrobacterales','Treat')
head(Solirubrobacterales_order_ab_SEED_up)

## Solirubrobacterales non_SEED order

Solirubrobacterales_order_ab<-otu_table(r_seed_order_up)@.Data[data.frame(tax_table(r_seed_order_up)@.Data[,1:4])$Order=="Solirubrobacterales",]
Solirubrobacterales_order_ab_up<-data.frame(Solirubrobacterales=Solirubrobacterales_order_ab,Treat=sample_data(r_seed_order_up)$Treat)
colnames(Solirubrobacterales_order_ab_up)<-c('Solirubrobacterales','Treat')
head(Solirubrobacterales_order_ab_up)

Solirubrobacterales_order_all<-rbind(Solirubrobacterales_order_ab_up,Solirubrobacterales_order_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Solirubrobacterales order abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Solirubrobacterales_order_all$Treat<-factor(Solirubrobacterales_order_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Solirubrobacterales_order_all,aes(x=Treat,y=Solirubrobacterales,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Solirubrobacterales Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Class level
r_seed_class_up<-tax_glom(r_seed_genus_up,taxrank = rank_names(r_seed_genus_up)[3],NArm = TRUE)
r_SEED_class<-tax_glom(r_SEED_genus,taxrank = rank_names(r_SEED_genus)[3],NArm = TRUE)

## Alphaproteobacteria SEED class

Alphaproteobacteria_class_ab_SEED<-otu_table(r_SEED_class)@.Data[data.frame(tax_table(r_SEED_class)@.Data[,1:3])$Class=="Alphaproteobacteria",]
Alphaproteobacteria_class_ab_SEED_up<-data.frame(Alphaproteobacteria=Alphaproteobacteria_class_ab_SEED,Treat=sample_data(r_SEED_class)$Treat)
colnames(Alphaproteobacteria_class_ab_SEED_up)<-c('Alphaproteobacteria','Treat')
head(Alphaproteobacteria_class_ab_SEED_up)

## Alphaproteobacteria non_SEED class
Alphaproteobacteria_class_ab<-otu_table(r_seed_class_up)@.Data[data.frame(tax_table(r_seed_class_up)@.Data[,1:3])$Class=="Alphaproteobacteria",]
Alphaproteobacteria_class_ab_up<-data.frame(Alphaproteobacteria=Alphaproteobacteria_class_ab,Treat=sample_data(r_seed_class_up)$Treat)
colnames(Alphaproteobacteria_class_ab_up)<-c('Alphaproteobacteria','Treat')
head(Alphaproteobacteria_class_ab_up)
Alphaproteobacteria_class_all<-rbind(Alphaproteobacteria_class_ab_up,Alphaproteobacteria_class_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Alphaproteobacteria class abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Alphaproteobacteria_class_all$Treat<-factor(Alphaproteobacteria_class_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))

ggplot(Alphaproteobacteria_class_all,aes(x=Treat,y=Alphaproteobacteria,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Alphaproteobacteria Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Gammaproteobacteria SEED class

Gammaproteobacteria_class_ab_SEED<-otu_table(r_SEED_class)@.Data[data.frame(tax_table(r_SEED_class)@.Data[,1:3])$Class=="Gammaproteobacteria",]
Gammaproteobacteria_class_ab_SEED_up<-data.frame(Gammaproteobacteria=Gammaproteobacteria_class_ab_SEED,Treat=sample_data(r_SEED_class)$Treat)
colnames(Gammaproteobacteria_class_ab_SEED_up)<-c('Gammaproteobacteria','Treat')
head(Gammaproteobacteria_class_ab_SEED_up)

## Gammaproteobacteria non_SEED class
Gammaproteobacteria_class_ab<-otu_table(r_seed_class_up)@.Data[data.frame(tax_table(r_seed_class_up)@.Data[,1:3])$Class=="Gammaproteobacteria",]
Gammaproteobacteria_class_ab_up<-data.frame(Gammaproteobacteria=Gammaproteobacteria_class_ab,Treat=sample_data(r_seed_class_up)$Treat)
colnames(Gammaproteobacteria_class_ab_up)<-c('Gammaproteobacteria','Treat')
head(Gammaproteobacteria_class_ab_up)
Gammaproteobacteria_class_all<-rbind(Gammaproteobacteria_class_ab_up,Gammaproteobacteria_class_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Gammaproteobacteria class abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Gammaproteobacteria_class_all$Treat<-factor(Gammaproteobacteria_class_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))

ggplot(Gammaproteobacteria_class_all,aes(x=Treat,y=Gammaproteobacteria,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Gammaproteobacteria Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Deltaproteobacteria SEED class

Deltaproteobacteria_class_ab_SEED<-otu_table(r_SEED_class)@.Data[data.frame(tax_table(r_SEED_class)@.Data[,1:3])$Class=="Deltaproteobacteria",]
Deltaproteobacteria_class_ab_SEED_up<-data.frame(Deltaproteobacteria=Deltaproteobacteria_class_ab_SEED,Treat=sample_data(r_SEED_class)$Treat)
colnames(Deltaproteobacteria_class_ab_SEED_up)<-c('Deltaproteobacteria','Treat')
head(Deltaproteobacteria_class_ab_SEED_up)

## Deltaproteobacteria non_SEED class
Deltaproteobacteria_class_ab<-otu_table(r_seed_class_up)@.Data[data.frame(tax_table(r_seed_class_up)@.Data[,1:3])$Class=="Deltaproteobacteria",]
Deltaproteobacteria_class_ab_up<-data.frame(Deltaproteobacteria=Deltaproteobacteria_class_ab,Treat=sample_data(r_seed_class_up)$Treat)
colnames(Deltaproteobacteria_class_ab_up)<-c('Deltaproteobacteria','Treat')
head(Deltaproteobacteria_class_ab_up)
Deltaproteobacteria_class_all<-rbind(Deltaproteobacteria_class_ab_up,Deltaproteobacteria_class_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Deltaproteobacteria class abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Deltaproteobacteria_class_all$Treat<-factor(Deltaproteobacteria_class_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))

ggplot(Deltaproteobacteria_class_all,aes(x=Treat,y=Deltaproteobacteria,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Deltaproteobacteria Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Phylum level
r_seed_phylum_up<-tax_glom(r_seed_genus_up,taxrank = rank_names(r_seed_genus_up)[2],NArm = TRUE)
r_SEED_phylum<-tax_glom(r_SEED_genus,taxrank = rank_names(r_SEED_genus)[2],NArm = TRUE)

## Firmicutes SEED phylum
Firmicutes_phylum_ab_SEED<-otu_table(r_SEED_phylum)@.Data[data.frame(tax_table(r_SEED_phylum)@.Data[,1:2])$Phylum=="Firmicutes",]
Firmicutes_phylum_ab_SEED_up<-data.frame(Firmicutes=Firmicutes_phylum_ab_SEED,Treat=sample_data(r_SEED_phylum)$Treat)
colnames(Firmicutes_phylum_ab_SEED_up)<-c('Firmicutes','Treat')
head(Firmicutes_phylum_ab_SEED_up)

## Firmicutes non_SEED phylum
Firmicutes_phylum_ab<-otu_table(r_seed_phylum_up)@.Data[data.frame(tax_table(r_seed_phylum_up)@.Data[,1:2])$Phylum=="Firmicutes",]
Firmicutes_phylum_ab_up<-data.frame(Firmicutes=Firmicutes_phylum_ab,Treat=sample_data(r_seed_phylum_up)$Treat)
colnames(Firmicutes_phylum_ab_up)<-c('Firmicutes','Treat')
head(Firmicutes_phylum_ab_up)

Firmicutes_phylum_all<-rbind(Firmicutes_phylum_ab_up,Firmicutes_phylum_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Firmicutes phylum abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Firmicutes_phylum_all$Treat<-factor(Firmicutes_phylum_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Firmicutes_phylum_all,aes(x=Treat,y=Firmicutes,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Firmicutes Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Verrucomicrobia SEED phylum
Verrucomicrobia_phylum_ab_SEED<-otu_table(r_SEED_phylum)@.Data[data.frame(tax_table(r_SEED_phylum)@.Data[,1:2])$Phylum=="Verrucomicrobia",]
Verrucomicrobia_phylum_ab_SEED_up<-data.frame(Verrucomicrobia=Verrucomicrobia_phylum_ab_SEED,Treat=sample_data(r_SEED_phylum)$Treat)
colnames(Verrucomicrobia_phylum_ab_SEED_up)<-c('Verrucomicrobia','Treat')
head(Verrucomicrobia_phylum_ab_SEED_up)

## Verrucomicrobia non_SEED phylum
Verrucomicrobia_phylum_ab<-otu_table(r_seed_phylum_up)@.Data[data.frame(tax_table(r_seed_phylum_up)@.Data[,1:2])$Phylum=="Verrucomicrobia",]
Verrucomicrobia_phylum_ab_up<-data.frame(Verrucomicrobia=Verrucomicrobia_phylum_ab,Treat=sample_data(r_seed_phylum_up)$Treat)
colnames(Verrucomicrobia_phylum_ab_up)<-c('Verrucomicrobia','Treat')
head(Verrucomicrobia_phylum_ab_up)

Verrucomicrobia_phylum_all<-rbind(Verrucomicrobia_phylum_ab_up,Verrucomicrobia_phylum_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Verrucomicrobia phylum abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Verrucomicrobia_phylum_all$Treat<-factor(Verrucomicrobia_phylum_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Verrucomicrobia_phylum_all,aes(x=Treat,y=Verrucomicrobia,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Verrucomicrobia Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Planctomycetes

## Planctomycetes SEED phylum
Planctomycetes_phylum_ab_SEED<-otu_table(r_SEED_phylum)@.Data[data.frame(tax_table(r_SEED_phylum)@.Data[,1:2])$Phylum=="Planctomycetes",]
Planctomycetes_phylum_ab_SEED_up<-data.frame(Planctomycetes=Planctomycetes_phylum_ab_SEED,Treat=sample_data(r_SEED_phylum)$Treat)
colnames(Planctomycetes_phylum_ab_SEED_up)<-c('Planctomycetes','Treat')
head(Planctomycetes_phylum_ab_SEED_up)

## Planctomycetes non_SEED phylum
Planctomycetes_phylum_ab<-otu_table(r_seed_phylum_up)@.Data[data.frame(tax_table(r_seed_phylum_up)@.Data[,1:2])$Phylum=="Planctomycetes",]
Planctomycetes_phylum_ab_up<-data.frame(Planctomycetes=Planctomycetes_phylum_ab,Treat=sample_data(r_seed_phylum_up)$Treat)
colnames(Planctomycetes_phylum_ab_up)<-c('Planctomycetes','Treat')
head(Planctomycetes_phylum_ab_up)

Planctomycetes_phylum_all<-rbind(Planctomycetes_phylum_ab_up,Planctomycetes_phylum_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Planctomycetes phylum abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Planctomycetes_phylum_all$Treat<-factor(Planctomycetes_phylum_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Planctomycetes_phylum_all,aes(x=Treat,y=Planctomycetes,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Planctomycetes Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Gemmatimonadetes SEED phylum
Gemmatimonadetes_phylum_ab_SEED<-otu_table(r_SEED_phylum)@.Data[data.frame(tax_table(r_SEED_phylum)@.Data[,1:2])$Phylum=="Gemmatimonadetes",]
Gemmatimonadetes_phylum_ab_SEED_up<-data.frame(Gemmatimonadetes=Gemmatimonadetes_phylum_ab_SEED,Treat=sample_data(r_SEED_phylum)$Treat)
colnames(Gemmatimonadetes_phylum_ab_SEED_up)<-c('Gemmatimonadetes','Treat')
head(Gemmatimonadetes_phylum_ab_SEED_up)

## Gemmatimonadetes non_SEED phylum
Gemmatimonadetes_phylum_ab<-otu_table(r_seed_phylum_up)@.Data[data.frame(tax_table(r_seed_phylum_up)@.Data[,1:2])$Phylum=="Gemmatimonadetes",]
Gemmatimonadetes_phylum_ab_up<-data.frame(Gemmatimonadetes=Gemmatimonadetes_phylum_ab,Treat=sample_data(r_seed_phylum_up)$Treat)
colnames(Gemmatimonadetes_phylum_ab_up)<-c('Gemmatimonadetes','Treat')
head(Gemmatimonadetes_phylum_ab_up)

Gemmatimonadetes_phylum_all<-rbind(Gemmatimonadetes_phylum_ab_up,Gemmatimonadetes_phylum_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Gemmatimonadetes phylum abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Gemmatimonadetes_phylum_all$Treat<-factor(Gemmatimonadetes_phylum_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Gemmatimonadetes_phylum_all,aes(x=Treat,y=Gemmatimonadetes,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Gemmatimonadetes Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Proteobacteria SEED phylum
Proteobacteria_phylum_ab_SEED<-otu_table(r_SEED_phylum)@.Data[data.frame(tax_table(r_SEED_phylum)@.Data[,1:2])$Phylum=="Proteobacteria",]
Proteobacteria_phylum_ab_SEED_up<-data.frame(Proteobacteria=Proteobacteria_phylum_ab_SEED,Treat=sample_data(r_SEED_phylum)$Treat)
colnames(Proteobacteria_phylum_ab_SEED_up)<-c('Proteobacteria','Treat')
head(Proteobacteria_phylum_ab_SEED_up)

## Proteobacteria non_SEED phylum
Proteobacteria_phylum_ab<-otu_table(r_seed_phylum_up)@.Data[data.frame(tax_table(r_seed_phylum_up)@.Data[,1:2])$Phylum=="Proteobacteria",]
Proteobacteria_phylum_ab_up<-data.frame(Proteobacteria=Proteobacteria_phylum_ab,Treat=sample_data(r_seed_phylum_up)$Treat)
colnames(Proteobacteria_phylum_ab_up)<-c('Proteobacteria','Treat')
head(Proteobacteria_phylum_ab_up)

Proteobacteria_phylum_all<-rbind(Proteobacteria_phylum_ab_up,Proteobacteria_phylum_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Proteobacteria phylum abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Proteobacteria_phylum_all$Treat<-factor(Proteobacteria_phylum_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Proteobacteria_phylum_all,aes(x=Treat,y=Proteobacteria,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Proteobacteria Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Bacteroidetes SEED phylum
Bacteroidetes_phylum_ab_SEED<-otu_table(r_SEED_phylum)@.Data[data.frame(tax_table(r_SEED_phylum)@.Data[,1:2])$Phylum=="Bacteroidetes",]
Bacteroidetes_phylum_ab_SEED_up<-data.frame(Bacteroidetes=Bacteroidetes_phylum_ab_SEED,Treat=sample_data(r_SEED_phylum)$Treat)
colnames(Bacteroidetes_phylum_ab_SEED_up)<-c('Bacteroidetes','Treat')
head(Bacteroidetes_phylum_ab_SEED_up)

## Bacteroidetes non_SEED phylum
Bacteroidetes_phylum_ab<-otu_table(r_seed_phylum_up)@.Data[data.frame(tax_table(r_seed_phylum_up)@.Data[,1:2])$Phylum=="Bacteroidetes",]
Bacteroidetes_phylum_ab_up<-data.frame(Bacteroidetes=Bacteroidetes_phylum_ab,Treat=sample_data(r_seed_phylum_up)$Treat)
colnames(Bacteroidetes_phylum_ab_up)<-c('Bacteroidetes','Treat')
head(Bacteroidetes_phylum_ab_up)

Bacteroidetes_phylum_all<-rbind(Bacteroidetes_phylum_ab_up,Bacteroidetes_phylum_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Bacteroidetes phylum abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Bacteroidetes_phylum_all$Treat<-factor(Bacteroidetes_phylum_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Bacteroidetes_phylum_all,aes(x=Treat,y=Bacteroidetes,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Bacteroidetes Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()


## Acidobacteria SEED phylum
Acidobacteria_phylum_ab_SEED<-otu_table(r_SEED_phylum)@.Data[data.frame(tax_table(r_SEED_phylum)@.Data[,1:2])$Phylum=="Acidobacteria",]
Acidobacteria_phylum_ab_SEED_up<-data.frame(Acidobacteria=Acidobacteria_phylum_ab_SEED,Treat=sample_data(r_SEED_phylum)$Treat)
colnames(Acidobacteria_phylum_ab_SEED_up)<-c('Acidobacteria','Treat')
head(Acidobacteria_phylum_ab_SEED_up)

## Acidobacteria non_SEED phylum
Acidobacteria_phylum_ab<-otu_table(r_seed_phylum_up)@.Data[data.frame(tax_table(r_seed_phylum_up)@.Data[,1:2])$Phylum=="Acidobacteria",]
Acidobacteria_phylum_ab_up<-data.frame(Acidobacteria=Acidobacteria_phylum_ab,Treat=sample_data(r_seed_phylum_up)$Treat)
colnames(Acidobacteria_phylum_ab_up)<-c('Acidobacteria','Treat')
head(Acidobacteria_phylum_ab_up)

Acidobacteria_phylum_all<-rbind(Acidobacteria_phylum_ab_up,Acidobacteria_phylum_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Acidobacteria phylum abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Acidobacteria_phylum_all$Treat<-factor(Acidobacteria_phylum_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Acidobacteria_phylum_all,aes(x=Treat,y=Acidobacteria,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Acidobacteria Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()

## Actinobacteria

## Actinobacteria SEED phylum
Actinobacteria_phylum_ab_SEED<-otu_table(r_SEED_phylum)@.Data[data.frame(tax_table(r_SEED_phylum)@.Data[,1:2])$Phylum=="Actinobacteria",]
Actinobacteria_phylum_ab_SEED_up<-data.frame(Actinobacteria=Actinobacteria_phylum_ab_SEED,Treat=sample_data(r_SEED_phylum)$Treat)
colnames(Actinobacteria_phylum_ab_SEED_up)<-c('Actinobacteria','Treat')
head(Actinobacteria_phylum_ab_SEED_up)

## Actinobacteria non_SEED phylum
Actinobacteria_phylum_ab<-otu_table(r_seed_phylum_up)@.Data[data.frame(tax_table(r_seed_phylum_up)@.Data[,1:2])$Phylum=="Actinobacteria",]
Actinobacteria_phylum_ab_up<-data.frame(Actinobacteria=Actinobacteria_phylum_ab,Treat=sample_data(r_seed_phylum_up)$Treat)
colnames(Actinobacteria_phylum_ab_up)<-c('Actinobacteria','Treat')
head(Actinobacteria_phylum_ab_up)

Actinobacteria_phylum_all<-rbind(Actinobacteria_phylum_ab_up,Actinobacteria_phylum_ab_SEED_up)

tiff('/Users/fangliu/Documents/2018_fugicide_project/Seed_treatment/16S/Seed_16S_Community_analysis/Silva_tax_based_results/rarefaction_13021/r_output/Actinobacteria phylum abundance boxplot.tiff', units="in", width=5, height=5, res=300,compression = 'lzw',bg="White")
Actinobacteria_phylum_all$Treat<-factor(Actinobacteria_phylum_all$Treat,levels = c('Soil','Bulk_1wk','Bulk_3wk','Bulk_4wk','Rhi_1wk','Rhi_3wk','Rhi_4wk','Endo_1wk','Endo_3wk','Endo_4wk','SEED'))
ggplot(Actinobacteria_phylum_all,aes(x=Treat,y=Actinobacteria,fill=Treat))+geom_boxplot(size=0.8)+
geom_jitter(shape=16, position=position_jitter(width = 0,height = 0),aes(fill=Treat))+scale_fill_manual(values = c(as.character(Treat_colors),"#ffee32"))+labs(title="Actinobacteria Relative Abundance",x="Treatment",y="Relative_abundance")+theme(legend.position="none",panel.background = element_rect(fill = "white",color="black"),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),plot.title=(element_text(size=13,family="Arial",face="bold",vjust = 3,hjust=0.5)),text=element_text(family="Arial",face="bold"),panel.border = element_rect(colour = "black", fill=NA, size=0.5),axis.text.y=element_text(size=12,family="Arial"),axis.text.x = element_text(family="Arial",size=13,angle = 90),axis.title=element_text(size = 15,face="bold",family="Arial",vjust = 1),legend.title = element_text(size=13,face="bold",family="Arial"),legend.text = (element_text(size=10,family="Arial")),plot.margin = unit(c(1,1,1,1), "cm"))
dev.off()
```


## Bacteria OTU overlap between treatments


```{r}
join_Seed_phyloseq_up<-subset_samples(join_Seed_phyloseq_up,Sample_ID!="Seed_CM_4")
join_Seed_phyloseq_up_merge<-merge_samples(join_Seed_phyloseq_up,group = "Treat",fun = mean)
join_Seed_phyloseq_up_merge_OTU<-data.frame(Treat=rownames(sample_data(join_Seed_phyloseq_up_merge)), otu_table(join_Seed_phyloseq_up_merge)@.Data)
join_Seed_phyloseq_up_merge_OTU[1:11,1:5]

Seed_to_Endowk1<-filter(join_Seed_phyloseq_up_merge_OTU,SEED>0&)
```
